---
title: "Congo_Mekong_CT_100ASV"
author: "Ted Bambakidis"
date: "12/11/2023"
output: html_document
---

# frequently used packages
```{r}
library(BiocManager)
library(phyloseq)
library(tidyverse)
library(ggplot2)
library(ggthemes)
library(vegan)
#library(MASS)
#library(cluster)
require(RColorBrewer)
#library(grid)
#library(cooccur)
#library(minerva)
#library(parallel)
#library(tidyr)
#library(forcats)
#library(usdm)
#library(cowplot)
#library(psych)
#library(dlookr)
#library(grDevices)
#library(scales)
#library(devtools)
#library(ggfortify) # for PCA autoplot
#library(tibble)
library(rbiom)
library(ape)
#library(shadowtext)
library(ade4)
#library(tidytree)
library(ggrepel)
library(microViz)
library(ggpubr)
library(apeglm)
```

# make phyloseq object from qiime2 exports - 100 ASVs
```{r}
OTU_table = read.table("ct_mek_con_merged_table_20230906_ed_100.txt", sep = "\t", header = TRUE, row.names = 1)
otu_mat = data.matrix(OTU_table)
OTU = otu_table(otu_mat, taxa_are_rows = TRUE)

tax_file = read.table("taxonomy_ed.tsv", sep = "\t", header = TRUE, row.names = 1)
tax_mat = as.matrix(tax_file)
TAX = tax_table(tax_mat)

treefile = "tree.nwk"
tree = read_tree(treefile)

designfile = read.csv("../consolidated_metadata_mek_con_ct.csv", row.names=1)
designfile$sample_id <- row.names(designfile)
design = sample_data(data.frame(designfile))

community = merge_phyloseq(OTU, TAX, design, tree)
print(community)
rank_names(community)
sample_variables(community)
summary(sample_sums(community))
```

# filter out chloroplasts, mitochondria, and singletons
```{r}
community = subset_taxa(community, Order!= "Chloroplast") # order
community = subset_taxa(community, Family!= "Mitochondria") # use family
community <- prune_taxa(taxa_sums(community) >1, community)

sort(sample_sums(community))
write.csv(sort(sample_sums(community)), file="output/ranked_library_size_ct_mek_cong.csv")
hist(sample_sums(community), main="Histogram: Read Counts", xlab="Total Reads", col="blue", las=1, breaks=20)
summary(sample_sums(community))
#rarecurve(t(otu_mat), step=50, cex=0.5)
```


# community3: same as community2, but also remove Djiri outlier
```{r}
# see mekong_sites_to_omit.xlsx for selected sites to omit
library(microViz)
# removing mekong duplicate samples with lower reads; see mekong_sites_to_omit.xlsx; also removing Djiri (CONGO_001) as highly polluted: https://www.scirp.org/journal/paperinformation?paperid=107628
community3 <-ps_filter(community, sample_id!="ME060025", sample_id!="ME060026", sample_id!="ME060027",sample_id!="ME060028",sample_id!="ME060031",sample_id!="ME060032",sample_id!="ME060035",sample_id!="ME060037",sample_id!="ME060039",sample_id!="ME060041",sample_id!="ME080033",sample_id!="ME080035",sample_id!="ME080037", sample_id!="ME080038", sample_id!="CONGO_070", sample_id!="ME060033", sample_id!="ME060034", sample_id!="ME060030", sample_id!="CONGO_001", sample_id!="PCA_261") #elim W9 PCA_261 with little metadata
# write metadata file out to compare / verify those samples were deleted
write.csv(sample_data(community3), file="output/community3.csv")

sort(sample_sums(community3))
```

# rareify 
```{r}
# community3
set.seed(42)
community_rare3 <- rarefy_even_depth(community3, sample.size = 9384, rngseed = FALSE, verbose = TRUE) # rareifying to CONGO_019 at 9384 reads
community_prunedrare3 <- prune_taxa(taxa_sums(community_rare3) > 0, community_rare3)
summary(sample_sums(community_rare3))
community_rare3

write.csv(t(otu_table(community_rare3)), file="output/OTUtable-rare9384_no_outliers_11Apr2024.csv")

```
# Set 1 - all samples (Mekong, Congo, CT) including Congo time-series (no Djoue), outliers removed
```{r}
# data prep and analysis of all sites (Congo, CT, Mekong)
# remove Djoue time-series and outliers Lefini and Mopanzi; and remove CT sites that we know are major outliers on CA analysis; just remove from all analysis

all.sites <- ps_filter(community_rare3, site_id.1 !="Lefini", site_id.1 !="Mopanzi", site_id.1 != "Djoue", site_id !="PCA_164", site_id!="PCA_234", site_id!="PCA_243") # from n=101 to n=85 (lost Djoue time series)

all.sites.df <- as.data.frame(sample_data(all.sites)) # check

# pull out OTU table and metadata
otu_all <- otu_table(all.sites)
otu_all <- as.data.frame(otu_all) # rows=taxa, columns=samples
otu_all.t <- as.data.frame(t(otu_all)) # rows=samples, columns=taxa
otu_all.t$sample.number <- row.names(otu_all.t)
end_rare_all <- ncol(otu_all.t)-1

# get metadata
all_meta <- data.frame(sample_data(all.sites))
#standardize variables
all_meta$tt_res_hr <- as.numeric(all_meta$tt_res_hr)
all_meta$tt_nores_hr <- as.numeric(all_meta$tt_nores_hr)
all_meta$ORD_STRA <- as.factor(all_meta$ORD_STRA)
all_meta$lat <- as.numeric(all_meta$lat)
all_meta$DIST_UP_KM <- as.numeric(all_meta$DIST_UP_KM)
all_meta$UPLAND_SKM <- as.numeric(all_meta$UPLAND_SKM)
all_meta$CO2_pct_sat <- as.numeric(all_meta$CO2_pct_sat)
all_meta$CH4_pct_sat <- as.numeric(all_meta$CH4_pct_sat)
all_meta$tmp_dc_month <- as.numeric(all_meta$tmp_dc_month)

# make sure some landcover continuous variables are numeric
all_meta$wet_pc_cg2 <- as.numeric(all_meta$wet_pc_cg2)
all_meta$wet_pc_ug2 <- as.numeric(all_meta$wet_pc_ug2)
all_meta$wet_pc_c04 <- as.numeric(all_meta$wet_pc_c04)
all_meta$wet_pc_c05 <- as.numeric(all_meta$wet_pc_c05)
all_meta$wet_pc_u04 <- as.numeric(all_meta$wet_pc_u04)
all_meta$wet_pc_u05 <- as.numeric(all_meta$wet_pc_u05)
all_meta$for_pc_cse <- as.numeric(all_meta$for_pc_cse)
all_meta$for_pc_use <- as.numeric(all_meta$for_pc_use)
all_meta$tmp_dc_month <- as.numeric(all_meta$tmp_dc_month)

all_meta$pre_mm_uyr <- as.numeric(all_meta$pre_mm_uyr)
all_meta$pre_mm_monthly <- as.numeric(all_meta$pre_mm_monthly)
all_meta$ele_mt_cav <- as.numeric(all_meta$ele_mt_cav)
all_meta$slp_dg_cav <- as.numeric(all_meta$slp_dg_cav)
all_meta$slp_dg_uav <- as.numeric(all_meta$slp_dg_uav)
all_meta$sgr_dk_rav <- as.numeric(all_meta$sgr_dk_rav)

# make wetland metric
all_meta$wet_pc_uall <- all_meta$wet_pc_u04 + all_meta$wet_pc_u05
# convert air temp to water temp from Liu et al 2022: Tw = 0.67Ta + 7.45
all_meta$temp_water_calc <- (0.67*((all_meta$tmp_dc_month)/10)+7.45)


all_meta$log10_tt <- log10(1+all_meta$tt_res_hr)
all_meta$log10_tt_nores <- log10(1+all_meta$tt_nores_hr)
all_meta$log10_q_daily_mean <- log10(1+all_meta$q_daily_mean)
all_meta$log10_DIST_UP_KM <- log10(all_meta$DIST_UP_KM)
all_meta$log10_UPLAND_SKM <- log10(all_meta$UPLAND_SKM) # merit wrong for Ndoki CONGO_020, use hydrosheds UPLAND_SKM instead; see notes
all_meta$log10_DOC <- log10(all_meta$doc_mcc_l)
all_meta$log10_NOx <- log10(1+all_meta$Nox)
all_meta$log10_TDN <- log10(1+all_meta$TDN_mgN_l)
all_meta$log10_PO4 <- log10(1+all_meta$PO4)
all_meta$log10_SPC <- log10(1+all_meta$SPC)
all_meta$log10_suva_254 <- log10(1+all_meta$suva245_mgc_l_m)
all_meta$log10_do_mg_l <- log10(1+all_meta$do_mg_l)
all_meta$log10_CO2_pct_sat <- log10(1+all_meta$CO2_pct_sat)
all_meta$log10_CH4_pct_sat <- log10(1+all_meta$CH4_pct_sat)

all_meta$log10_pre_mm_uyr <- log10(1+all_meta$pre_mm_uyr)
all_meta$log10_pre_mm_monthly <- log10(1+all_meta$pre_mm_monthly)
all_meta$log10_ele_mt_cav <- log10(1+all_meta$ele_mt_cav)
all_meta$log10_slp_dg_cav <- log10(1+all_meta$slp_dg_cav)
all_meta$log10_slp_dg_uav <- log10(1+all_meta$slp_dg_uav)
all_meta$log10_sgr_dk_rav <- log10(1+all_meta$sgr_dk_rav)

# combine metadata and OTUs into one df
all_meta$sample_number <- row.names(all_meta)
meta_otu_all <- merge(all_meta, otu_all.t, by=0, all.x=TRUE)

all_meta_col <- ncol(all_meta)
otu_all_col <- ncol(otu_all.t)
otu_all_start <- all_meta_col+2
otu_all_end <- ncol(meta_otu_all)-1

# pull out metadata table; was prev called "all_meta.2" but renamed to 1
all_meta.1 <- meta_otu_all[,c(1:all_meta_col)]
write.csv(all_meta.1, "output/all_meta.1.csv")
# create new column that codes for synoptic or monthly, and the actual month
all_meta.1$type_month <- ifelse(all_meta.1$sam_type == "synoptic", "synoptic",
                                ifelse(all_meta.1$sam_type =="monthly", paste(all_meta.1$month), "synoptic"))
all_meta.1$type_month <- as.factor(all_meta.1$type_month)

# pull out OTU table
all_otu <- meta_otu_all[,c(otu_all_start:otu_all_end)]

# descriptive statistucs
library(dplyr)
stats.all.meta.1 <- all_meta.1 %>%
  group_by(project) %>%
  summarise(
    count = n(),
    mean_slp = mean(slp_dg_uav, na.rm=TRUE),
    sd_slp = sd(slp_dg_uav, na.rm=TRUE),
    min_slp = min(slp_dg_uav, na.rm=TRUE),
    max_slp = max(slp_dg_uav, na.rm=TRUE),
    median_slp = median(slp_dg_uav, na.rm=TRUE),
    Q1_slp = quantile(slp_dg_uav, probs=0.25, na.rm=TRUE),
    Q3_slp = quantile(slp_dg_uav, probs=0.75, na.rm=TRUE),
        
    mean_sgr = mean(sgr_dk_rav, na.rm=TRUE),
    sd_sgr = sd(sgr_dk_rav, na.rm=TRUE),
    min_sgr = min(sgr_dk_rav, na.rm=TRUE),
    max_sgr = max(sgr_dk_rav, na.rm=TRUE),
    median_sgr = median(sgr_dk_rav, na.rm=TRUE),
    Q1_sgr = quantile(sgr_dk_rav, probs=0.25, na.rm=TRUE),
    Q3_sgr = quantile(sgr_dk_rav, probs=0.75, na.rm=TRUE),
    
    mean_pre = mean(pre_mm_uyr, na.rm=TRUE),
    sd_pre = sd(pre_mm_uyr, na.rm=TRUE),
    min_pre = min(pre_mm_uyr, na.rm=TRUE),
    max_pre = max(pre_mm_uyr, na.rm=TRUE),
    median_pre = median(pre_mm_uyr, na.rm=TRUE),
    Q1_pre = quantile(pre_mm_uyr, probs=0.25, na.rm=TRUE),
    Q3_pre = quantile(pre_mm_uyr, probs=0.75, na.rm=TRUE),
    
    mean_tt = mean(tt_nores_hr, na.rm=TRUE),
    sd_tt = sd(tt_nores_hr, na.rm=TRUE),
    min_tt = min(tt_nores_hr, na.rm=TRUE),
    max_tt = max(tt_nores_hr, na.rm=TRUE),
    median_tt = median(tt_nores_hr, na.rm=TRUE),
    Q1_tt = quantile(tt_nores_hr, probs=0.25, na.rm=TRUE),
    Q3_tt = quantile(tt_nores_hr, probs=0.75, na.rm=TRUE),
    
    mean_wet = mean(wet_pc_uall, na.rm=TRUE),
    sd_wet = sd(wet_pc_uall, na.rm=TRUE),
    min_wet = min(wet_pc_uall, na.rm=TRUE),
    max_wet = max(wet_pc_uall, na.rm=TRUE),
    median_wet = median(wet_pc_uall, na.rm=TRUE),
    Q1_wet = quantile(wet_pc_uall, probs=0.25, na.rm=TRUE),
    Q3_wet = quantile(wet_pc_uall, probs=0.75, na.rm=TRUE),
    
    mean_temp = mean(temp_water_calc, na.rm=TRUE),
    sd_temp = sd(temp_water_calc, na.rm=TRUE),
    min_temp = min(temp_water_calc, na.rm=TRUE),
    max_temp = max(temp_water_calc, na.rm=TRUE),
    median_temp = median(temp_water_calc, na.rm=TRUE),
    Q1_temp = quantile(temp_water_calc, probs=0.25, na.rm=TRUE),
    Q3_temp = quantile(temp_water_calc, probs=0.75, na.rm=TRUE),
  )
write.csv(stats.all.meta.1, "output/stats.all.meta.1.csv", row.names=FALSE)


# biplots of TTs
  # w/RES
pdf("output/biplot_set1_tt_wres_basins.pdf") # totally skewed by huge MILL TT of 10^5
ggplot()+
  geom_text(data=all_meta.1, aes(project, tt_res_hr, label=site_id.1, color=project))+
  theme_classic()+
  ggtitle("TT with Res - all basins")
dev.off()

# elim 2 MILL samples
all_meta.1.noMILL <- subset(all_meta.1, site_id.1 !="MILL") # worked, from 88 to 86 rows

pdf("output/biplot_set1_tt_wres_basins_noMILL.pdf")
ggplot()+
  geom_text(data=all_meta.1.noMILL, aes(project, tt_res_hr, label=site_id.1, color=project))+
  theme_classic()+
  ggtitle("TT with Res - all basins (no MILL)")
dev.off()

pdf("output/biplot_set1_tt_wres_basins_noMILL_log10.pdf")
ggplot()+
  geom_text(data=all_meta.1.noMILL, aes(project, tt_res_hr, label=site_id.1, color=project))+
  theme_classic()+
  scale_y_log10()+
  ggtitle("TT with Res - all basins (no MILL)")
dev.off()

  # no RES
pdf("output/biplot_set1_tt_nores_basins.pdf")
ggplot()+
  geom_text(data=all_meta.1, aes(project, tt_nores_hr, label=site_id.1, color=project))+
  theme_classic()+
  ggtitle("TT No Res - all basins")
dev.off()

library(scales)
pdf("output/biplot_set1_tt_nores_basins_log10.pdf")
ggplot()+
  geom_text(data=all_meta.1, aes(project, tt_nores_hr, label=site_id.1, color=project))+
  theme_classic()+
  scale_y_continuous(trans = 'log10',
                     breaks = c(1,10,100,1000),
                     labels = label_number())
  ggtitle("TT No Res - all basins")
dev.off()



# PCoA wunifrac - all - using ape
# OTU table must be transformed back to rows=taxa, columns=samples; so use otu_all.t
otu_all.m <- as.matrix(otu_all)
otu_all.wuni <- beta.div(otu_all.m, "unifrac", weighted=TRUE, tree=tree)

pcoa.all.wuni <- pcoa (otu_all.wuni, correction="none", rn=NULL)

pcoa.all.wuni.comp <- pcoa.all.wuni$vectors
pcoa.all.wuni.1 <- -1*pcoa.all.wuni.comp[,1]
pcoa.all.wuni.2 <- -2*pcoa.all.wuni.comp[,2]

all_meta.1.pcoa <- all_meta.1

all_meta.1.pcoa$pcoa.all.wuni.1 <- as.numeric(pcoa.all.wuni.1)
all_meta.1.pcoa$pcoa.all.wuni.2 <- as.numeric(pcoa.all.wuni.2)

print(pcoa.all.wuni$values$Eigenvalues) # negative eigenvalues starting with dim 63 + thereafter

# basin - wuni
pdf("output/pcoa.all.wuni.basin.pdf", width=9, height=8)
ggplot()+
  geom_text(data=all_meta.1.pcoa, aes(pcoa.all.wuni.1, pcoa.all.wuni.2, label=site_id.1, color=project), size=4, position=position_nudge(y=0.003), fontface="bold")+
              theme_classic()+
  scale_fill_brewer(palette="Set1")+
  #scale_x_reverse()+
  #scale_y_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.all.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.all.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA all sites wuni ASV")

# basin w/ Cailliez correction
pcoa.all.wuni.cai <- pcoa (otu_all.wuni, correction="cailliez", rn=NULL)

pcoa.all.wuni.cai.comp <- pcoa.all.wuni.cai$vectors
pcoa.all.wuni.cai.1 <- -1*pcoa.all.wuni.cai.comp[,1]
pcoa.all.wuni.cai.2 <- -2*pcoa.all.wuni.cai.comp[,2]

all_meta.1.pcoa$pcoa.all.wuni.cai.1 <- as.numeric(pcoa.all.wuni.cai.1)
all_meta.1.pcoa$pcoa.all.wuni.cai.2 <- as.numeric(pcoa.all.wuni.cai.2)

print(pcoa.all.wuni.cai$values$Corr_eig) # all positive corrected eigenvalues

# basin - wuni w/ cailliez correction
pdf("output/pcoa.all.wuni.basin.cailliez.pdf", width=9, height=8)
ggplot()+
  geom_text(data=all_meta.1.pcoa, aes(pcoa.all.wuni.cai.1, pcoa.all.wuni.cai.2, label=site_id.1, color=project), size=4, position=position_nudge(y=0.003), fontface="bold")+
              theme_classic()+
  scale_fill_brewer(palette="Set1")+
  #scale_x_reverse()+
  #scale_y_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.all.wuni.cai$values$Rel_corr_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.all.wuni.cai$values$Rel_corr_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA all sites wuni ASV - cailliez correction")

library(RColorBrewer)
#water_shapes <- c(21,22,23,24) 
#names(water_shapes) <-levels(com_meta.pcoa$watershed)
#shape_scale_sheds <- scale_shape_manual(name="watershed", values=water_shapes)
all_meta.1.pcoa$type_month <- factor(all_meta.1.pcoa$type_month, levels=c("January", "February", "March", "April", "May", "August", "October", "November", "December", "synoptic"))

month_color <- c ("January"="#008000", "February"="#008080", "March"="#0000FF", "April"="#800080", "May"="#FF00FF", "August"="#FF0000", "October"="#ca6f1e", "November"="#f7dc6f", "December"="#00FF00", "synoptic"="black")

# basin - cailliez - shapes
pdf("output/pcoa.all.wuni.basin.cailliez.shapes.pdf", width=11, height=8)
ggplot(all_meta.1.pcoa, aes(pcoa.all.wuni.cai.1, pcoa.all.wuni.cai.2, shape=project))+
         geom_point(aes(color=factor(all_meta.1.pcoa$type_month), size=4))+
         theme_classic()+
  scale_color_manual(values=month_color)+
  #scale_fill_brewer(palette="Set1")+
  #scale_x_reverse()+
  #scale_y_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.all.wuni.cai$values$Rel_corr_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.all.wuni.cai$values$Rel_corr_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA all sites wuni ASV - cailliez correction")

shapes_sam_type <- c("synoptic" = 16, "monthly" = 18 )
# basin - caillez - sheds as colors; Congo time-series with added text of month abbrev
pdf("output/pcoa.all.wuni.basin.cailliez.1shape.pdf", width=11, height=8)
ggplot(all_meta.1.pcoa, aes(pcoa.all.wuni.cai.1, pcoa.all.wuni.cai.2, color=project, shape=sam_type))+
  geom_point(size=5)+#aes(color=all_meta.1.pcoa$project), size=4)+
  geom_text(aes(label = ifelse(all_meta.1.pcoa$type_month !="synoptic", as.character(all_meta.1.pcoa$type_month), "")), size=5,  hjust=0.5, nudge_y=0.02)+
  scale_shape_manual(values=shapes_sam_type)+
  theme_classic()+
  xlab(paste("PCoA-1 (",round(pcoa.all.wuni.cai$values$Rel_corr_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.all.wuni.cai$values$Rel_corr_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA all sites wuni ASV - cailliez correction")
dev.off()


all_meta.2.pcoa$type_month <- factor(all_meta.2.pcoa$type_month, levels=c("January", "February", "March", "April", "May", "August", "October", "November", "December"))
# new PCoA
pdf("output/pcoa.all.wuni.basin.test.pdf", width=9, height=8)
ggplot(all_meta.2.pcoa, aes(pcoa.all.wuni.1, pcoa.all.wuni.2))+
  geom_text(aes(label=site_id.1, color=project), size=4, position=position_nudge(y=0.003), fontface="bold")+
  geom_point(aes(shape=type_month, color=factor(all_meta.2.pcoa$type_month)))+
  #scale_color_manual(values=factor(all_meta.2.pcoa$type_month))+
  #geom_point(data=meta_syn_pcoa, aes(pcoa.otu.syn.wuni.1, pcoa.otu.syn.wuni.2, shape=project, color=order_merit), size=3)+
              theme_classic()+
  scale_fill_brewer(palette="Set1")+
  #scale_x_reverse()+
  scale_y_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.all.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.all.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA all sites wuni ASV")

# PCoA using dudi.pco (ade4)
# https://adeverse.github.io/ade4/articles/ChapSpeciesStruct.html#principal-coordinate-analysis-pcoa
# https://adeverse.github.io/ade4/articles/Chap2tables.html
# test if weighted unifrac distance matrix is euclidean (required for PCoA)
is.euclid(otu_all.wuni) # false
# pcoa on non-euclidean wunifrac 
pcoa.all.wuni.dudi.no.euc <- dudi.pco(otu_all.wuni, scannf=FALSE) # technicaly wrong approach with non-Euclidean dist matrix, but produces no negative eigenvalues
print(pcoa.all.wuni.dudi.no.euc$eig)

# convert wuni to euclidean via transformation
otu_all.wuni.euc <- cailliez(otu_all.wuni)
is.euclid(otu_all.wuni.euc) # true
pcoa.all.wuni.dudi <- dudi.pco(otu_all.wuni.euc, scannf=FALSE)
# pull out coordinates

pcoa.all.wuni.dudi.li <- as.data.frame(pcoa.all.wuni.dudi$li)
pcoa.all.wuni.dudi.li.1 <- -1*pcoa.all.wuni.dudi.li[,1]
pcoa.all.wuni.dudi.li.2 <- -2*pcoa.all.wuni.dudi.li[,2]

all_meta.1.pcoa$pcoa.all.wuni.dudi.1 <- as.numeric(pcoa.all.wuni.dudi.li.1)
all_meta.1.pcoa$pcoa.all.wuni.dudi.2 <- as.numeric(pcoa.all.wuni.dudi.li.2)

# basin - wuni dudi.pco cailliez -0.163933320 0.0485618435; 0.163933320 -0.09712368
pdf("output/pcoa.all.wuni.basin.dudi.euc.pdf", width=9, height=8)
ggplot()+
  geom_text(data=all_meta.1.pcoa, aes(pcoa.all.wuni.dudi.1, pcoa.all.wuni.dudi.2, label=site_id.1, color=project), size=4, position=position_nudge(y=0.003), fontface="bold")+
              theme_classic()+
  scale_fill_brewer(palette="Set1")+
  #scale_x_reverse()+
  #scale_y_reverse()+
  xlab("25.7% Projected Inertia")+ 
  ylab("11.4% Projected Inertia")+ 
  ggtitle("PCoA all sites wuni ASV - dudi.pco cailliez trans")
dev.off()


########
# COA
all_meta_coa<-all_meta.1
otu_coa_all <- dudi.coa(all_otu, scannf=FALSE, nf=8)
# plot COA - export coordinates
otu_coa_all_li <- as.data.frame(otu_coa_all$li) # site scores
coa_all <- merge (otu_coa_all_li, all_meta_coa, by=0)

otu_coa_all_co <- as.data.frame(otu_coa_all$co) # species scores
otu_coa_all_co$ASV <- rownames(otu_coa_all_co)

# plot sites only
# basin
pdf("output/coa.all.basin.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_all,aes(Axis1,Axis2,label=site_id.1, color=project), size=4, fontface="bold")+
  #scale_color_gradient(low="blue", high="green") +
  theme_classic()+
  ggtitle("CA all sites")+
  xlab("3.9% projected inertia") +
  ylab("3.6% projected inertia")
dev.off()

# plot sites + species
pdf("output/coa.all.basin.species.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_all,aes(Axis1,Axis2,label=site_id.1, color=project), size=4, fontface="bold")+
  geom_text(data=otu_coa_all_co, aes(Comp1, Comp2, label=ASV))+
  #scale_color_gradient(low="blue", high="green") +
  theme_classic()+
  ggtitle("CA all sites + ASVs")+
  xlab("3.9% projected inertia") +
  ylab("3.6% projected inertia")
dev.off()


# co-inertia using COA
# 1. subset variables of interest - using UPLAND_SKM because no other good hydro data for CT (no Shaoda TT data yet)
all_meta.coin.pca.1 <- all_meta.1[,c("log10_UPLAND_SKM", "wet_pc_uall", "log10_pre_mm_uyr", "log10_pre_mm_monthly", "tmp_dc_month", "log10_ele_mt_cav", "log10_slp_dg_cav", "log10_slp_dg_uav", "log10_sgr_dk_rav")]
# PCA
env_pca_all.1 <- dudi.pca(d=all_meta.coin.pca.1, row.w=otu_coa_all$lw, scale=TRUE, scannf=FALSE, nf=10)
# co-inertia
coin_all.1 <- coinertia(env_pca_all.1, otu_coa_all, scan = FALSE, nf = 5)

enviro_loadings_coin_all.1 <- as.data.frame(coin_all.1$co)
enviro_loadings_coin_all.1$variable <- row.names(enviro_loadings_coin_all.1)

site_scores_coin_all.1_0 <- as.data.frame(coin_all.1$mX)
site_scores_coin_all.1 <- bind_cols(site_scores_coin_all.1_0, all_meta.2)

# by upland km2
pdf("output/coin.all.1.km2.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.1,aes(NorS1,NorS2,label=site_id.1, color=log10_UPLAND_SKM), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin_all.1, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.1, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.8% Projected Inertia")+ # 0.3693518*61.795 = 22.82409
  ylab("5.8% Projected Inertia")+ # 0.3693518*15.812 = 5.840191
  ggtitle("All sites coin.all.1.km2.pdf RV=0.370")
dev.off()

# by basin
pdf("output/coin.all.1.basin.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.1,aes(NorS1,NorS2,label=site_id.1, color=project), size=4, fontface="bold")+
  #scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin_all.1, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.1, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.8% Projected Inertia")+ # 0.3693518*61.795 = 22.82409
  ylab("5.8% Projected Inertia")+ # 0.3693518*15.812 = 5.840191
  ggtitle("All sites coin.all.1.basin.pdf RV=0.370")
dev.off()

# pre_mm_uyr
pdf("output/coin.all.1.precip.yr.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.1,aes(NorS1,NorS2,label=site_id.1, color=log10_pre_mm_uyr), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="blue")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin_all.1, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.1, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.8% Projected Inertia")+ # 0.3693518*61.795 = 22.82409
  ylab("5.8% Projected Inertia")+ # 0.3693518*15.812 = 5.840191
  ggtitle("All sites coin.all.1.precip.yr.pdf RV=0.370")
dev.off()

#################################
# Models 2+3 - use TT with res 
#################################
### FULL MODEL ####
# 2. using TT w/ res
all_meta.coin.pca.2.tt <- all_meta.1[,c("log10_tt", "wet_pc_uall", "log10_pre_mm_uyr", "log10_pre_mm_monthly", "temp_water_calc", "log10_ele_mt_cav", "log10_slp_dg_cav", "log10_slp_dg_uav", "log10_sgr_dk_rav")]
# PCA
env_pca_all.2.tt <- dudi.pca(d=all_meta.coin.pca.2.tt, row.w=otu_coa_all$lw, scale=TRUE, scannf=FALSE, nf=10)
# co-inertia
coin_all.2.tt <- coinertia(env_pca_all.2.tt, otu_coa_all, scan = FALSE, nf = 5)

enviro_loadings_coin_all.2.tt <- as.data.frame(coin_all.2.tt$co)
enviro_loadings_coin_all.2.tt$variable <- row.names(enviro_loadings_coin_all.2.tt)

site_scores_coin_all.2.tt_0 <- as.data.frame(coin_all.2.tt$mX)
site_scores_coin_all.2.tt <- bind_cols(site_scores_coin_all.2.tt_0, all_meta.1)

# by TT w/res
pdf("output/coin.all.2.tt.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.2.tt,aes(NorS1,NorS2,label=site_id.1, color=log10_tt), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin_all.2.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.2.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("24.8% Projected Inertia")+ # 0.3684298*67.390 = 24.82848
  ylab("5.0% Projected Inertia")+ # 0.3684298*13.474 = 4.964223
  ggtitle("Set 1-All sites TT, except Djoue. coin.all.2.tt  RV=0.368")
dev.off()

# by basin
pdf("output/coin.all.2.basin.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.2.tt,aes(NorS1,NorS2,label=site_id.1, color=project), size=4, fontface="bold")+
  #scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin_all.2.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.2.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("24.8% Projected Inertia")+ # 0.3684298*67.390 = 24.82848
  ylab("5.0% Projected Inertia")+ # 0.3684298*13.474 = 4.964223
  ggtitle("Set 1-All sites TT, except Djoue. coin.all.2.tt  RV=0.368")
dev.off()

# by precip (stronvest vector)
pdf("output/coin.all.2.precip.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.2.tt,aes(NorS1,NorS2,label=site_id.1, color=log10_pre_mm_uyr), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="blue")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin_all.2.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.2.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("24.8% Projected Inertia")+ # 0.3684298*67.390 = 24.82848
  ylab("5.0% Projected Inertia")+ # 0.3684298*13.474 = 4.964223
  ggtitle("Set 1-All sites pre_mm_uyr, except Djoue. coin.all.2.tt  RV=0.368")
dev.off()

## REDUCED MODEL - omitting redundant variables
# eliminate the weaker variables in the above model coin_all.2.tt, esp ones that are redundant/co-vary
    # elim: log10_pre_mm_monthly (the annual is strongest vector above); log10_ele_mt_cav (elevation in reach catchment); log10_slp_dg_cav (average slope in reach catchment; instead keep upstream one, it's more informative, gives idea of average stream gradient in entire watershed)
# 3 elim log10_pre_mm_uyr (annual precip in upstream watershed), log10_ele_mt_cav (elevation in reach catchment), log10_slp_dg_cav (average slope in reach catchment)
all_meta.coin.pca.3.tt <- all_meta.1[,c("log10_tt", "wet_pc_uall",  "log10_pre_mm_uyr", "temp_water_calc", "log10_slp_dg_uav", "log10_sgr_dk_rav")]
# PCA
env_pca_all.3.tt <- dudi.pca(d=all_meta.coin.pca.3.tt, row.w=otu_coa_all$lw, scale=TRUE, scannf=FALSE, nf=10)
# co-inertia
coin_all.3.tt <- coinertia(env_pca_all.3.tt, otu_coa_all, scan = FALSE, nf = 5)

enviro_loadings_coin_all.3.tt <- as.data.frame(coin_all.3.tt$co)
enviro_loadings_coin_all.3.tt$variable <- row.names(enviro_loadings_coin_all.3.tt)

site_scores_coin_all.3.tt_0 <- as.data.frame(coin_all.3.tt$mX)
site_scores_coin_all.3.tt <- bind_cols(site_scores_coin_all.3.tt_0, all_meta.1)

# by TT
pdf("output/coin.all.3.tt.reduced.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.3.tt,aes(NorS1,NorS2,label=site_id.1, color=log10_tt), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin_all.3.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.3.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.1% Projected Inertia")+ # 0.3528798*71.153 = 25.10846
  ylab("5.4% Projected Inertia")+ # 0.3528798*15.219 = 5.370478
  ggtitle("Set 1-All sites TT, except Djoue. REDUCED coin.all.3.tt  RV=0.353")
dev.off()

# by basin
pdf("output/coin.all.3.basin.reduced.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.3.tt,aes(NorS1,NorS2,label=site_id.1, color=project), size=4, fontface="bold")+
  #scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin_all.3.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.3.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.1% Projected Inertia")+ # 0.3528798*71.153 = 25.10846
  ylab("5.4% Projected Inertia")+ # 0.3528798*15.219 = 5.370478
  ggtitle("Set 1-All sites TT, except Djoue. REDUCED coin.all.3.tt  RV=0.353")
dev.off()

# by basin, precip color
pdf("output/coin.all.3.precip.reduced.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.3.tt,aes(NorS1,NorS2,label=site_id.1, color=log10_pre_mm_uyr), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="blue")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin_all.3.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.3.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.1% Projected Inertia")+ # 0.3528798*71.153 = 25.10846
  ylab("5.4% Projected Inertia")+ # 0.3528798*15.219 = 5.370478
  ggtitle("Set 1-All sites pre_mm_uyr, except Djoue. REDUCED coin.all.3.tt  RV=0.353")
dev.off()

#################################
# Models 4+5 - use TT NO res 
#################################
    # especially affects CT data, which have extremely long TTs when including reservoirs (above)
### FULL MODEL ####
# 4. using TT NO res
all_meta.coin.pca.4.tt.nores <- all_meta.1[,c("log10_tt_nores", "wet_pc_uall", "log10_pre_mm_uyr", "log10_pre_mm_monthly", "temp_water_calc", "log10_ele_mt_cav", "log10_slp_dg_cav", "log10_slp_dg_uav", "log10_sgr_dk_rav")]
# PCA
env_pca_all.4.tt.nores <- dudi.pca(d=all_meta.coin.pca.4.tt.nores, row.w=otu_coa_all$lw, scale=TRUE, scannf=FALSE, nf=10)
# co-inertia
coin_all.4.tt.nores <- coinertia(env_pca_all.4.tt.nores, otu_coa_all, scan = FALSE, nf = 5)

enviro_loadings_coin_all.4.tt.nores <- as.data.frame(coin_all.4.tt.nores$co)
enviro_loadings_coin_all.4.tt.nores$variable <- row.names(enviro_loadings_coin_all.4.tt.nores)

site_scores_coin_all.4.tt.nores_0 <- as.data.frame(coin_all.4.tt.nores$mX)
site_scores_coin_all.4.tt.nores <- bind_cols(site_scores_coin_all.4.tt.nores_0, all_meta.1)

# by TT w/res
pdf("output/coin.all.4.tt.nores.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.4.tt.nores,aes(NorS1,NorS2,label=site_id.1, color=log10_tt_nores), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  #scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin_all.4.tt.nores, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.4.tt.nores, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("24.8% Projected Inertia")+ # 0.3757175*65.902 = 24.76053
  ylab("6.2% Projected Inertia")+ # 0.3757175*16.396 = 6.160264
  ggtitle("Set 1-All sites TT No Res, except Djoue. coin.all.4.tt.nores  RV=0.376")
dev.off()

# by basin
pdf("output/coin.all.4.basin.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.4.tt.nores,aes(NorS1,NorS2,label=site_id.1, color=project), size=4, fontface="bold")+
  #scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  #scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin_all.4.tt.nores, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.4.tt.nores, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
   xlab("24.8% Projected Inertia")+ # 0.3757175*65.902 = 24.76053
  ylab("6.2% Projected Inertia")+ # 0.3757175*16.396 = 6.160264
  ggtitle("Set 1-All sites TT No Res, except Djoue. coin.all.4.tt.nores  RV=0.376")
dev.off()

# by precip (stronvest vector)
pdf("output/coin.all.4.precip.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.4.tt.nores,aes(NorS1,NorS2,label=site_id.1, color=log10_pre_mm_uyr), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="blue")+
  theme_classic()+
  #scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin_all.4.tt.nores, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.4.tt.nores, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("24.8% Projected Inertia")+ # 0.3757175*65.902 = 24.76053
  ylab("6.2% Projected Inertia")+ # 0.3757175*16.396 = 6.160264
  ggtitle("Set 1-All sites TT No Res, except Djoue. coin.all.4.tt.nores  RV=0.376")
dev.off()

## REDUCED MODEL - omitting redundant variables
# eliminate the weaker variables in the above model coin_all.2.tt, esp ones that are redundant/co-vary
    # elim: log10_pre_mm_monthly (the annual is strongest vector above); log10_ele_mt_cav (elevation in reach catchment); log10_slp_dg_cav (average slope in reach catchment; instead keep upstream one, it's more informative, gives idea of average stream gradient in entire watershed)
# 3 elim log10_pre_mm_uyr (annual precip in upstream watershed), log10_ele_mt_cav (elevation in reach catchment), log10_slp_dg_cav (average slope in reach catchment)
all_meta.coin.pca.5.tt.nores <- all_meta.1[,c("log10_tt_nores", "wet_pc_uall",  "log10_pre_mm_uyr", "temp_water_calc", "log10_slp_dg_uav", "log10_sgr_dk_rav")]
# PCA
env_pca_all.5.tt.nores <- dudi.pca(d=all_meta.coin.pca.5.tt.nores, row.w=otu_coa_all$lw, scale=TRUE, scannf=FALSE, nf=10)
# co-inertia
coin_all.5.tt.nores <- coinertia(env_pca_all.5.tt.nores, otu_coa_all, scan = FALSE, nf = 5)

enviro_loadings_coin_all.5.tt.nores <- as.data.frame(coin_all.5.tt.nores$co)
enviro_loadings_coin_all.5.tt.nores$variable <- row.names(enviro_loadings_coin_all.5.tt.nores)

site_scores_coin_all.5.tt.nores_0 <- as.data.frame(coin_all.5.tt.nores$mX)
site_scores_coin_all.5.tt.nores <- bind_cols(site_scores_coin_all.5.tt.nores_0, all_meta.1)

# by TT no Res
pdf("output/coin.all.5.tt.nores.reduced.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.5.tt.nores,aes(NorS1,NorS2,label=site_id.1, color=log10_tt_nores), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  #scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin_all.5.tt.nores, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.5.tt.nores, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("27.3% Projected Inertia")+ # 0.3942039*69.137 = 27.25408
  ylab("8.0% Projected Inertia")+ # 0.3942039*20.242 = 7.979475
  ggtitle("Set 1-All sites TT, except Djoue. REDUCED coin_all.5.tt.nores (3 cT removed)  RV=0.394")
dev.off()

# by basin
pdf("output/coin.all.5.basin.reduced.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.5.tt.nores,aes(NorS1,NorS2,label=site_id.1, color=project), size=4, fontface="bold")+
  #scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  #scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin_all.5.tt.nores, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.5.tt.nores, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("27.3% Projected Inertia")+ # 0.3942039*69.137 = 27.25408
  ylab("8.0% Projected Inertia")+ # 0.3942039*20.242 = 7.979475
  ggtitle("Set 1-All sites TT, except Djoue. REDUCED coin_all.5.tt.nores (3 cT removed)  RV=0.394")
dev.off()

# by basin w/ shapes for CT and Mekong
pdf("output/coin.all.5.basin.reduced.shapes.pdf", width=9, height=8)
ggplot(site_scores_coin_all.5.tt.nores, aes(NorS1, NorS2, color=project))+
  geom_point(data = subset(site_scores_coin_all.5.tt.nores, project != "congo"), size=5)+
  geom_text(data = subset(site_scores_coin_all.5.tt.nores, project == "congo"), aes(label = as.character(site_id.1)), size=5, hjust=0.5, nudge_y=0.01)+
  #scale_shape_manual(values=)
  theme_classic()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin_all.5.tt.nores, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.5.tt.nores, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("27.3% Projected Inertia")+ # 0.3942039*69.137 = 27.25408
  ylab("8.0% Projected Inertia")+ # 0.3942039*20.242 = 7.979475
  ggtitle("Set 1-All sites TT, except Djoue. REDUCED coin_all.5.tt.nores (3 cT removed)  RV=0.394")
dev.off()
  

# by basin, precip color
pdf("output/coin.all.5.precip.reduced.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.5.tt.nores,aes(NorS1,NorS2,label=site_id.1, color=log10_pre_mm_uyr), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="blue")+
  theme_classic()+
  #scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin_all.5.tt.nores, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.5.tt.nores, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("27.2% Projected Inertia")+ # 0.3906271*69.655 = 27.20913
  ylab("7.7% Projected Inertia")+ # 0.3906271*19.597 = 7.655119
  ggtitle("Set 1-All sites TT, except Djoue. REDUCED coin_all.5.tt.nores (3 cT removed)  RV=0.391")
dev.off()


#### using PCoA #########
# 5.1 reduced model using PCoA created via dudi.pco
# PCA - remake using row weights from PCoA (prob unnecessary, but def don't make using COA row weights)
env_pca_all.5.1.tt.nores <- dudi.pca(d=all_meta.coin.pca.5.tt.nores, row.w=pcoa.all.wuni.dudi$lw, scale=TRUE, scannf=FALSE, nf=10)
# co-inertia using pcoa
coin_all.5.1.tt.nores <- coinertia(env_pca_all.5.1.tt.nores, pcoa.all.wuni.dudi, scan = FALSE, nf = 5)

enviro_loadings_coin_all.5.1.tt.nores <- as.data.frame(coin_all.5.1.tt.nores$co)
enviro_loadings_coin_all.5.1.tt.nores$variable <- row.names(enviro_loadings_coin_all.5.1.tt.nores)

site_scores_coin_all.5.1.tt.nores_0 <- as.data.frame(coin_all.5.1.tt.nores$mX)
site_scores_coin_all.5.1.tt.nores <- bind_cols(site_scores_coin_all.5.1.tt.nores_0, all_meta.1)

# by TT no Res
pdf("output/coin.all.5.1.tt.nores.reduced.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_all.5.1.tt.nores,aes(NorS1,NorS2,label=site_id.1, color=log10_tt_nores), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  #scale_y_reverse()+
  #scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin_all.5.1.tt.nores, aes(x=0, y=0, xend=Comp1*30, yend=Comp2*30), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.5.1.tt.nores, aes(Comp1*30, Comp2*30, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("26.4% Projected Inertia")+ # 0.292075*90.2684 = 26.36514
  ylab("1.3% Projected Inertia")+ # 0.292075*4.4535 = 1.300756
  ggtitle("Set 1-All sites TT, except Djoue. REDUCED coin.all.5.1.tt.nores PCoA  RV=0.292")
dev.off()

# by basin
pdf("output/coin.all.5.1.basin.reduced.pdf", width=9, height=8)
ggplot()+
 geom_text(data=site_scores_coin_all.5.1.tt.nores,aes(NorS1,NorS2,label=site_id.1, color=project), size=4, fontface="bold")+
  #scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  #scale_y_reverse()+
  #scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin_all.5.1.tt.nores, aes(x=0, y=0, xend=Comp1*30, yend=Comp2*30), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.5.1.tt.nores, aes(Comp1*30, Comp2*30, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("26.4% Projected Inertia")+ # 0.292075*90.2684 = 26.36514
  ylab("1.3% Projected Inertia")+ # 0.292075*4.4535 = 1.300756
  ggtitle("Set 1-All sites TT, except Djoue. REDUCED coin.all.5.1.tt.nores PCoA  RV=0.292")
dev.off()

# by basin, precip color
pdf("output/coin.all.5.1.precip.reduced.pdf", width=9, height=8)
ggplot()+
 geom_text(data=site_scores_coin_all.5.1.tt.nores,aes(NorS1,NorS2,label=site_id.1, color=log10_pre_mm_uyr), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="blue")+
  theme_classic()+
  #scale_y_reverse()+
  #scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin_all.5.1.tt.nores, aes(x=0, y=0, xend=Comp1*30, yend=Comp2*30), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin_all.5.1.tt.nores, aes(Comp1*30, Comp2*30, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("26.4% Projected Inertia")+ # 0.292075*90.2684 = 26.36514
  ylab("1.3% Projected Inertia")+ # 0.292075*4.4535 = 1.300756
  ggtitle("Set 1-All sites TT, except Djoue. REDUCED coin.all.5.1.tt.nores PCoA  RV=0.292")
dev.off()

```

# set 2: all sites (Congo, Mekong, CT) without Congo Monthly sites, without Congo outliers (Mopanzi and Djiri)
```{r}
all.sites.2 <- ps_filter(community_rare3, sam_type == "synoptic")  # to only have synoptic sites, no time series

# remove outliers; Lefini and Mopanzi
all.sites.2 <- ps_filter(all.sites.2, site_id.1 !="Lefini", site_id.1 !="Mopanzi")
all.sites.2.df <- as.data.frame(sample_data(all.sites.2))

# manually pull out OTU table, metadata, etc as in CT paper
otu_all.sites.2 <- otu_table(all.sites.2)
otu_all.sites.2 <- as.data.frame(otu_all.sites.2) # rows=taxa, columns=samples
otu_all.sites.2.t <- as.data.frame(t(otu_all.sites.2)) # transform: rows=samples, columns=taxa
otu_all.sites.2.t$sample.number <- row.names(otu_all.sites.2.t)
end_rare.all.site.2<-ncol(otu_all.sites.2.t)-1

all.sites.2.meta <- data.frame(sample_data(all.sites.2))

# standardize some variables
all.sites.2.meta$tt_res_hr <- as.numeric(all.sites.2.meta$tt_res_hr)
all.sites.2.meta$ORD_STRA <- as.factor(all.sites.2.meta$ORD_STRA)
all.sites.2.meta$lat <- as.numeric(all.sites.2.meta$lat)
all.sites.2.meta$DIST_UP_KM <- as.numeric(all.sites.2.meta$DIST_UP_KM)
all.sites.2.meta$UPLAND_SKM <- as.numeric(all.sites.2.meta$UPLAND_SKM)
all.sites.2.meta$CO2_pct_sat <- as.numeric(all.sites.2.meta$CO2_pct_sat)
all.sites.2.meta$CH4_pct_sat <- as.numeric(all.sites.2.meta$CH4_pct_sat)

# make sure some landcover continuous variables are numeric
all.sites.2.meta$wet_pc_cg2 <- as.numeric(all.sites.2.meta$wet_pc_cg2)
all.sites.2.meta$wet_pc_ug2 <- as.numeric(all.sites.2.meta$wet_pc_ug2)
all.sites.2.meta$wet_pc_c04 <- as.numeric(all.sites.2.meta$wet_pc_c04)
all.sites.2.meta$wet_pc_c05 <- as.numeric(all.sites.2.meta$wet_pc_c05)
all.sites.2.meta$wet_pc_u04 <- as.numeric(all.sites.2.meta$wet_pc_u04)
all.sites.2.meta$wet_pc_u05 <- as.numeric(all.sites.2.meta$wet_pc_u05)
all.sites.2.meta$for_pc_cse <- as.numeric(all.sites.2.meta$for_pc_cse)
all.sites.2.meta$for_pc_use <- as.numeric(all.sites.2.meta$for_pc_use)

#code categorical variable land_cover as a numeric; 1 = forest, 2 = savanna, 3 = swamp
all.sites.2.meta$land_Mann_num <- as.numeric(as.factor(all.sites.2.meta$land_Mann))
# convert land_upstill to factor
all.sites.2.meta$land_upstill<- as.factor(all.sites.2.meta$land_upstill)
all.sites.2.meta$land_upstill_num<- as.numeric(as.factor(all.sites.2.meta$land_upstill)) # 1=forest, 2=savanna, 3=swamp

# combine wetland upstream variables
all.sites.2.meta$wet_pc_uall <- all.sites.2.meta$wet_pc_u04 + congo_syn_meta$wet_pc_u05


all.sites.2.meta$log10_tt <- log10(1+all.sites.2.meta$tt_res_hr)
all.sites.2.meta$log10_q_daily_mean <- log10(1+all.sites.2.meta$q_daily_mean)
all.sites.2.meta$log10_DIST_UP_KM <- log10(all.sites.2.meta$DIST_UP_KM)
all.sites.2.meta$log10_UPLAND_SKM <- log10(all.sites.2.meta$UPLAND_SKM) # merit wrong for Ndoki CONGO_020, use hydrosheds UPLAND_SKM instead; see notes
all.sites.2.meta$log10_DOC <- log10(all.sites.2.meta$doc_mcc_l)
all.sites.2.meta$log10_NOx <- log10(1+all.sites.2.meta$Nox)
all.sites.2.meta$log10_TDN <- log10(1+all.sites.2.meta$TDN_mgN_l)
all.sites.2.meta$log10_PO4 <- log10(1+all.sites.2.meta$PO4)
all.sites.2.meta$log10_SPC <- log10(1+all.sites.2.meta$SPC)
all.sites.2.meta$log10_suva_254 <- log10(1+all.sites.2.meta$suva245_mgc_l_m)
all.sites.2.meta$log10_do_mg_l <- log10(1+all.sites.2.meta$do_mg_l)
all.sites.2.meta$log10_CO2_pct_sat <- log10(1+all.sites.2.meta$CO2_pct_sat)
all.sites.2.meta$log10_CH4_pct_sat <- log10(1+all.sites.2.meta$CH4_pct_sat)

# combine metadata and OTUs into one df
all.sites.2.meta$sample.number <- row.names(all.sites.2.meta)
all.sites.2.otu.meta <- merge(all.sites.2.meta, otu_all.sites.2.t, by=0, all.x=TRUE)

all.sites.2.meta.col <-ncol(all.sites.2.meta)
otu_all.sites.2.col <- ncol(otu_all.sites.2.t)
otu_all.sites.2.start <- all.sites.2.meta.col+2
otu_all.sites.2.end <- ncol(all.sites.2.otu.meta)-1

########
# PCoA - set 2
#######
otu_all.sites.2.m <- as.matrix(otu_all.sites.2)
all.sites.2.wuni <- beta.div(otu_all.sites.2.m, "unifrac", weighted=TRUE, tree=tree)
pcoa.all.sites.2.wuni <- pcoa(all.sites.2.wuni, correction="none", rn=NULL)

meta.all.2 <- all.sites.2.otu.meta[,c(1:all.sites.2.meta.col)]

pcoa.all.sites.2.wuni.comp <- pcoa.all.sites.2.wuni$vectors
pcoa.all.sites.2.wuni.1 <- -1*pcoa.all.sites.2.wuni.comp[,1]
pcoa.all.sites.2.wuni.2 <- -2*pcoa.all.sites.2.wuni.comp[,2]

meta.all.2.pcoa <- meta.all.2
meta.all.2.pcoa$pcoa.all.sites.2.ax1 <- as.numeric(pcoa.all.sites.2.wuni.1)
meta.all.2.pcoa$pcoa.all.sites.2.ax2 <- as.numeric(pcoa.all.sites.2.wuni.2)

pdf("output/pcoa.all.sites2.wuni.basin.pdf", width=9, height=8) 
ggplot()+
  geom_text(data=meta.all.2.pcoa, aes(pcoa.all.sites.2.ax1, pcoa.all.sites.2.ax2, label=site_id.1, color=project), size=4, position=position_nudge(y=0.003), fontface="bold")+
              theme_classic()+
  scale_fill_brewer(palette="Set1")+
  #scale_x_reverse()+
  scale_y_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.all.sites.2.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.all.sites.2.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA all sites.2 wuni ASV; no monthly or outliers")
```

# Congo subset
```{r}
## separate / subset out the Congo dataset from main phyloseq object, including its taxonomy, OTUs etc using microviz
# main congo object (synoptic and time-series)
congo_com3 <- ps_filter(community_rare3, project=="congo")
sort(sample_sums(congo_com3))

# synoptic only
congo_syn <- ps_filter(community_rare3, project=="congo", sam_type == "synoptic") # n=23
sort(sample_sums(congo_syn))

# monthly time-series only
congo_mon <- ps_filter(community_rare3, project=="congo", sam_type == "monthly")
sort(sample_sums(congo_mon))

#______________Congo Synoptic____________

# manually pull out OTU table, metadata, etc as in CT paper
#tree_congo_syn <- phy_tree(congo_syn)
#tree_congo_syn <- as.phylo(tree_congo_syn)
otu_congo_syn <- otu_table(congo_syn)
otu_congo_syn <- as.data.frame(otu_congo_syn) # rows=taxa, columns=samples
otu_congo_syn.t <- as.data.frame(t(otu_congo_syn)) # transform: rows=samples, columns=taxa
otu_congo_syn.t$sample.number <- row.names(otu_congo_syn.t)
end_rare<-ncol(otu_congo_syn.t)-1

congo_syn_meta <- data.frame(sample_data(congo_syn))

# standardize some variables
congo_syn_meta$tt_res_hr <- as.numeric(congo_syn_meta$tt_res_hr)
congo_syn_meta$tt_nores_hr <- as.numeric(congo_syn_meta$tt_nores_hr)
congo_syn_meta$ORD_STRA <- as.factor(congo_syn_meta$ORD_STRA)
congo_syn_meta$lat <- as.numeric(congo_syn_meta$lat)
congo_syn_meta$DIST_UP_KM <- as.numeric(congo_syn_meta$DIST_UP_KM)
congo_syn_meta$UPLAND_SKM <- as.numeric(congo_syn_meta$UPLAND_SKM)
congo_syn_meta$CO2_pct_sat <- as.numeric(congo_syn_meta$CO2_pct_sat)
congo_syn_meta$CH4_pct_sat <- as.numeric(congo_syn_meta$CH4_pct_sat)

# make sure some landcover continuous variables are numeric
congo_syn_meta$wet_pc_cg2 <- as.numeric(congo_syn_meta$wet_pc_cg2)
congo_syn_meta$wet_pc_ug2 <- as.numeric(congo_syn_meta$wet_pc_ug2)
congo_syn_meta$wet_pc_c04 <- as.numeric(congo_syn_meta$wet_pc_c04)
congo_syn_meta$wet_pc_c05 <- as.numeric(congo_syn_meta$wet_pc_c05)
congo_syn_meta$wet_pc_u04 <- as.numeric(congo_syn_meta$wet_pc_u04)
congo_syn_meta$wet_pc_u05 <- as.numeric(congo_syn_meta$wet_pc_u05)
congo_syn_meta$for_pc_cse <- as.numeric(congo_syn_meta$for_pc_cse)
congo_syn_meta$for_pc_use <- as.numeric(congo_syn_meta$for_pc_use)

#code categorical variable land_cover as a numeric; 1 = forest, 2 = savanna, 3 = swamp
congo_syn_meta$land_Mann_num <- as.numeric(as.factor(congo_syn_meta$land_Mann))
# convert land_upstill to factor
congo_syn_meta$land_upstill<- as.factor(congo_syn_meta$land_upstill)
congo_syn_meta$land_upstill_num<- as.numeric(as.factor(congo_syn_meta$land_upstill)) # 1=forest, 2=savanna, 3=swamp

# combine wetland upstream variables
congo_syn_meta$wet_pc_uall <- congo_syn_meta$wet_pc_u04 + congo_syn_meta$wet_pc_u05


congo_syn_meta$log10_tt <- log10(1+congo_syn_meta$tt_res_hr)
congo_syn_meta$log10_tt_nores <- log10(1+congo_syn_meta$tt_nores_hr)
congo_syn_meta$log10_q_daily_mean <- log10(1+congo_syn_meta$q_daily_mean)
congo_syn_meta$log10_DIST_UP_KM <- log10(congo_syn_meta$DIST_UP_KM)
congo_syn_meta$log10_UPLAND_SKM <- log10(congo_syn_meta$UPLAND_SKM) # merit wrong for Ndoki CONGO_020, use hydrosheds UPLAND_SKM instead; see notes
congo_syn_meta$log10_DOC <- log10(congo_syn_meta$doc_mcc_l)
congo_syn_meta$log10_NOx <- log10(1+congo_syn_meta$Nox)
congo_syn_meta$log10_TDN <- log10(1+congo_syn_meta$TDN_mgN_l)
congo_syn_meta$log10_PO4 <- log10(1+congo_syn_meta$PO4)
congo_syn_meta$log10_SPC <- log10(1+congo_syn_meta$SPC)
congo_syn_meta$log10_suva_254 <- log10(1+congo_syn_meta$suva245_mgc_l_m)
congo_syn_meta$log10_do_mg_l <- log10(1+congo_syn_meta$do_mg_l)
congo_syn_meta$log10_CO2_pct_sat <- log10(1+congo_syn_meta$CO2_pct_sat)
congo_syn_meta$log10_CH4_pct_sat <- log10(1+congo_syn_meta$CH4_pct_sat)

# combine metadata and OTUs into one df
congo_syn_meta$sample.number <- row.names(congo_syn_meta)
meta_otu_congo_syn <- merge(congo_syn_meta, otu_congo_syn.t, by=0, all.x=TRUE)

congo_syn_meta_col <-ncol(congo_syn_meta)
otu_congo_syn_col <- ncol(otu_congo_syn.t)
otu_congo_syn_start <-congo_syn_meta_col+2 
otu_congo_syn_end <- ncol(meta_otu_congo_syn)-1

```

# PCoA-congo
```{r}
# PCoA wunifrac, must use rows as taxa OTU table
# uses all available synoptic Congo samples (n=23)
#meta_con_syn <- meta_otu_congo_syn[,c(1:congo_syn_meta_col)]
#otu_con_syn <- meta_otu_congo_syn[,c(otu_congo_syn_start:otu_congo_syn_end)]

# this uses all synoptic samples:
otu_congo_syn.m <- as.matrix(otu_congo_syn)

otu_congo_syn.wuni.test <- beta.div(otu_congo_syn.m, "unifrac", weighted=TRUE, tree=tree)

# alt use subset without NAs, that will also be used as input into co-inertias
meta_otu_congo_syn.ci<- subset(meta_otu_congo_syn, !is.na(ORD_STRA) & !is.na(log10_tt) & !is.na(log10_q_daily_mean) & !is.na(log10_DIST_UP_KM) & !is.na(log10_UPLAND_SKM) & !is.na(temp)
 & !is.na(pH) & !is.na(log10_SPC) & !is.na(log10_suva_254) & !is.na(log10_DOC) & !is.na(Sr) & !is.na(FI) & !is.na(do_pct) & !is.na(CO2_pct_sat) & !is.na(CH4_pct_sat) & !is.na(log10_NOx) & !is.na(log10_PO4))

congo_syn_meta.pcoa <- meta_otu_congo_syn.ci[,c(1:congo_syn_meta_col)]
otu_con_syn.pcoa <- meta_otu_congo_syn.ci[,c((otu_congo_syn_start-1):otu_congo_syn_end)]
row.names(otu_con_syn.pcoa)<-otu_con_syn.pcoa$sample.number.x
otu_con_syn.pcoa$sample.number.x <- NULL
otu_con_syn.pcoa.t <- t(otu_con_syn.pcoa)
otu_con_syn.pcoa.t.m <- as.matrix(otu_con_syn.pcoa.t)

otu_con_syn.pcoa.t.m.wuni <- beta.div(otu_con_syn.pcoa.t.m, "unifrac", weighted=TRUE, tree=tree)
#otu_congo_syn.wuni <- beta.div(otu_con_syn.t, "unifrac", weighted=TRUE, tree=tree_congo_syn)
pcoa.congo.syn.wuni <- pcoa (otu_con_syn.pcoa.t.m.wuni, correction="none", rn=NULL)


pcoa.congo.syn.wuni.comp <- pcoa.congo.syn.wuni$vectors
pcoa.congo.syn.wuni.1 <- -1*pcoa.congo.syn.wuni.comp[,1]
pcoa.congo.syn.wuni.2 <- -2*pcoa.congo.syn.wuni.comp[,2]

congo_syn_meta.pcoa <- congo_syn_meta.pcoa
congo_syn_meta.pcoa$pcoa.congo.syn.wuni.1 <- as.numeric(pcoa.congo.syn.wuni.1)
congo_syn_meta.pcoa$pcoa.congo.syn.wuni.2 <- as.numeric(pcoa.congo.syn.wuni.2)

# travel time
pdf("output/pcoa_congo_syn_wu_tt.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=log10_tt), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_continuous(low="blue", high="green")+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni-TT")

# water temp
pdf("output/pcoa_congo_syn_wu_temp.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=temp), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_continuous(low="yellow", high="red")+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni-tempC")

# order
order_color <- c("1"="#000080", "2"="#1E90FF", "3"="#008B8B", "4"="#008000", "5"="#FFFF00", "6"="#FFD700", "7"="#FF8C00", "8"="red", "9"="#FF1493")
pdf("output/pcoa_congo_syn_wu_order.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=ORD_STRA), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_manual(values=order_color)+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni-order")

# dominant landcover glc_cl_cmj
glc_color <- c("1"="#008000", "12" = "#FFD700", "13" = "#FFD700", "20" = "#000080", "7" = "#000080","15" = "#000080")

pdf("output/pcoa_congo_syn_wu_glc_cl_cmj.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=glc_cl_cmj), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_manual(values=glc_color)+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni- landcover_dominant")

# pH
pdf("output/pcoa_congo_syn_wu_pH.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=pH), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_continuous(low="blue", high="yellow")+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni-pH")

# DOC conc
pdf("output/pcoa_congo_syn_wu_doc.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=log10_DOC), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_continuous(low="blue", high="yellow")+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni-log10[DOC]")

# Sr
pdf("output/pcoa_congo_syn_wu_Sr.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=Sr), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_continuous(low="blue", high="yellow")+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni-Sr")

# SPC
pdf("output/pcoa_congo_syn_wu_spc.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=SPC), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_continuous(low="blue", high="yellow")+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni-SPC")

# NOx
pdf("output/pcoa_congo_syn_wu_NOx.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=log10_NOx), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_continuous(low="blue", high="green")+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni-NOx")

# TDN
pdf("output/pcoa_congo_syn_wu_TDN.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=TDN_mgN_l), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_continuous(low="blue", high="green")+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni-TDN")

# PO4
pdf("output/pcoa_congo_syn_wu_PO4.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=log10_PO4), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_continuous(low="yellow", high="red")+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni-PO4")

# latitude
pdf("output/pcoa_congo_syn_wu_lat.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=lat), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_continuous(low="blue", high="green")+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni-latitude")

# landcover - Mann
lc_colors <- c("savanna"="darkkhaki", "forest" = "green", "swamp" = "blue")

pdf("output/pcoa_congo_syn_wu_land_Mann.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=land_Mann), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_manual(values=lc_colors)+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni- landcover_Mann")

pdf("output/pcoa_congo_syn_wu_land_composite.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=land_composite), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_manual(values=lc_colors)+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni- landcover_composite")

pdf("output/pcoa_congo_syn_wu_land_glc_maj_up.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.pcoa, aes(pcoa.congo.syn.wuni.1, pcoa.congo.syn.wuni.2, label=site_id.1, color=glc_major_up), size=4, position=position_nudge(y=0.01), fontface="bold")+
  theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_manual(values=lc_colors)+
  #scale_x_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.congo.syn.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA Congo synoptic wuni- landcover_glc_major_upstream")


```

# congo synoptic - COA
```{r}
# all congo synoptic samples
otu_coa_congo_syn <- dudi.coa(otu_congo_syn.t, scannf=FALSE, nf=8)
```

# Correspondence analysis - all syn sites n=23
```{r}
# all synoptic sites
# don't subset to remove NAs, include as many synoptic sites as possible for COA examination (but will need to subset for CI)

congo_syn_meta.coa <- meta_otu_congo_syn[,c(1:congo_syn_meta_col)]
otu_con_syn.coa <- meta_otu_congo_syn[,c(otu_congo_syn_start:otu_congo_syn_end)]

# COA
otu_coa_congo_syn <- dudi.coa(otu_con_syn.coa, scannf=FALSE, nf=8)

# plot COA
# export COA coordinates
otu_coa_congo_syn_li <- as.data.frame(otu_coa_congo_syn$li)
coa_meta_congo_syn <- merge(otu_coa_congo_syn_li, congo_syn_meta.coa, by=0)
# plot
  # tt
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.all.tt.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn,aes(Axis1,Axis2,label=river, color=log10_tt), size=4, fontface="bold")+
  scale_color_gradient(low="blue", high="green") +
  theme_classic()+
  ggtitle("CA synoptic by TT all rivers n=23")+
  xlab("12.3% projected inertia") +
  ylab("10.8% projected inertia")
dev.off()

  # dendritic distance
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.all.dendist.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn,aes(Axis1,Axis2,label=river, color=log10_DIST_UP_KM), size=4, fontface="bold")+
  scale_color_gradient(low="blue", high="green") +
  theme_classic()+
  ggtitle("CA synoptic by dend distance all rivers n=23")+
  xlab("12.3% projected inertia") +
  ylab("10.8% projected inertia")
dev.off()

  # temp
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.all.temp.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn,aes(Axis1,Axis2,label=river, color=temp), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="red") +
  theme_classic()+
  ggtitle("CA synoptic by temp all rivers n=23")+
  xlab("12.3% projected inertia") +
  ylab("10.8% projected inertia")
dev.off()

  # doc
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.all.doc.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn,aes(Axis1,Axis2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="magenta") +
  theme_classic()+
  ggtitle("CA synoptic by DOC all rivers n=23")+
  xlab("12.3% projected inertia") +
  ylab("10.8% projected inertia")
dev.off()

  # TDN
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.all.TDN.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn,aes(Axis1,Axis2,label=river, color=log10_TDN), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="magenta") +
  theme_classic()+
  ggtitle("CA synoptic by TDN all rivers n=23")+
  xlab("12.3% projected inertia") +
  ylab("10.8% projected inertia")
dev.off()

  # NOx
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.all.NOx.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn,aes(Axis1,Axis2,label=river, color=log10_NOx), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="magenta") +
  theme_classic()+
  ggtitle("CA synoptic by NOx all rivers n=23")+
  xlab("12.3% projected inertia") +
  ylab("10.8% projected inertia")
dev.off()

  # pH
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.all.pH.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn,aes(Axis1,Axis2,label=river, color=pH), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="magenta") +
  theme_classic()+
  ggtitle("CA synoptic by pH all rivers n=23")+
  xlab("12.3% projected inertia") +
  ylab("10.8% projected inertia")
dev.off()
```


# congo synoptic - COA, PCA + co-inertia
# includes most sites (outliers Mopanzi etc)
```{r}
meta_otu_congo_syn.ci<- subset(meta_otu_congo_syn, !is.na(ORD_STRA) & !is.na(log10_tt) & !is.na(log10_q_daily_mean) & !is.na(log10_DIST_UP_KM) & !is.na(log10_UPLAND_SKM) & !is.na(temp)
 & !is.na(pH) & !is.na(log10_SPC) & !is.na(log10_suva_254) & !is.na(log10_DOC) & !is.na(Sr) & !is.na(FI) & !is.na(log10_do_mg_l) & !is.na(CO2_pct_sat) & !is.na(CH4_pct_sat) & !is.na(log10_NOx) & !is.na(log10_PO4))

congo_syn_meta.ci <- meta_otu_congo_syn.ci[,c(1:congo_syn_meta_col)]
otu_con_syn.ci <- meta_otu_congo_syn.ci[,c(otu_congo_syn_start:otu_congo_syn_end)]

# COA
otu_coa_congo_syn <- dudi.coa(otu_con_syn.ci, scannf=FALSE, nf=8)

# plot COA
# export COA coordinates
otu_coa_congo_syn_li <- as.data.frame(otu_coa_congo_syn$li)
coa_meta_congo_syn <- merge(otu_coa_congo_syn_li, congo_syn_meta.ci, by=0)
# plot
  # tt
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.all.tt.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn,aes(Axis1,Axis2,label=river, color=log10_tt), size=4, fontface="bold")+
  scale_color_gradient(low="blue", high="green") +
  theme_classic()+
  ggtitle("CA synoptic by TT all rivers")+
  xlab("13.2% projected inertia") +
  ylab("11.6% projected inertia")
dev.off()

  # dendritic distance
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.all.dendist.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn,aes(Axis1,Axis2,label=river, color=log10_DIST_UP_KM), size=4, fontface="bold")+
  scale_color_gradient(low="blue", high="green") +
  theme_classic()+
  ggtitle("CA synoptic by dend distance all rivers")+
  xlab("13.2% projected inertia") +
  ylab("11.6% projected inertia")
dev.off()

  # temp
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.all.temp.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn,aes(Axis1,Axis2,label=river, color=temp), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="red") +
  theme_classic()+
  ggtitle("CA synoptic by temp all rivers")+
  xlab("13.2% projected inertia") +
  ylab("11.6% projected inertia")
dev.off()

  # doc
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.all.doc.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn,aes(Axis1,Axis2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="magenta") +
  theme_classic()+
  ggtitle("CA synoptic by DOC all rivers")+
  xlab("13.2% projected inertia") +
  ylab("11.6% projected inertia")
dev.off()

# subset env variables of interest
congo_syn_meta.ci.pca <- congo_syn_meta.ci[,c("ORD_STRA", "log10_tt", "log10_q_daily_mean", "log10_DIST_UP_KM", "log10_UPLAND_SKM", "temp"
, "pH", "log10_SPC", "log10_suva_254", "log10_DOC", "Sr", "FI", "log10_do_mg_l", "CO2_pct_sat", "CH4_pct_sat", "log10_NOx", "log10_PO4")]
congo_syn_meta.ci.pca$ORD_STRA <- as.numeric(congo_syn_meta.ci.pca$ORD_STRA)


# PCA
env_pca_congo_syn <- dudi.pca(d=congo_syn_meta.ci.pca, row.w=otu_coa_congo_syn$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.1 <- coinertia(env_pca_congo_syn, otu_coa_congo_syn, scan = FALSE, nf = 5)

enviro_loadings_coin.1 <- as.data.frame(coin_con_syn.1$co)
enviro_loadings_coin.1$variable <- row.names(enviro_loadings_coin.1)

site_scores_coin_con_syn.1_0 <- as.data.frame(coin_con_syn.1$mX)
site_scores_coin_con_syn.1 <- bind_cols(site_scores_coin_con_syn.1_0, congo_syn_meta.ci)

pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.1.tt.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.1,aes(NorS1,NorS2,label=river, color=log10_tt), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.1, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.1, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("24.0% Projected Inertia")+
  ylab("13.9% Projected Inertia")+
  ggtitle("Congo Synoptic co-inertia.1, RV=0.57")
dev.off()


# remove Order.2
congo_syn_meta.ci.pca.2 <- congo_syn_meta.ci[,c("log10_tt", "log10_q_daily_mean", "log10_DIST_UP_KM", "log10_UPLAND_SKM", "temp"
, "pH", "log10_SPC", "log10_suva_254", "log10_DOC", "Sr", "FI", "log10_do_mg_l", "CO2_pct_sat", "CH4_pct_sat", "log10_NOx", "log10_PO4")]

# PCA
env_pca_congo_syn.2 <- dudi.pca(d=congo_syn_meta.ci.pca.2, row.w=otu_coa_congo_syn$lw, scale = TRUE, scannf = FALSE, nf = 10)
# co-inertia
coin_con_syn.2 <- coinertia(env_pca_congo_syn.2, otu_coa_congo_syn, scan = FALSE, nf = 5)

enviro_loadings_coin.2 <- as.data.frame(coin_con_syn.2$co)
enviro_loadings_coin.2$variable <- row.names(enviro_loadings_coin.2)

site_scores_coin_con_syn.2_0 <- as.data.frame(coin_con_syn.2$mX)
site_scores_coin_con_syn.2 <- bind_cols(site_scores_coin_con_syn.2_0, congo_syn_meta.ci)

# color by tt
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.2.ttnoDjiri.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.2,aes(NorS1,NorS2,label=river, color=log10_tt), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  #scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.2, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.2, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.9% Projected Inertia")+ # 0.569613 x  45.433 = 25.879
  ylab("11.3% Projected Inertia")+ # 0.569613 x 19.865  = 11.315
  ggtitle("Congo Synoptic co-inertia.2, no Djiri, RV=0.57")
dev.off()

# color by DOC
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.2.DOCnoDjiri.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.2,aes(NorS1,NorS2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="magenta")+
  theme_classic()+
  scale_y_reverse()+
  #scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.2, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.2, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.9% Projected Inertia")+ # 0.569613 x  45.433 = 25.879
  ylab("11.3% Projected Inertia")+ # 0.569613 x 19.865  = 11.315
  ggtitle("Congo Synoptic co-inertia.2, no Djiri, RV=0.57")
dev.off()


# tt only.3________________
congo_syn_meta.ci.pca.3 <- congo_syn_meta.ci[,c("log10_tt",  "temp" , "pH", "log10_SPC", "log10_suva_254", "log10_DOC", "Sr", "FI", "do_pct", "CO2_pct_sat", "CH4_pct_sat", "log10_NOx", "log10_PO4")]
# PCA
env_pca_congo_syn.3 <- dudi.pca(d=congo_syn_meta.ci.pca.3, row.w=otu_coa_congo_syn$lw, scale = TRUE, scannf = FALSE, nf = 10)
# co-inertia
coin_con_syn.3 <- coinertia(env_pca_congo_syn.3, otu_coa_congo_syn, scan = FALSE, nf = 5)

enviro_loadings_coin.3 <- as.data.frame(coin_con_syn.3$co)
enviro_loadings_coin.3$variable <- row.names(enviro_loadings_coin.3)

site_scores_coin_con_syn.3_0 <- as.data.frame(coin_con_syn.3$mX)
site_scores_coin_con_syn.3 <- bind_cols(site_scores_coin_con_syn.3_0, congo_syn_meta.ci)

#___tt color by tt
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.3.tt.noDjiri.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.3,aes(NorS1,NorS2,label=river, color=log10_tt), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.3, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.3, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.2% Projected Inertia")+
  ylab("10.9% Projected Inertia")+
  ggtitle("Congo Synoptic co-inertia.3, no Djiri, RV=0.58")
dev.off()

#___tt color by landcover - Mann
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.3.tt.noDjiri.landMann.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.3,aes(NorS1,NorS2,label=river, color=land_Mann), size=4, fontface="bold")+
  scale_color_manual(values=lc_colors)+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.3, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.3, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.2% Projected Inertia")+
  ylab("10.9% Projected Inertia")+
  ggtitle("Congo Synoptic co-inertia.3, no Djiri, RV=0.58")
dev.off()

#___tt color by landcover - composite
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.3.tt.noDjiri.landcomp.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.3,aes(NorS1,NorS2,label=river, color=land_composite), size=4, fontface="bold")+
  scale_color_manual(values=lc_colors)+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.3, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.3, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.2% Projected Inertia")+
  ylab("10.9% Projected Inertia")+
  ggtitle("Congo Synoptic co-inertia.3, no Djiri, RV=0.58")
dev.off()

#___tt color by landcover - glc_major_upstream
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.3.tt.noDjiri.landmajup.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.3,aes(NorS1,NorS2,label=river, color=glc_major_up), size=4, fontface="bold")+
  scale_color_manual(values=lc_colors)+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.3, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.3, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.2% Projected Inertia")+
  ylab("10.9% Projected Inertia")+
  ggtitle("Congo Synoptic co-inertia.3, no Djiri, RV=0.58")
dev.off()

# upland skm only.4__________
congo_syn_meta.ci.pca.4 <- congo_syn_meta.ci[,c("log10_UPLAND_SKM",  "temp" , "pH", "log10_SPC", "log10_suva_254", "log10_DOC", "Sr", "FI", "do_pct", "CO2_pct_sat", "CH4_pct_sat", "log10_NOx", "log10_PO4")]
# PCA
env_pca_congo_syn.4 <- dudi.pca(d=congo_syn_meta.ci.pca.4, row.w=otu_coa_congo_syn$lw, scale = TRUE, scannf = FALSE, nf = 10)
# co-inertia
coin_con_syn.4 <- coinertia(env_pca_congo_syn.4, otu_coa_congo_syn, scan = FALSE, nf = 5)

enviro_loadings_coin.4 <- as.data.frame(coin_con_syn.4$co)
enviro_loadings_coin.4$variable <- row.names(enviro_loadings_coin.4)

site_scores_coin_con_syn.4_0 <- as.data.frame(coin_con_syn.4$mX)
site_scores_coin_con_syn.4 <- bind_cols(site_scores_coin_con_syn.4_0, congo_syn_meta.ci)

pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.4.km2.noDjiri.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.4,aes(NorS1,NorS2,label=river, color=log10_UPLAND_SKM), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.4, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.4, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.7% Projected Inertia")+
  ylab("10.7% Projected Inertia")+
  ggtitle("Congo Synoptic co-inertia.4, no Djiri, RV=0.58")
dev.off()

# dendritic distance only.5__________
congo_syn_meta.ci.pca.5 <- congo_syn_meta.ci[,c("log10_DIST_UP_KM",  "temp" , "pH", "log10_SPC", "log10_suva_254", "log10_DOC", "Sr", "FI", "log10_do_mg_l", "CO2_pct_sat", "CH4_pct_sat", "log10_NOx", "log10_PO4")]
# PCA
env_pca_congo_syn.5 <- dudi.pca(d=congo_syn_meta.ci.pca.5, row.w=otu_coa_congo_syn$lw, scale = TRUE, scannf = FALSE, nf = 10)
# co-inertia
coin_con_syn.5 <- coinertia(env_pca_congo_syn.5, otu_coa_congo_syn, scan = FALSE, nf = 5)

enviro_loadings_coin.5 <- as.data.frame(coin_con_syn.5$co)
enviro_loadings_coin.5$variable <- row.names(enviro_loadings_coin.5)

site_scores_coin_con_syn.5_0 <- as.data.frame(coin_con_syn.5$mX)
site_scores_coin_con_syn.5 <- bind_cols(site_scores_coin_con_syn.5_0, congo_syn_meta.ci)

pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.5.dd.DOCnoDjiri.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.5,aes(NorS1,NorS2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="magenta")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.5, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.5, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.1% Projected Inertia")+ # 0.5806316 x 38.100
  ylab("10.9% Projected Inertia")+ # 0.5806316 x 18.824
  ggtitle("Congo Synoptic co-inertia.5, no Djiri, RV=0.58")
dev.off()
```


# Correspondence outliers removed (Mopanzi)
```{r}
# # Elim Lefini; do not subset to remove NA for FI; just eliminate it from the co-inertia, as it co-varys with Sr anyway. this preserves Eohala and Likoula aux Herbes; n=22
meta_otu_congo_syn.out1 <- subset(meta_otu_congo_syn, !is.na(log10_tt) & !is.na(log10_q_daily_mean) & !is.na(log10_DIST_UP_KM) & !is.na(log10_UPLAND_SKM) & !is.na(temp)
 & !is.na(pH) & !is.na(log10_suva_254) & !is.na(log10_DOC) & !is.na(Sr) & !is.na(log10_do_mg_l) & !is.na(log10_CO2_pct_sat) & !is.na(log10_CH4_pct_sat) & !is.na(log10_NOx) & !is.na(log10_PO4))

# eliminate Mopanzi outlier
# remove Mopanzi, major outlier
meta_otu_congo_syn.out1 <- subset(meta_otu_congo_syn.out1, Nox<400) # eliminates samples with NOx < 400 (and also Lefini as NA, which is ok its also missing many values); n=21 now no Lefini, no Mopanzi

congo_syn_meta.coa.out1 <- meta_otu_congo_syn.out1[,c(1:congo_syn_meta_col)]
otu_con_syn.coa.out1 <- meta_otu_congo_syn.out1[,c(otu_congo_syn_start:otu_congo_syn_end)]
# check zero occurrences
zero.otu_con_syn.coa.out1 <- colSums(otu_con_syn.coa.out1) == 0
otu_con_syn.coa.out1.nozero <- otu_con_syn.coa.out1[, !zero.otu_con_syn.coa.out1] # went from 3114 columns to 2706 columns

# COA
otu_coa_congo_syn.out1 <- dudi.coa(otu_con_syn.coa.out1.nozero, scannf=FALSE, nf=8)

# plot COA
# export COA coordinates
otu_coa_congo_syn.out1_li <- as.data.frame(otu_coa_congo_syn.out1$li)
coa_meta_congo_syn.out1 <- merge(otu_coa_congo_syn.out1_li, congo_syn_meta.coa.out1, by=0)
# plot
  # tt
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.tt.test.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=log10_tt), size=4, fontface="bold")+
  scale_color_gradient(low="blue", high="green") +
  theme_classic()+
  ggtitle("CA synoptic by TT no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

  # dendritic distance
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.dendist.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=log10_DIST_UP_KM), size=4, fontface="bold")+
  scale_color_gradient(low="blue", high="green") +
  theme_classic()+
  ggtitle("CA synoptic by dend distance no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

  # temp
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.temp.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=temp), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="red") +
  theme_classic()+
  ggtitle("CA synoptic by temp no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

  # doc
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.doc.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="magenta") +
  theme_classic()+
  ggtitle("CA synoptic by DOC no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

  # TDN
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.TDN.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=log10_TDN), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="magenta") +
  theme_classic()+
  ggtitle("CA synoptic by TDN no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

  # NOx
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.NOx.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=log10_NOx), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="magenta") +
  theme_classic()+
  ggtitle("CA synoptic by NOx no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

  # pH
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.pH.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=pH), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="magenta") +
  theme_classic()+
  ggtitle("CA synoptic by pH no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

  # land cover - Upstill Goddard
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.land.upstill.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=land_upstill), size=4, fontface="bold")+
  scale_color_manual(values=lc_colors)+
  theme_classic()+
  ggtitle("CA synoptic by land-Upstill no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

  # land cover - Mann
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.land.Mann.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=land_Mann), size=4, fontface="bold")+
  scale_color_manual(values=lc_colors)+
  theme_classic()+
  ggtitle("CA synoptic by land-Mann no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

  # land tec_cl_cmj = ecoregion; 63=western congolian forest-savanna; 26 = northwest congolian lowland forests; 29 = western congolian swamp forests
ecoregion_col <- c("63"="darkkhaki", "26" = "green", "29" = "blue")
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.land.tec.cl.cmj.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=tec_cl_cmj), size=4, fontface="bold")+
  scale_color_manual(values=ecoregion_col)+
  theme_classic()+
  ggtitle("CA synoptic by land-tec_cl_cmj no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()
  
# wet_pc_cg2 = catchment, percent of wetland class g2 (excludes, rivers, lakes reservoirs)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.wet.pc.cg2.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=wet_pc_cg2), size=4, fontface="bold")+
  scale_color_gradient(low="green", high="blue")+
  theme_classic()+
  ggtitle("CA synoptic by wet.pc.cg2 no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

# wet_pc_ug2 = watershed, percent of wetland class g2 (excludes, rivers, lakes reservoirs)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.wet.pc.ug2.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=wet_pc_ug2), size=4, fontface="bold")+
  scale_color_gradient(low="green", high="blue")+
  theme_classic()+
  ggtitle("CA synoptic by wet.pc.cg2 no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

# wet_pc_c04 = catchment, percent "freshwater marsh, floodplain"
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.wet.pc.c04.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=wet_pc_c04), size=4, fontface="bold")+
  scale_color_gradient(low="green", high="blue")+
  theme_classic()+
  ggtitle("CA synoptic by wet.pc.c04 no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

# wet_pc_u04 = watershed, percent "freshwater marsh, floodplain"
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.wet.pc.u04.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=wet_pc_u04), size=4, fontface="bold")+
  scale_color_gradient(low="green", high="blue")+
  theme_classic()+
  ggtitle("CA synoptic by wet.pc.u04 no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

# wet_pc_c05 = catchment, percent "swamp forest, flooded forest"
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.wet.pc.c05.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=wet_pc_c05), size=4, fontface="bold")+
  scale_color_gradient(low="green", high="blue")+
  theme_classic()+
  ggtitle("CA synoptic by wet.pc.c05 no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

# wet_pc_u05 = watershed, percent "swamp forest, flooded forest"
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.wet.pc.u05.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=wet_pc_u05), size=4, fontface="bold")+
  scale_color_gradient(low="green", high="blue")+
  theme_classic()+
  ggtitle("CA synoptic by wet.pc.u05 no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

# for_pc_cse = Forest cover, percent, catchment
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.for.pc.cse.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=for_pc_cse), size=4, fontface="bold")+
  scale_color_gradient(low="blue", high="yellow")+
  theme_classic()+
  ggtitle("CA synoptic by for.pc.cse no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()

# for_pc_use = Forest cover, percent, watershed
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.for.pc.use.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1,aes(Axis1,Axis2,label=river, color=for_pc_use), size=4, fontface="bold")+
  scale_color_gradient(low="blue", high="yellow")+
  theme_classic()+
  ggtitle("CA synoptic by for.pc.use no Mopanzi Lefini n=21")+
  xlab("13.0% projected inertia") +
  ylab("11.5% projected inertia")
dev.off()


##################
# COA with zero occurrences removed
#################
# check zero occurrences
zero.otu_con_syn.coa.out1 <- colSums(otu_con_syn.coa.out1) == 0
otu_con_syn.coa.out1.nozero <- otu_con_syn.coa.out1[, !zero.otu_con_syn.coa.out1] # went from 3118 columns to 2705 columns - so lots of species that are absent from these samples; should have used microViz!  but this zero_occurrances approach works too; but doesn't make a difference for COA plot (this looks same as coa.con.syn.out1.tt.test.pdf above)

# COA
otu_coa_congo_syn.out1.8 <- dudi.coa(otu_con_syn.coa.out1.nozero, scannf=FALSE, nf=8)

# plot COA
# export COA coordinates
otu_coa_congo_syn.out1.8_li <- as.data.frame(otu_coa_congo_syn.out1.8$li)
coa_meta_congo_syn.out1.8 <- merge(otu_coa_congo_syn.out1.8_li, congo_syn_meta.coa.out1, by=0)
# plot
  # tt
pdf("output/coa.con.syn.out1.8.tt.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1.8,aes(Axis1,Axis2,label=river, color=log10_tt), size=4, fontface="bold")+
  scale_color_gradient(low="blue", high="green") +
  theme_classic()+
  ggtitle("CA synoptic set 8 by TT no Mopanzi Lefini n=21")+
  xlab("13.1% projected inertia") +
  ylab("11.6% projected inertia")
dev.off()
```

# Set 8: co-inertia - all rivers with outliers removed (Mopanzi and Djiri)
```{r}
# next try co-inertia using COA otu_coa_congo_syn.out1 and a PCA that matches it at beginning of this subsection
# subset env variables of interest
congo_syn_meta.out1.pca <- congo_syn_meta.coa.out1[,c("log10_tt_nores", "log10_q_daily_mean", "log10_DIST_UP_KM", "log10_UPLAND_SKM", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_uall")]

congo_syn_meta.coa.out1.talk <- congo_syn_meta.coa.out1[,c("river", "ORD_STRA", "UPLAND_SKM", "tt_res_hr", "wet_pc_uall")]
write.csv(congo_syn_meta.coa.out1.talk, "output/set8.meta.ISME.talk.csv")
# PCA
env_pca_congo_syn.out1 <- dudi.pca(d=congo_syn_meta.out1.pca, row.w=otu_coa_congo_syn.out1$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1 <- coinertia(env_pca_congo_syn.out1, otu_coa_congo_syn.out1, scan = FALSE, nf = 5)

enviro_loadings_coin.out1 <- as.data.frame(coin_con_syn.out1$co)
enviro_loadings_coin.out1$variable <- row.names(enviro_loadings_coin.out1)

site_scores_coin_con_syn.out1_0 <- as.data.frame(coin_con_syn.out1$mX)
site_scores_coin_con_syn.out1 <- bind_cols(site_scores_coin_con_syn.out1_0, congo_syn_meta.coa.out1)

# tt color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.8.allhydro.tt.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1,aes(NorS1,NorS2,label=river, color=log10_tt), size=5)+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("28.3% Projected Inertia")+ # 0.5558849*50.896 = 28.29232
  ylab("10.1% Projected Inertia")+ # 0.5558849*18.094 = 10.05818
  ggtitle("Congo Synoptic Set8 coin.con.syn.out1.8.allhydro.tt.pdf RV=0.556")
dev.off()

# dendritic distance color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.8.dd.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1,aes(NorS1,NorS2,label=river, color=log10_DIST_UP_KM), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("28.4% Projected Inertia")+ # 0.5567725*50.955 = 28.37034
  ylab("10.1% Projected Inertia")+ # 0.5567725*18.075 = 10.06366
  ggtitle("Congo Synoptic Set8 coin.con.syn.out1.8.dd.pdf RV=0.557")
dev.off()

# DOC color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.8.DOC.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1,aes(NorS1,NorS2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="magenta")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("28.4% Projected Inertia")+ # 0.5567725*50.955 = 28.37034
  ylab("10.1% Projected Inertia")+ # 0.5567725*18.075 = 10.06366
  ggtitle("Congo Synoptic Set 8 coin.con.syn.out1.8.DOC.pdf, RV=0.557")
dev.off()

# pH
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.8.pH.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1,aes(NorS1,NorS2,label=river, color=pH), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="magenta")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("28.4% Projected Inertia")+ # 0.5567725*50.955 = 28.37034
  ylab("10.1% Projected Inertia")+ # 0.5567725*18.075 = 10.06366
  ggtitle("Congo Synoptic Set 8 coin.con.syn.out1.8.pH.pdf, RV=0.557")
dev.off()


# using single hydro variables - set 8.5
congo_syn_meta.out1.8.tt <- congo_syn_meta.coa.out1[,c("log10_tt_nores", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_uall")]

# PCA
env_pca_congo_syn.out1.8.tt <- dudi.pca(d=congo_syn_meta.out1.8.tt, row.w=otu_coa_congo_syn.out1$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.8.tt <- coinertia(env_pca_congo_syn.out1.8.tt, otu_coa_congo_syn.out1, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.8.tt <- as.data.frame(coin_con_syn.out1.8.tt$co)
enviro_loadings_coin.out1.8.tt$variable <- row.names(enviro_loadings_coin.out1.8.tt)

site_scores_coin_con_syn.out1.8.tt_0 <- as.data.frame(coin_con_syn.out1.8.tt$mX)
site_scores_coin_con_syn.out1.8.tt <- bind_cols(site_scores_coin_con_syn.out1.8.tt_0, congo_syn_meta.coa.out1)

# tt color
pdf("output/coin.con.syn.out1.8.tt.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.8.tt,aes(NorS1,NorS2,label=river, color=log10_tt), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.8.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.8.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("26.3% Projected Inertia")+ # 0.5848485 *44.849 = 26.22987
  ylab("10.2% Projected Inertia")+ # 0.5848485*17.412 = 10.18338
  ggtitle("Congo Synoptic co-inertia.outliers1.8.tt, RV=0.585")
dev.off()
# tt color - text edits
pdf("output/coin.con.syn.out1.8.tt.font6.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.8.tt,aes(NorS1,NorS2,label=river, color=log10_tt), size=6)+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.8.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.8.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("26.3% Projected Inertia")+ # 0.5848485 *44.849 = 26.22987
  ylab("10.2% Projected Inertia")+ # 0.5848485*17.412 = 10.18338
  ggtitle("Congo Synoptic co-inertia.outliers1.8.tt, RV=0.585")
dev.off()

# pH color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.8.pH.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.8.tt,aes(NorS1,NorS2,label=river, color=pH), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="yellow")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.8.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.8.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
 xlab("25.1 Projected Inertia")+ # 0.5905048 *42.487 = 25.08878
  ylab("11.2% Projected Inertia")+ # 0.5905048*18.918 = 11.17117
  ggtitle("Congo Synoptic co-inertia.outliers1.8.pH, RV=0.5905")
dev.off()

# DOC
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.8.DOC.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.8.tt,aes(NorS1,NorS2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="yellow")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.8.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.8.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.1 Projected Inertia")+ # 0.5905048 *42.487 = 25.08878
  ylab("11.2% Projected Inertia")+ # 0.5905048*18.918 = 11.17117
  ggtitle("Congo Synoptic co-inertia.outliers1.8.DOC, RV=0.5905")
dev.off()

## Set 8.1.1: PCoA using wunifrac distances - with and without Euclidean correction of the distance matrix (cailliez); + PCoA using CI test
# original CA: otu_coa_congo_syn.out1; original OTU table (n=21): otu_con_syn.coa.out1; we know this contains extra OTUs that have all zeroes bc it was not properly removed via microviz or zero colsums. but here running a test for now to compare this PCoA and CI with the original one based on CA (coin.con.syn.out1.8.tt.pdf)

# ape PCoA
# samples=columns, OTUs=rows
otu_con_syn.coa.out1.t <- t(otu_con_syn.coa.out1)
otu.8.wuni <- beta.div(otu_con_syn.coa.out1.t, "unifrac", weighted=TRUE, tree=tree)
is.euclid(otu.8.wuni) # false

pcoa.8.wuni <- pcoa(otu.8.wuni, correction="none", rn=NULL)
pcoa.8.wuni.comp <- pcoa.8.wuni$vectors
pcoa.8.wuni.1 <- -1*pcoa.8.wuni.comp[,1]
pcoa.8.wuni.2 <- -2*pcoa.8.wuni.comp[,2]

congo_syn_meta.coa.out1.8.pcoa <- congo_syn_meta.coa.out1

congo_syn_meta.coa.out1.8.pcoa$pcoa.8.wuni.1 <- as.numeric(pcoa.8.wuni.1)
congo_syn_meta.coa.out1.8.pcoa$pcoa.8.wuni.2 <- as.numeric(pcoa.8.wuni.2)

# tt - 8. wuni using ape no correction
pdf("output/pcoa.8.1.1.wuni.tt.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.coa.out1.8.pcoa, aes(pcoa.8.wuni.1, pcoa.8.wuni.2, label=site_id.1, color=log10_tt), size=4, position=position_nudge(y=0.003), fontface="bold")+
  theme_classic()+
  scale_color_continuous(low="blue", high="green")+
  #scale_fill_brewer(palette="Set1")+
  #scale_x_reverse()+
  #scale_y_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.8.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.8.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA set 8 wuni ASV - ape no correction")

print(pcoa.8.wuni$values$Eigenvalues) # only last 2 dimensions (20 + 21) have negative eigenvalues, which likely does not matter or affect the euclidean nature; strictly speaking the entire distance matrix is non-euclidean, but it effectively is. see: https://uw.pressbooks.pub/appliedmultivariatestatistics/chapter/pcoa/

# tt -8.1.2 - wuni using ape w/ cailliez correction
pcoa.8.wuni.cailliez <- pcoa(otu.8.wuni, correction="cailliez", rn=NULL)
pcoa.8.wuni.cailliez.comp <- pcoa.8.wuni.cailliez$vectors
pcoa.8.wuni.cailliez.1 <- -1*pcoa.8.wuni.cailliez.comp[,1]
pcoa.8.wuni.cailliez.2 <- -2*pcoa.8.wuni.cailliez.comp[,2]

congo_syn_meta.coa.out1.8.pcoa$pcoa.8.wuni.cailliez.1 <- as.numeric(pcoa.8.wuni.cailliez.1)
congo_syn_meta.coa.out1.8.pcoa$pcoa.8.wuni.cailliez.2 <- as.numeric(pcoa.8.wuni.cailliez.2)

# tt - 8.1.2 wuni using ape w/cailliez correction
pdf("output/pcoa.8.1.2.wuni.tt.cailliez.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.coa.out1.8.pcoa, aes(pcoa.8.wuni.cailliez.1, pcoa.8.wuni.cailliez.2, label=site_id.1, color=log10_tt), size=4, position=position_nudge(y=0.003), fontface="bold")+
  theme_classic()+
  scale_color_continuous(low="blue", high="green")+
  #scale_fill_brewer(palette="Set1")+
  #scale_x_reverse()+
  #scale_y_reverse()+
  xlab(paste("PCoA-1 (",round(pcoa.8.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.8.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA set 8 wuni ASV - ape w/cailliez correction")
# save as above but with text edits
pdf("output/pcoa.8.1.2.wuni.tt.cailliez.font5.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.coa.out1.8.pcoa, aes(pcoa.8.wuni.cailliez.1, pcoa.8.wuni.cailliez.2, label=site_id.1, color=log10_tt), size=5, position=position_nudge(y=0.003))+
  theme_classic()+
  scale_color_continuous(low="blue", high="green")+
  #scale_fill_brewer(palette="Set1")+
  #scale_x_reverse()+
  #scale_y_reverse()+
  xlim(-0.25, 0.2)+
  xlab(paste("PCoA-1 (",round(pcoa.8.wuni$values$Relative_eig,digits=3)[1]*100,"% variance explained)",sep="")) +
  ylab(paste("PCoA-2,(",round(pcoa.8.wuni$values$Relative_eig,digits=3)[2]*100,"% variance explained)",sep="")) +
              ggtitle("PCoA set 8 wuni ASV - ape w/cailliez correction")


# tt - 8.1.3 - wuni using dudi.pco w/cailliez correction on unifrac matrix first 
# test if weighted unifrac distance matrix is euclidean (required for PCoA)
is.euclid(otu.8.wuni) # false
# convert wuni to euclidean via transformation
otu.8.wuni.euc <- cailliez(otu.8.wuni)
is.euclid(otu.8.wuni.euc) # true

pcoa.8.wuni.dudi <- dudi.pco(otu.8.wuni.euc, scannf=FALSE)
# pull out coordinates
pcoa.8.wuni.dudi.li <- as.data.frame(pcoa.8.wuni.dudi$li)
pcoa.8.wuni.dudi.li.1 <- -1*pcoa.8.wuni.dudi.li[,1]
pcoa.8.wuni.dudi.li.2 <- -2*pcoa.8.wuni.dudi.li[,2]

congo_syn_meta.coa.out1.8.pcoa$pcoa.8.wuni.dudi.1 <- as.numeric(pcoa.8.wuni.dudi.li.1)
congo_syn_meta.coa.out1.8.pcoa$pcoa.8.wuni.dudi.2 <- as.numeric(pcoa.8.wuni.dudi.li.2)

# tt - wuni dudi.pco cailliez
pdf("output/pcoa.8.1.3wuni.tt.dudi.euc.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.coa.out1.8.pcoa, aes(pcoa.8.wuni.dudi.1, pcoa.8.wuni.dudi.2, label=site_id.1, color=log10_tt), size=4, position=position_nudge(y=0.003), fontface="bold")+
              theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_continuous(low="blue", high="green")+
  #scale_x_reverse()+
  #scale_y_reverse()+
  xlab("31.0% Projected Inertia")+ 
  ylab("24.2% Projected Inertia")+ 
  ggtitle("PCoA set 8 wuni ASV - dudi.pco cailliez trans")
dev.off()

print(pcoa.8.wuni.dudi$eig)

# co-inertia using PCoA as input - 8.1.3
# make new PCA with weighted PCoA row weights
env_pca_congo_syn.out1.8.1.3.tt <- dudi.pca(d=congo_syn_meta.out1.8.tt, row.w=pcoa.8.wuni.dudi$lw, scale=TRUE, scannf=FALSE, nf=10)
# co-inertia
coin_con_syn.out1.8.1.3.tt <- coinertia(env_pca_congo_syn.out1.8.1.3.tt, pcoa.8.wuni.dudi, scan=FALSE, nf=5)

enviro_loadings_coin.out1.8.1.3.tt <- as.data.frame(coin_con_syn.out1.8.1.3.tt$co)
enviro_loadings_coin.out1.8.1.3.tt$variable <- row.names(enviro_loadings_coin.out1.8.1.3.tt)

site_scores_coin_con_syn.out1.8.1.3.tt_0 <- as.data.frame(coin_con_syn.out1.8.1.3.tt$mX)
site_scores_coin_con_syn.out1.8.1.3.tt <- bind_cols(site_scores_coin_con_syn.out1.8.1.3.tt_0, congo_syn_meta.coa.out1)

pdf("output/coin.con.syn.out1.8.1.3.tt.pcoa.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.8.1.3.tt,aes(NorS1,NorS2,label=site_id.1, color=log10_tt), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  #scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.8.1.3.tt, aes(x=0, y=0, xend=Comp1*30, yend=Comp2*30), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.8.1.3.tt, aes(Comp1*30, Comp2*30, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("26.7% Projected Inertia")+ # 0.4389187*60.913 = 26.73585
  ylab("7.4% Projected Inertia")+ # 0.4389187*16.894 = 7.415093
  ggtitle("Set 8-TT coin_con_syn.out1.8.1.3.tt w/PCoA  RV=0.439")
dev.off()


# log10_q_daily_mean
congo_syn_meta.out1.8.q <- congo_syn_meta.coa.out1[,c("log10_q_daily_mean", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4")]

# PCA
env_pca_congo_syn.out1.8.q <- dudi.pca(d=congo_syn_meta.out1.8.q, row.w=otu_coa_congo_syn.out1$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.8.q <- coinertia(env_pca_congo_syn.out1.8.q, otu_coa_congo_syn.out1, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.8.q <- as.data.frame(coin_con_syn.out1.8.q$co)
enviro_loadings_coin.out1.8.q$variable <- row.names(enviro_loadings_coin.out1.8.q)

site_scores_coin_con_syn.out1.8.q_0 <- as.data.frame(coin_con_syn.out1.8.q$mX)
site_scores_coin_con_syn.out1.8.q <- bind_cols(site_scores_coin_con_syn.out1.8.q_0, congo_syn_meta.coa.out1)

# q color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.8.q.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.8.tt,aes(NorS1,NorS2,label=river, color=log10_q_daily_mean), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.8.q, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.8.q, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("24.9 Projected Inertia")+ # 0.5686331*43.856 = 24.93797
  ylab("10.9% Projected Inertia")+ # 0.5686331*19.088 = 10.85407
  ggtitle("Congo Synoptic co-inertia.outliers1.8.q_daily_mean, RV=0.5686")
dev.off()


# log10_DIST_UP_KM
congo_syn_meta.out1.8.dd <- congo_syn_meta.coa.out1[,c("log10_DIST_UP_KM", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4")]

# PCA
env_pca_congo_syn.out1.8.dd <- dudi.pca(d=congo_syn_meta.out1.8.dd, row.w=otu_coa_congo_syn.out1$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.8.dd <- coinertia(env_pca_congo_syn.out1.8.dd, otu_coa_congo_syn.out1, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.8.dd <- as.data.frame(coin_con_syn.out1.8.dd$co)
enviro_loadings_coin.out1.8.dd$variable <- row.names(enviro_loadings_coin.out1.8.dd)

site_scores_coin_con_syn.out1.8.dd_0 <- as.data.frame(coin_con_syn.out1.8.dd$mX)
site_scores_coin_con_syn.out1.8.dd <- bind_cols(site_scores_coin_con_syn.out1.8.dd_0, congo_syn_meta.coa.out1)

# dd color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.8.dd.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.8.tt,aes(NorS1,NorS2,label=river, color=log10_DIST_UP_KM), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.8.dd, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.8.dd, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.5Projected Inertia")+ #  0.5913856*43.047 = 25.45738
  ylab("11.1% Projected Inertia")+ #  0.5913856*18.842 = 11.14289
  ggtitle("Congo Synoptic co-inertia.outliers1.8.dendritic_dist, RV=0.5914")
dev.off()


# log10_UPLAND_SKM
congo_syn_meta.out1.8.km2 <- congo_syn_meta.coa.out1[,c("log10_UPLAND_SKM", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4")]

# PCA
env_pca_congo_syn.out1.8.km2 <- dudi.pca(d=congo_syn_meta.out1.8.km2, row.w=otu_coa_congo_syn.out1$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.8.km2 <- coinertia(env_pca_congo_syn.out1.8.km2, otu_coa_congo_syn.out1, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.8.km2 <- as.data.frame(coin_con_syn.out1.8.km2$co)
enviro_loadings_coin.out1.8.km2$variable <- row.names(enviro_loadings_coin.out1.8.km2)

site_scores_coin_con_syn.out1.8.km2_0 <- as.data.frame(coin_con_syn.out1.8.km2$mX)
site_scores_coin_con_syn.out1.8.km2 <- bind_cols(site_scores_coin_con_syn.out1.8.km2_0, congo_syn_meta.coa.out1)

# dd color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.8.km2.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.8.tt,aes(NorS1,NorS2,label=river, color=log10_UPLAND_SKM), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.8.km2, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.8.km2, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.6 Projected Inertia")+ #  0.5906178*43.299 = 25.57316
  ylab("11.0% Projected Inertia")+ #  0.5906178*18.545 = 10.95301
  ggtitle("Congo Synoptic co-inertia.outliers1.8.area_km2, RV=0.5906")
dev.off()

# indicator species on big vs small rivers
congo.syn.meta.out1.8.indic <- congo_syn_meta.coa.out1

# tt
  # histograms
hist(congo.syn.meta.out1.8.indic$tt_res_hr, main="Histogram: tt_res_hr sample distribution", xlab="tt_res (hr)", col="blue", las=1, breaks=30)
hist(congo.syn.meta.out1.8.indic$log10_tt, main="Histogram: log10_tt_res_hr sample distribution", xlab="log10_tt_res (hr)", col="blue", las=1, breaks=30)
  # hi-low
congo.syn.meta.out1.8.indic$tt_high_low <- ifelse(congo.syn.meta.out1.8.indic$tt_res_hr <=168, "low",
                                      ifelse(congo.syn.meta.out1.8.indic$tt_res_hr >=168, "high", "null"))
# must set OTU table to RA data
# use original phyloseq object to eliminate samples from n=23 to n=21
library(microViz)
congo_syn.8 <- ps_filter(congo_syn, sample_id!="CONGO_003", sample_id!="CONGO_019") # elim Mopanzi, Djiri already eliminated;  matches set 9 here
# convert OTU data to relative abundance
library(phylosmith)
congo_syn.8.ra<-relative_abundance(congo_syn.8, sig_fig=6)
otu_congo_syn.8.ra <- otu_table(congo_syn.8.ra)
write.csv(otu_congo_syn.8.ra, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/otu_congo_syn.8.ra.csv")
otu_congo_syn.8.ra.t <- as.data.frame(t(otu_congo_syn.8.ra)) # transpose OTU table, so rows=samples, columns=OTUs. This is input format for indicator species
write.csv(otu_congo_syn.8.ra.t, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/otu_congo_syn.8.ra.t.csv")

  # indicator
indic.8.tt.big.small <- multipatt(otu_congo_syn.8.ra.t, congo.syn.meta.out1.8.indic$tt_high_low, control=how(nperm=999))
indic.8.tt.big.small.results <- as.data.frame(indic.8.tt.big.small$sign)
indic.8.tt.big.small.results.2 <- subset(indic.8.tt.big.small.results, p.value<0.05&!is.na(p.value))
indic.8.tt.big.small.results.2$OTU_ID2 <- row.names(indic.8.tt.big.small.results.2)

indic.8.tt.big.small.results.2.tax <- merge(indic.8.tt.big.small.results.2,tax,by="OTU_ID2")
write.csv(indic.8.tt.big.small.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.8.tt.big.small.results.2.tax.csv")

summary(indic.8.tt.big.small, indvalcomp=TRUE)

# indicator re-do; above code uses "double-rarified" data from community_rare3 and then computed RA again
otu.congo_syn.8 <- otu_table(congo_syn.8)
otu.congo_syn.8.t <- t(otu.congo_syn.8)
otu.congo_syn.8.t <- as.data.frame(otu.congo_syn.8.t)
  # indicator
library(indicspecies)
indic.8.tt.big.small.3 <- multipatt(otu.congo_syn.8.t, congo.syn.meta.out1.8.indic$tt_high_low, control=how(nperm=999))
indic.8.tt.big.small.3.results <- as.data.frame(indic.8.tt.big.small.3$sign)
indic.8.tt.big.small.3.results.2 <- subset(indic.8.tt.big.small.3.results, p.value<0.05&!is.na(p.value))
indic.8.tt.big.small.3.results.2$OTU_ID2 <- row.names(indic.8.tt.big.small.3.results.2) 

indic.8.tt.big.small.3.results.2.tax <- merge(indic.8.tt.big.small.3.results.2,tax,by="OTU_ID2")
write.csv(indic.8.tt.big.small.3.results.2.tax, "output/indic.8.tt.big.small.3.results.2.tax.csv")

summary(indic.8.tt.big.small.3, indvalcomp=TRUE)
```

# set 8: indicator spp using labdsv (alt R package, used in Kellog et al)
```{r}
# indicator spp using labdsv (alt R package, used in Kellog et al)
library(labdsv)
otu_con_syn.coa.out2 <- meta_otu_congo_syn.out1[,c((otu_congo_syn_start-1):otu_congo_syn_end)] # same OTU table we've been using, but keep a column for sample names
rownames(otu_con_syn.coa.out2) <- otu_con_syn.coa.out2[,1] # assign first column as row names
otu_con_syn.coa.out2 <- otu_con_syn.coa.out2[,-1] # remove column 1 containing row names; otu table matches otu table prev extracted from rarified phyloseq object above; n=21
# check rowsums
otu_con_syn.coa.out2$row.sums <- rowSums(otu_con_syn.coa.out2) # all add to 9384
otu_con_syn.coa.out2 <- otu_con_syn.coa.out2[, -ncol(otu_con_syn.coa.out2)] # remove row sums

# metadata file from above: congo_syn_meta.out1.8.tt
congo_syn_meta.out1.8.tt.labdsv <- congo_syn_meta.coa.out1
rownames(congo_syn_meta.out1.8.tt.labdsv) <- congo_syn_meta.out1.8.tt.labdsv[,1]
congo_syn_meta.out1.8.tt.labdsv <- congo_syn_meta.out1.8.tt.labdsv[,-1]
# assign groups
congo_syn_meta.out1.8.tt.labdsv$tt_group <- ifelse(congo_syn_meta.out1.8.tt.labdsv$tt_res_hr <=168, "low",
                                      ifelse(congo_syn_meta.out1.8.tt.labdsv$tt_res_hr >=168, "high", "null"))
# make numeric vector for tt_group
#congo_syn_meta.out1.8.tt.labdsv$tt_group <- as.factor(congo_syn_meta.out1.8.tt.labdsv$tt_group)
library(dplyr)
tt.8.groups <- recode(congo_syn_meta.out1.8.tt.labdsv$tt_group, "low"=1, "high"=2)
# checks
length(tt.8.groups) == nrow(otu_con_syn.coa.out2) # dimensions TRUE
zero_occurances.8 <- colSums(otu_con_syn.coa.out2) == 0
otu_con_syn.coa.out2 <- otu_con_syn.coa.out2[, !zero_occurances.8] # from 3114 to 2700 columns - so lots of species that are absent from these samples; should have used microViz!  but this zero_occurrances approach works too
otu_con_syn.coa.out2$row.sums <- rowSums(otu_con_syn.coa.out2) # all add to 9384
otu_con_syn.coa.out2 <- otu_con_syn.coa.out2[, -ncol(otu_con_syn.coa.out2)] # remove row sums

indic.labdsv.8.tt<- indval(otu_con_syn.coa.out2, tt.8.groups, numitr=999)

summary(indic.labdsv.8.tt, type="short") # 148 sig indicators; group 1(low) = 3 ASVs, group 2(high) = 145 ASVs

# save results
indic.labdsv.8.tt.res <- data.frame(
 # ASV = rownames(indic.labdsv.8.tt$indval),
  relfrq = indic.labdsv.8.tt$relfrq,
  relabu = indic.labdsv.8.tt$relabu,
  indval = indic.labdsv.8.tt$indval,
  pval = indic.labdsv.8.tt$pval
)
write.csv(indic.labdsv.8.tt.res, "output/indic.labdsv.8.tt.res.csv")

# compare ASVs in high vs low TT rivers
# TT high
sel.ind.tt.hi <- which(congo_syn_meta.out1.8.tt.labdsv$tt_group == "high") # rows 11 16 21
tt.otu.hi.sam.8 <- otu_con_syn.coa.out2[sel.ind.tt.hi, ] # 2701 ASVs
zero.occ.tt.otu.hi.sam.8 <- colSums(tt.otu.hi.sam.8) == 0
tt.otu.hi.sam.8 <- tt.otu.hi.sam.8[, !zero.occ.tt.otu.hi.sam.8] # 788 ASVs
# TT low
sel.ind.tt.low <- which(congo_syn_meta.out1.8.tt.labdsv$tt_group == "low")
tt.otu.low.sam.8 <- otu_con_syn.coa.out2[sel.ind.tt.low, ] # 2701 ASVs
zero.occ.tt.otu.low.sam.8 <- colSums(tt.otu.low.sam.8) == 0
tt.otu.low.sam.8 <- tt.otu.low.sam.8[, !zero.occ.tt.otu.low.sam.8] # 2211 ASVs
# compare TT ASVs
tt.hi.otus <- colnames(tt.otu.hi.sam.8) # headers for high OTUs
tt.low.otus <- colnames(tt.otu.low.sam.8) # headers for low OTUs
tt.hi.unique <- setdiff(tt.hi.otus, tt.low.otus) # unique OTUs in high-TT # 490
tt.low.unique <- setdiff(tt.low.otus, tt.hi.otus) # unique OTUs in low-TT # 1913
tt.common <- intersect(tt.hi.otus, tt.low.otus) # 298 common OTUs
# assign tax and output
tt.hi.unique.df <- data.frame(OTU_ID2 = tt.hi.unique) # convert list to df
tt.hi.unique.df.tax <- merge(tt.hi.unique.df, tax, by="OTU_ID2")
write.csv(tt.hi.unique.df.tax, "output/tt.hi.unique.df.tax.csv")

tt.low.unique.df <- data.frame(OTU_ID2 = tt.low.unique)
tt.low.unique.df.tax <- merge(tt.low.unique.df, tax, by="OTU_ID2")
write.csv(tt.low.unique.df.tax, "output/tt.low.unique.df.tax.csv")

tt.common.df <- data.frame(OTU_ID2 = tt.common)
tt.common.df.tax <- merge(tt.common.df, tax, by="OTU_ID2")
write.csv(tt.common.df.tax, "output/tt.common.df.tax.csv")



ald.8.tt.hilo.all.results <- ald.8.tt.hilo.all
ald.8.tt.hilo.all.results$OTU_ID2 <- row.names(ald.8.tt.hilo.all.results)
ald.8.tt.hilo.all.results.tax <- merge (ald.8.tt.hilo.all.results, tax, by="OTU_ID2")
write.csv(ald.8.tt.hilo.all.results.tax, "output/ald.8.tt.hilo.all.results.tax.csv")

## match our sig ASVs with their DNA sequence, create new fata file
# dna-sequences.fasta is a qiime2 export from ct_mek_con_merged_rep_seqs_bac_arc_20230908.qza
# SILVA taxonomy
library(seqinr)
fasta.file <-  "../../ct_qiime_100/dna-sequences.fasta"
dna.seq <- read.fasta(file = fasta.file)
# otu list for set 8.tt
set.8.indic.tt.ASVlist.file <- "output/set.8.indic.tt.ASVlist.csv"
set.8.indic.tt.ASVlist <- readLines(set.8.indic.tt.ASVlist.file)
# Extract sequences for specific OTUs
set.8.indic.tt.ASV.seq <- lapply(set.8.indic.tt.ASVlist, function(otu) {
  if (otu %in% names(dna.seq)) {
    seq <- dna.seq[[otu]]
    return(toupper(seq)) # ensure uppercase sequence
  } else {
    return(NULL)
  }
})
# Remove NULL entries
set.8.indic.tt.ASV.seq <- set.8.indic.tt.ASV.seq[!sapply(set.8.indic.tt.ASV.seq, is.null)]
# write extracted set8.tt sequences into a new fasta file
set.8.out.fasta <- "output/set.8.indic.tt.ASV.seq.fasta"
write.fasta(sequences=set.8.indic.tt.ASV.seq, names=set.8.indic.tt.ASVlist, file.out=set.8.out.fasta)

# GTDB taxonomy
fasta.file.gtdb <- "../../ct_qiime_100_GTDB_tax/dna-sequences-GTDB.fasta"
dna.seq.gtdb <- read.fasta(file=fasta.file.gtdb)
# otu list provided above: set.8.indic.tt.ASVlist
# Extract sequences for specific OTUs
set.8.indic.tt.ASV.seq.gtdb <- lapply(set.8.indic.tt.ASVlist, function(otu) {
  if (otu %in% names(dna.seq)) {
    seq <- dna.seq.gtdb[[otu]]
    return(toupper(seq)) # ensure uppercase sequence
  } else {
    return(NULL)
  }
})

```


```{r}
```



# Set 8 - differential abundance with ALDEx2 & DESeq2
```{r}
library(ALDEx2)
# go back to original phyloseq object, before rarefying: community3 <-ps_filter(community, sample_id!="ME060025", s
# remove all samples with reads sample.size < 9383
sort(sample_sums(community3))
com_4_unrare <- prune_samples(sample_sums(community3) > 9383, community3)

# get congo synoptics
com_4_unrare <- ps_filter(com_4_unrare, project=="congo", sam_type == "synoptic") # n=23
sort(sample_sums(com_4_unrare))
# remove large rivers + outliers
com_8_unrare <- ps_filter(com_4_unrare, site_id.1!="Lefini", site_id.1!="Mopanzi") # n=18, samples match set 9


# pull out OTU
otu_com_8 <- otu_table(com_8_unrare)
otu_com_8 <- as.data.frame(otu_com_8) # rows=taxa, columns=samples

# pull out groups 
meta_com_8 <- data.frame(sample_data(com_8_unrare))
meta_com_8$sample_number <- row.names(meta_com_8)

meta_com_8$tt_res_hr <- as.numeric(meta_com_8$tt_res_hr)
meta_com_8$log10_tt <- log10(1+meta_com_8$tt_res_hr)

# assign hi-lo for TTas in code above
meta_com_8$tt_hilo <- ifelse(meta_com_8$tt_res_hr <=192, "low",
                                             ifelse(meta_com_8$tt_res_hr >=192, "high", "null"))

cond_com_8_tt_hilo <- meta_com_8$tt_hilo

# make sure your conditions/group order in metadata matches those in the OTU table
ald.8.tt.hilo <- aldex(reads=otu_com_8, conditions = cond_com_8_tt_hilo, mc.samples = 128, test="t", effect=TRUE,
             	include.sample.summary = FALSE, verbose=T, denom="all", paired.test=FALSE)


aldex.plot(ald.8.tt.hilo, type="volcano", test="welch", xlab="Difference",
    ylab="-1(log10(q))", main='Volcano plot') 

# modular, step by step approach
ald.8.tt.hilo.clr <- aldex.clr(otu_com_8, cond_com_8_tt_hilo, mc.samples=128, verbose=TRUE, denom="all", gamma=1e-3)
# t-test - Welch's and Wilcoxon
ald.8.tt.hilo.tt <- aldex.ttest(ald.8.tt.hilo.clr, hist.plot=TRUE, paired.test=FALSE, verbose=TRUE)
# effect sizes for within and between groups
ald.8.tt.hilo.eff <- aldex.effect(ald.8.tt.hilo.clr, CI=TRUE, verbose=TRUE, include.sample.summary=TRUE, paired.test=TRUE)
# aldex plot
ald.8.tt.hilo.all <- data.frame(ald.8.tt.hilo.tt, ald.8.tt.hilo.eff)

pdf("output/ald.8.tt.hilo.all.pdf")
par(mfrow=c(1,3))
aldex.plot(ald.8.tt.hilo.all, type="MA", test="welch", main="MA plot")
aldex.plot(ald.8.tt.hilo.all, type="MW", test="welch", main="effect plot")
aldex.plot(ald.8.tt.hilo.all, type="volcano", test="welch", main="volcano plot")
dev.off()

# save results
ald.8.tt.hilo.all.results <- ald.8.tt.hilo.all
ald.8.tt.hilo.all.results$OTU_ID2 <- row.names(ald.8.tt.hilo.all.results)
ald.8.tt.hilo.all.results.tax <- merge (ald.8.tt.hilo.all.results, tax, by="OTU_ID2")
write.csv(ald.8.tt.hilo.all.results.tax, "output/ald.8.tt.hilo.all.results.tax.csv")

########
# DESeq2 via phyloseq ---- https://www.bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html
#######
# add high-low TT variable in phyloseq object
sample_data(com_8_unrare) <- sample_data(meta_com_8)
# convert to deseq2 object
library(DESeq2)
dsq.com.8.tt <- phyloseq_to_deseq2(com_8_unrare, ~ tt_hilo)
# set control/reference vs. treatment group; by default it does this alphabetically, but because "high" comes before "low", that means high=ref, low=treat. We want the opposite, so we get LFC wrt the low group.
dsq.com.8.tt$tt_hilo <- factor(dsq.com.8.tt$tt_hilo, levels = c("low", "high"))

# examine sparsity of data
dsq.com.8.tt.rs <- rowSums(counts(dsq.com.8.tt)) # get row sums of each count (ASV)
dsq.com.8.tt.rmx <- apply(counts(dsq.com.8.tt), 1, max) # proportion of count in a single sample, to see if the ASV is concentrated in one sample
plot(dsq.com.8.tt.rs+1, dsq.com.8.tt.rmx/dsq.com.8.tt.rs, log="x")
dsq.com.8.tt.rs.df <- as.data.frame(rowSums(counts(dsq.com.8.tt))) # get row sums of each count (ASV)

###########
# 3 approaches for calc geometric means prior to estimating size factors
###########

## 1) standard estimateSizeFactrs (default) = geometric means calculated from the data
dsq.com.8.tt.def = estimateSizeFactors(dsq.com.8.tt) # no separate geometric mean is calculated first
dsq.com.8.tt.def = DESeq(dsq.com.8.tt.def, fitType="local")
# investigate results
res.dsq.com.8.tt.def <- results(dsq.com.8.tt.def) # results object
res.dsq.com.8.tt.def <- res.dsq.com.8.tt.def[order(res.dsq.com.8.tt.def$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.8.tt.def <- res.dsq.com.8.tt.def[(res.dsq.com.8.tt.def$padj < alpha), ] # only sig results
sigtab.dsq.com.8.tt.def <- cbind(as(sigtab.dsq.com.8.tt.def, "data.frame"), as(tax_table(com_8_unrare)[rownames(sigtab.dsq.com.8.tt.def), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.8.tt.def, "output/sigtab.dsq.com.8.tt.def.csv")

# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.8.tt.def = sigtab.dsq.com.8.tt.def[sigtab.dsq.com.8.tt.def[, "log2FoldChange"] > 0, ]
posigtab.dsq.com.8.tt.def = posigtab.dsq.com.8.tt.def[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.8.tt.def.genus = subset(sigtab.dsq.com.8.tt.def, !is.na(Genus))
# Class order
x.8.tt.def = tapply(sigtab.dsq.com.8.tt.def.genus$log2FoldChange, sigtab.dsq.com.8.tt.def.genus$Class, function(x.8.tt.def) max(x.8.tt.def))
x.8.tt.def=sort(x.8.tt.def, TRUE)
sigtab.dsq.com.8.tt.def.genus$Class = factor(as.character(sigtab.dsq.com.8.tt.def.genus$Class), levels=names(x.8.tt.def))
# Genus order
x.8.tt.def = tapply(sigtab.dsq.com.8.tt.def.genus$log2FoldChange, sigtab.dsq.com.8.tt.def.genus$Genus, function(x.8.tt.def) max(x.8.tt.def))
x.8.tt.def=sort(x.8.tt.def, TRUE)
sigtab.dsq.com.8.tt.def.genus$Genus = factor(as.character(sigtab.dsq.com.8.tt.def.genus$Genus), levels=names(x.8.tt.def))
# plot with x=log2FC, y=genus, color=Class
ggplot(sigtab.dsq.com.8.tt.def.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -80, hjust = 0, vjust = 0.5))

## 2) geoMeans = custom phyloseq approach developed by McMurdie, which deals with lots of zeroes in 16S data
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans.8.tt = apply(counts(dsq.com.8.tt), 1, gm_mean) # custom gm_mean function applied to all rows
dsq.com.8.tt.gm = estimateSizeFactors(dsq.com.8.tt, geoMeans = geoMeans.8.tt) # estimate size factors based on that computed gm_mean
dsq.com.8.tt.gm = DESeq(dsq.com.8.tt.gm, fitType="local")
# investigate results
res.dsq.com.8.tt.gm <- results(dsq.com.8.tt.gm, contrast=c("tt_hilo", "high", "low")) # results object, specify our contrasts
res.dsq.com.8.tt.gm <- res.dsq.com.8.tt.gm[order(res.dsq.com.8.tt.gm$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.8.tt.gm <- res.dsq.com.8.tt.gm[(res.dsq.com.8.tt.gm$padj < alpha), ] # only sig results
sigtab.dsq.com.8.tt.gm <- cbind(as(sigtab.dsq.com.8.tt.gm, "data.frame"), as(tax_table(com_8_unrare)[rownames(sigtab.dsq.com.8.tt.gm), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.8.tt.gm, "output/sigtab.dsq.com.8.tt.gm.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.8.tt.gm = sigtab.dsq.com.8.tt.gm[sigtab.dsq.com.8.tt.gm[, "log2FoldChange"] > 0, ] # only shows ASVs with positive FC
posigtab.dsq.com.8.tt.gm = posigtab.dsq.com.8.tt.gm[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.8.tt.gm.genus = subset(sigtab.dsq.com.8.tt.gm, !is.na(Genus))
# Class order
x.8.tt.gm = tapply(sigtab.dsq.com.8.tt.gm.genus$log2FoldChange, sigtab.dsq.com.8.tt.gm.genus$Class, function(x.8.tt.gm) max(x.8.tt.gm))
x.8.tt.gm=sort(x.8.tt.gm, TRUE)
sigtab.dsq.com.8.tt.gm.genus$Class = factor(as.character(sigtab.dsq.com.8.tt.gm.genus$Class), levels=names(x.8.tt.gm))
# Genus order
x.8.tt.gm = tapply(sigtab.dsq.com.8.tt.gm.genus$log2FoldChange, sigtab.dsq.com.8.tt.gm.genus$Genus, function(x.8.tt.gm) max(x.8.tt.gm))
x.8.tt.gm=sort(x.8.tt.gm, TRUE)
sigtab.dsq.com.8.tt.gm.genus$Genus = factor(as.character(sigtab.dsq.com.8.tt.gm.genus$Genus), levels=names(x.8.tt.gm))
# plot with x=log2FC, y=genus, color=Class
pdf("output/sigtab.dsq.com.8.tt.gm.genus.pdf")
ggplot(sigtab.dsq.com.8.tt.gm.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5)) +
  ggtitle("Big vs. Small Rivers - DESeq2 (sigtab.dsq.com.8.tt.gm.genus.pdf)")
dev.off()

# 2 with strict pre-filtering
dsq.com.8.tt.keep <- rowSums(counts(dsq.com.8.tt) >= 5) >=2  # original = 2837 ASV; kept = 576 ASV
dsq.com.8.tt.kept <- dsq.com.8.tt[dsq.com.8.tt.keep, ]

geoMeans.8.tt.kept = apply(counts(dsq.com.8.tt.kept), 1, gm_mean) # custom gm_mean function applied to all rows
dsq.com.8.tt.kept = estimateSizeFactors(dsq.com.8.tt.kept, geoMeans = geoMeans.8.tt.kept) # estimate size factors based on that computed gm_mean
dsq.com.8.tt.kept = DESeq(dsq.com.8.tt.kept, fitType="local")
# investigate results
res.dsq.com.8.tt.gm.kept <- results(dsq.com.8.tt.kept, contrast=c("tt_hilo", "high", "low")) # low comes last, set as reference
res.dsq.com.8.tt.gm.kept <- res.dsq.com.8.tt.gm.kept[order(res.dsq.com.8.tt.gm.kept$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.8.tt.gm.kept <- res.dsq.com.8.tt.gm.kept[(res.dsq.com.8.tt.gm.kept$padj < alpha), ] # only sig results
sigtab.dsq.com.8.tt.gm.kept <- cbind(as(sigtab.dsq.com.8.tt.gm.kept, "data.frame"), as(tax_table(com_8_unrare)[rownames(sigtab.dsq.com.8.tt.gm.kept), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.8.tt.gm.kept, "output/sigtab.dsq.com.8.tt.gm.kept.csv")
##
res.dsq.com.10.CH4.gm.plot.cons <- results(dsq.com.10.CH4.gm.out, contrast = c("CH4_grp", "high", "low")) # low comes last, set as reference
res.dsq.com.10.CH4.gm.plot.cons <- res.dsq.com.10.CH4.gm.plot.cons[order(res.dsq.com.10.CH4.gm.plot.cons$padj, na.last=NA), ]

sigtab.dsq.com.10.CH4.gm.plot.cons <- res.dsq.com.10.CH4.gm.plot.cons[(res.dsq.com.10.CH4.gm.plot.cons$padj < alpha), ] # only sig; old good code
sigtab.dsq.com.10.CH4.gm.plot.cons <- cbind(as(sigtab.dsq.com.10.CH4.gm.plot.cons, "data.frame"), as(tax_table(com_10_unrare)[rownames(sigtab.dsq.com.10.CH4.gm.plot.cons), ], "matrix")) 
write.csv(sigtab.dsq.com.10.CH4.gm.plot.cons, "output/sigtab.dsq.com.10.CH4.gm.plot.cons.csv")

# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.8.tt.gm = sigtab.dsq.com.8.tt.gm[sigtab.dsq.com.8.tt.gm[, "log2FoldChange"] > 0, ] # only shows ASVs with positive FC
posigtab.dsq.com.8.tt.gm = posigtab.dsq.com.8.tt.gm[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.8.tt.gm.genus = subset(sigtab.dsq.com.8.tt.gm, !is.na(Genus))
# Class order
x.8.tt.gm = tapply(sigtab.dsq.com.8.tt.gm.genus$log2FoldChange, sigtab.dsq.com.8.tt.gm.genus$Class, function(x.8.tt.gm) max(x.8.tt.gm))
x.8.tt.gm=sort(x.8.tt.gm, TRUE)
sigtab.dsq.com.8.tt.gm.genus$Class = factor(as.character(sigtab.dsq.com.8.tt.gm.genus$Class), levels=names(x.8.tt.gm))
# Genus order
x.8.tt.gm = tapply(sigtab.dsq.com.8.tt.gm.genus$log2FoldChange, sigtab.dsq.com.8.tt.gm.genus$Genus, function(x.8.tt.gm) max(x.8.tt.gm))
x.8.tt.gm=sort(x.8.tt.gm, TRUE)
sigtab.dsq.com.8.tt.gm.genus$Genus = factor(as.character(sigtab.dsq.com.8.tt.gm.genus$Genus), levels=names(x.8.tt.gm))
# plot with x=log2FC, y=genus, color=Class
pdf("output/sigtab.dsq.com.8.tt.gm.genus.pdf")
ggplot(sigtab.dsq.com.8.tt.gm.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5)) +
  ggtitle("Big vs. Small Rivers - DESeq2 (sigtab.dsq.com.8.tt.gm.genus.pdf)")
dev.off()


## 3) poscounts = "deals with a gene with some zeros, by calculating a modified geometric mean by taking the n-th root of the product of the non-zero counts. This evolved out of use cases with Paul McMurdie’s phyloseq package for metagenomic (16S, sic) samples"
dsq.com.8.tt.ps = estimateSizeFactors(dsq.com.8.tt, type="poscounts")
dsq.com.8.tt.ps = DESeq(dsq.com.8.tt.ps, fitType="local")
# investigate results
res.dsq.com.8.tt.ps <- results(dsq.com.8.tt.ps) # results object
res.dsq.com.8.tt.ps <- res.dsq.com.8.tt.ps[order(res.dsq.com.8.tt.ps$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.8.tt.ps <- res.dsq.com.8.tt.ps[(res.dsq.com.8.tt.ps$padj < alpha), ] # only sig results
sigtab.dsq.com.8.tt.ps <- cbind(as(sigtab.dsq.com.8.tt.ps, "data.frame"), as(tax_table(com_8_unrare)[rownames(sigtab.dsq.com.8.tt.ps), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.8.tt.ps, "output/sigtab.dsq.com.8.tt.ps.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.8.tt.ps = sigtab.dsq.com.8.tt.ps[sigtab.dsq.com.8.tt.ps[, "log2FoldChange"] > 0, ] # only includes ASVs with positive FC
posigtab.dsq.com.8.tt.ps = posigtab.dsq.com.8.tt.ps[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.8.tt.ps.genus = subset(sigtab.dsq.com.8.tt.ps, !is.na(Genus))
# Class order
x.8.tt.ps = tapply(sigtab.dsq.com.8.tt.ps.genus$log2FoldChange, sigtab.dsq.com.8.tt.ps.genus$Class, function(x.8.tt.ps) max(x.8.tt.ps))
x.8.tt.ps=sort(x.8.tt.ps, TRUE)
sigtab.dsq.com.8.tt.ps.genus$Class = factor(as.character(sigtab.dsq.com.8.tt.ps.genus$Class), levels=names(x.8.tt.ps))
# Genus order
x.8.tt.ps = tapply(sigtab.dsq.com.8.tt.ps.genus$log2FoldChange, sigtab.dsq.com.8.tt.ps.genus$Genus, function(x.8.tt.ps) max(x.8.tt.ps))
x.8.tt.ps=sort(x.8.tt.ps, TRUE)
sigtab.dsq.com.8.tt.ps.genus$Genus = factor(as.character(sigtab.dsq.com.8.tt.ps.genus$Genus), levels=names(x.8.tt.ps))
# plot with x=log2FC, y=genus, color=Class
ggplot(sigtab.dsq.com.8.tt.ps.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5))
```

# Set 6 - congo synoptic.2 - smaller rivers only but with Mopanzi - COA and co-inertia
```{r}
meta_otu_congo_syn.ci<- subset(meta_otu_congo_syn, !is.na(ORD_STRA) & !is.na(log10_tt) & !is.na(log10_q_daily_mean) & !is.na(log10_DIST_UP_KM) & !is.na(log10_UPLAND_SKM) & !is.na(temp)
 & !is.na(pH) & !is.na(log10_SPC) & !is.na(log10_suva_254) & !is.na(log10_DOC) & !is.na(Sr) & !is.na(FI) & !is.na(log10_do_mg_l) & !is.na(CO2_pct_sat) & !is.na(CH4_pct_sat) & !is.na(log10_NOx) & !is.na(log10_PO4))

# remove Congo, Oubangui, and Sangha - 3 largest sites (based on TT)
meta_otu_congo_syn.ci.2 <- subset(meta_otu_congo_syn.ci, Row.names!="CONGO_017" & Row.names!="CONGO_031" & Row.names!="CONGO_044")

congo_syn_meta.ci.2 <- meta_otu_congo_syn.ci.2[,c(1:congo_syn_meta_col)]
otu_con_syn.ci.2 <- meta_otu_congo_syn.ci.2[,c(otu_congo_syn_start:otu_congo_syn_end)]

# COA
otu_coa_congo_syn.2 <- dudi.coa(otu_con_syn.ci.2, scannf=FALSE, nf=8)


# tt.6 - small synoptic rivers (but still contains Mopanzi!)
congo_syn_meta.ci.pca.6 <- congo_syn_meta.ci.2[,c("log10_tt",  "temp" , "pH", "log10_SPC", "log10_suva_254", "log10_DOC", "Sr", "FI", "do_pct", "CO2_pct_sat", "CH4_pct_sat", "log10_NOx", "log10_PO4")]
# PCA
env_pca_congo_syn.6 <- dudi.pca(d=congo_syn_meta.ci.pca.6, row.w=otu_coa_congo_syn.2$lw, scale = TRUE, scannf = FALSE, nf = 10)
# co-inertia
coin_con_syn.6 <- coinertia(env_pca_congo_syn.6, otu_coa_congo_syn.2, scan = FALSE, nf = 5)

enviro_loadings_coin.6 <- as.data.frame(coin_con_syn.6$co)
enviro_loadings_coin.6$variable <- row.names(enviro_loadings_coin.6)

site_scores_coin_con_syn.6_0 <- as.data.frame(coin_con_syn.6$mX)
site_scores_coin_con_syn.6 <- bind_cols(site_scores_coin_con_syn.6_0, congo_syn_meta.ci.2)

# color by tt
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.6.tt.small.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.6,aes(NorS1,NorS2,label=river, color=log10_tt), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.6, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.6, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("19.7% Projected Inertia")+ # 30.859 x 0.639251
  ylab("14.7% Projected Inertia")+ # 22.992 x 0.639251
  ggtitle("Congo Synoptic co-inertia.6, small synoptic rivers, RV=0.639251")
dev.off()

# color by DOC
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.6.DOC.small.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.6,aes(NorS1,NorS2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="yellow")+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.6, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.6, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("19.7% Projected Inertia")+ # 30.859 x 0.639251
  ylab("14.7% Projected Inertia")+ # 22.992 x 0.639251
  ggtitle("Congo Synoptic co-inertia.6, small synoptic rivers, RV=0.639251")
dev.off()

# color by pH
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.6.pH.small.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.6,aes(NorS1,NorS2,label=river, color=pH), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="yellow")+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.6, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.6, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("19.7% Projected Inertia")+ # 30.859 x 0.639251
  ylab("14.7% Projected Inertia")+ # 22.992 x 0.639251
  ggtitle("Congo Synoptic co-inertia.6, small synoptic rivers, RV=0.639251")
dev.off()


# all hydro.7 - small synoptic rivers
congo_syn_meta.ci.pca.7 <- congo_syn_meta.ci.2[,c("log10_tt", "log10_q_daily_mean", "log10_DIST_UP_KM", "log10_UPLAND_SKM", "temp", "pH", "log10_SPC", "log10_suva_254", "log10_DOC", "Sr", "FI", "do_pct", "CO2_pct_sat", "CH4_pct_sat", "log10_NOx", "log10_PO4")]
# PCA
env_pca_congo_syn.7 <- dudi.pca(d=congo_syn_meta.ci.pca.7, row.w=otu_coa_congo_syn.2$lw, scale = TRUE, scannf = FALSE, nf = 10)
# co-inertia
coin_con_syn.7 <- coinertia(env_pca_congo_syn.7, otu_coa_congo_syn.2, scan = FALSE, nf = 5)

enviro_loadings_coin.7 <- as.data.frame(coin_con_syn.7$co)
enviro_loadings_coin.7$variable <- row.names(enviro_loadings_coin.7)

site_scores_coin_con_syn.7_0 <- as.data.frame(coin_con_syn.7$mX)
site_scores_coin_con_syn.7 <- bind_cols(site_scores_coin_con_syn.7_0, congo_syn_meta.ci.2)

# color by TT
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.7.allhydro.small.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.7,aes(NorS1,NorS2,label=river, color=log10_tt), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.7, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.7, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("24.4% Projected Inertia")+ # 37.780 * 0.6463951
  ylab("15.1% Projected Inertia")+ # 23.348 * 0.6463951
  ggtitle("Congo Synoptic co-inertia.7, all hydrosmall synoptic rivers, RV=0.646")
dev.off()

# color by [DOC]
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.7.allhydro.small.DOC.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.7,aes(NorS1,NorS2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="yellow")+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.7, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.7, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("24.4% Projected Inertia")+ # 37.780 * 0.6463951
  ylab("15.1% Projected Inertia")+ # 23.348 * 0.6463951
  ggtitle("Congo Synoptic co-inertia.7, all hydrosmall synoptic rivers, RV=0.646")
dev.off()

# color by pH
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.7.allhydro.small.pH.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.7,aes(NorS1,NorS2,label=river, color=pH), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="yellow")+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.7, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.7, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("24.4% Projected Inertia")+ # 37.780 * 0.6463951
  ylab("15.1% Projected Inertia")+ # 23.348 * 0.6463951
  ggtitle("Congo Synoptic co-inertia.7, all hydrosmall synoptic rivers, RV=0.646")
dev.off()

```

# Set 9: congo synoptic - smaller rivers only - exploratory COA and CI; also Mopanzi and outliers removed
```{r}
# remove 3 largest rivers Congo, Sangha, Obangui; Mopanzi and Djiri already removed above in *out.1 object
meta_otu_congo_syn.out1.9 <- subset(meta_otu_congo_syn.out1, Row.names!="CONGO_017" & Row.names!="CONGO_031" & Row.names!="CONGO_044")

congo_syn_meta.out1.9 <- meta_otu_congo_syn.out1.9[,c(1:congo_syn_meta_col)]
otu_con_syn.out1.9 <- meta_otu_congo_syn.out1.9[,c(otu_congo_syn_start:otu_congo_syn_end)]

# check zero occurrences
zero.otu_con_syn.out1.9 <- colSums(otu_con_syn.out1.9) == 0
otu_con_syn.out1.9.nozero <- otu_con_syn.out1.9[, !zero.otu_con_syn.out1.9] # 3115 to 2203 columns

# COA
otu_coa_congo_syn.out1.9 <- dudi.coa(otu_con_syn.out1.9.nozero, scannf=FALSE, nf=8)
# plot COA
# export COA coordinates
otu_coa_congo_syn.out1.9_li <- as.data.frame(otu_coa_congo_syn.out1.9$li)
coa_meta_congo_syn.out1.9 <- merge(otu_coa_congo_syn.out1.9_li, congo_syn_meta.out1.9, by=0)

  # land tec_cl_cmj = ecoregion; 63=western congolian forest-savanna; 26 = northwest congolian lowland forests; 29 = western congolian swamp forests
ecoregion_col <- c("63"="darkkhaki", "26" = "green", "29" = "blue")
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.9.land.tec.cl.cmj.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1.9,aes(Axis1,Axis2,label=river, color=tec_cl_cmj), size=4, fontface="bold")+
  scale_color_manual(values=ecoregion_col)+
  theme_classic()+
  ggtitle("CA synoptic by land-tec_cl_cmj small rivers n=18")+
  xlab("14.2% projected inertia") +
  ylab("12.0% projected inertia")
dev.off()

# wet_pc_cg2 = catchment, percent of wetland class g2 (excludes, rivers, lakes reservoirs)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.9.wet.pc.cg2.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1.9,aes(Axis1,Axis2,label=river, color=wet_pc_cg2), size=4, fontface="bold")+
  scale_color_gradient(low="green", high="blue")+
  theme_classic()+
  ggtitle("CA synoptic by wet.pc.cg2 small rivers n=18")+
  xlab("14.2% projected inertia") +
  ylab("12.0% projected inertia")
dev.off()

# wet_pc_ug2 = Watershed, percent of wetland class g2 (excludes, rivers, lakes reservoirs)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.9.wet.pc.ug2.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1.9,aes(Axis1,Axis2,label=river, color=wet_pc_ug2), size=4, fontface="bold")+
  scale_color_gradient(low="green", high="blue")+
  theme_classic()+
  ggtitle("CA synoptic by wet.pc.ug2 small rivers n=18")+
  xlab("14.2% projected inertia") +
  ylab("12.0% projected inertia")
dev.off()

# wet_pc_c04 = catchment, percent "freshwater marsh, floodplain"
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.9.wet.pc.c04.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1.9,aes(Axis1,Axis2,label=river, color=wet_pc_c04), size=4, fontface="bold")+
  scale_color_gradient(low="green", high="blue")+
  theme_classic()+
  ggtitle("CA synoptic by wet.pc.c04 small rivers n=18")+
  xlab("14.2 projected inertia") +
  ylab("12.0% projected inertia")
dev.off()

# wet_pc_u04 = watershed, percent "freshwater marsh, floodplain"
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.9.wet.pc.u04.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1.9,aes(Axis1,Axis2,label=river, color=wet_pc_u04), size=4, fontface="bold")+
  scale_color_gradient(low="green", high="blue")+
  theme_classic()+
  ggtitle("CA synoptic by wet.pc.u04 small rivers n=18")+
  xlab("14.2% projected inertia") +
  ylab("12.0% projected inertia")
dev.off()

# wet_pc_c05 = catchment, percent "swamp forest, flooded forest"
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.9.wet.pc.c05.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1.9,aes(Axis1,Axis2,label=river, color=wet_pc_c05), size=4, fontface="bold")+
  scale_color_gradient(low="green", high="blue")+
  theme_classic()+
  ggtitle("CA synoptic by wet.pc.c05 small rivers n=18")+
  xlab("14.2% projected inertia") +
  ylab("12.0% projected inertia")
dev.off()

# wet_pc_u05 = watershed, percent "swamp forest, flooded forest"
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.9.wet.pc.u05.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1.9,aes(Axis1,Axis2,label=river, color=wet_pc_u05), size=4, fontface="bold")+
  scale_color_gradient(low="green", high="blue")+
  theme_classic()+
  ggtitle("CA synoptic by wet.pc.u05 small rivers n=18")+
  xlab("14.2% projected inertia") +
  ylab("12.0% projected inertia")
dev.off()

# for_pc_cse = Forest cover, percent, catchment
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.9.for.pc.cse.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1.9,aes(Axis1,Axis2,label=river, color=for_pc_cse), size=4, fontface="bold")+
  scale_color_gradient(low="blue", high="yellow")+
  theme_classic()+
  ggtitle("CA synoptic by for.pc.cse small rivers n=18")+
  xlab("14.2% projected inertia") +
  ylab("12.0% projected inertia")
dev.off()

# for_pc_use = Forest cover, percent, watershed
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coa.con.syn.out1.9.for.pc.use.pdf", width=9, height=8)
ggplot()+
  geom_text(data=coa_meta_congo_syn.out1.9,aes(Axis1,Axis2,label=river, color=for_pc_use), size=4, fontface="bold")+
  scale_color_gradient(low="blue", high="yellow")+
  theme_classic()+
  ggtitle("CA synoptic by for.pc.use small rivers n=18")+
  xlab("14.2% projected inertia") +
  ylab("12.0% projected inertia")
dev.off()

# all hydro:
congo_syn_meta.out1.9.hydro <- congo_syn_meta.out1.9[,c("log10_tt", "log10_q_daily_mean", "log10_DIST_UP_KM", "log10_UPLAND_SKM", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4")]
# PCA
env_pca_congo_syn.out1.9.hydro <- dudi.pca(d=congo_syn_meta.out1.9.hydro, row.w=otu_coa_congo_syn.out1.9$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.9.hydro <- coinertia(env_pca_congo_syn.out1.9.hydro, otu_coa_congo_syn.out1.9, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.hydro <- as.data.frame(coin_con_syn.out1.9.hydro$co)
enviro_loadings_coin.out1.9.hydro$variable <- row.names(enviro_loadings_coin.out1.9.hydro)

site_scores_coin_con_syn.out1.9.hydro_0 <- as.data.frame(coin_con_syn.out1.9.hydro$mX)
site_scores_coin_con_syn.out1.9.hydro <- bind_cols(site_scores_coin_con_syn.out1.9.hydro_0, congo_syn_meta.out1.9)

# tt color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.hydro.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.hydro,aes(NorS1,NorS2,label=river, color=log10_tt), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.hydro, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.hydro, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.2 Projected Inertia")+ # 0.6442599*34.503 = 22.2289
  ylab("16.5% Projected Inertia")+ # 0.6442599*25.578 = 16.47888
  ggtitle("Congo Synoptic co-inertia.outliers1.9.all.hydro, RV=0.6443")
dev.off()


# tt only
congo_syn_meta.out1.9.tt <- congo_syn_meta.out1.9[,c("log10_tt", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4")]
# PCA
env_pca_congo_syn.out1.9.tt <- dudi.pca(d=congo_syn_meta.out1.9.tt, row.w=otu_coa_congo_syn.out1.9$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.9.tt <- coinertia(env_pca_congo_syn.out1.9.tt, otu_coa_congo_syn.out1.9, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.tt <- as.data.frame(coin_con_syn.out1.9.tt$co)
enviro_loadings_coin.out1.9.tt$variable <- row.names(enviro_loadings_coin.out1.9.tt)

site_scores_coin_con_syn.out1.9.tt_0 <- as.data.frame(coin_con_syn.out1.9.tt$mX)
site_scores_coin_con_syn.out1.9.tt <- bind_cols(site_scores_coin_con_syn.out1.9.tt_0, congo_syn_meta.out1.9)

# tt color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.tt.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.tt,aes(NorS1,NorS2,label=river, color=log10_tt), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.5 Projected Inertia")+ # 0.6704577*34.100 = 22.48406
  ylab("15.8% Projected Inertia")+ # 0.6704577*23.661= 15.77896
  ggtitle("Congo Synoptic co-inertia.outliers1.9.tt, RV=0.6704")
dev.off()

    # pH color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.tt.pH.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.tt,aes(NorS1,NorS2,label=river, color=pH), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="magenta")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.5 Projected Inertia")+ # 0.6704577*34.100 = 22.48406
  ylab("15.8% Projected Inertia")+ # 0.6704577*23.661= 15.77896
  ggtitle("Congo Synoptic co-inertia.outliers1.9.tt, RV=0.6704")
dev.off()
    # DOC color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.tt.DOC.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.tt,aes(NorS1,NorS2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="magenta")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.5 Projected Inertia")+ # 0.6704577*34.100 = 22.48406
  ylab("15.8% Projected Inertia")+ # 0.6704577*23.661= 15.77896
  ggtitle("Congo Synoptic co-inertia.outliers1.9.tt, RV=0.6704")
dev.off()
    # CH4 color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.tt.CH4.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.tt,aes(NorS1,NorS2,label=river, color=log10_CH4_pct_sat), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="yellow")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.5 Projected Inertia")+ # 0.6704577*34.100 = 22.48406
  ylab("15.8% Projected Inertia")+ # 0.6704577*23.661= 15.77896
  ggtitle("Congo Synoptic co-inertia.outliers1.9.tt, RV=0.6704")
dev.off()

# q
congo_syn_meta.out1.9.q <- congo_syn_meta.out1.9[,c("log10_q_daily_mean", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4")]
# PCA
env_pca_congo_syn.out1.9.q <- dudi.pca(d=congo_syn_meta.out1.9.q, row.w=otu_coa_congo_syn.out1.9$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.9.q <- coinertia(env_pca_congo_syn.out1.9.q, otu_coa_congo_syn.out1.9, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.q <- as.data.frame(coin_con_syn.out1.9.q$co)
enviro_loadings_coin.out1.9.q$variable <- row.names(enviro_loadings_coin.out1.9.q)

site_scores_coin_con_syn.out1.9.q_0 <- as.data.frame(coin_con_syn.out1.9.q$mX)
site_scores_coin_con_syn.out1.9.q <- bind_cols(site_scores_coin_con_syn.out1.9.q_0, congo_syn_meta.out1.9)

# q color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.q.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.q,aes(NorS1,NorS2,label=river, color=log10_q_daily_mean), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.q, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.q, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.4 Projected Inertia")+ # 0.6614477*33.812 = 22.36487
  ylab("17.5% Projected Inertia")+ # 0.6614477*26.495 = 17.52506
  ggtitle("Congo Synoptic co-inertia.outliers1.9.q, RV=0.6614")
dev.off()

# dendritic distance
congo_syn_meta.out1.9.dd <- congo_syn_meta.out1.9[,c("log10_DIST_UP_KM", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4")]
# PCA
env_pca_congo_syn.out1.9.dd <- dudi.pca(d=congo_syn_meta.out1.9.dd, row.w=otu_coa_congo_syn.out1.9$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.9.dd <- coinertia(env_pca_congo_syn.out1.9.dd, otu_coa_congo_syn.out1.9, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.dd <- as.data.frame(coin_con_syn.out1.9.dd$co)
enviro_loadings_coin.out1.9.dd$variable <- row.names(enviro_loadings_coin.out1.9.dd)

site_scores_coin_con_syn.out1.9.dd_0 <- as.data.frame(coin_con_syn.out1.9.dd$mX)
site_scores_coin_con_syn.out1.9.dd <- bind_cols(site_scores_coin_con_syn.out1.9.dd_0, congo_syn_meta.out1.9)

# dd color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.dd.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.dd,aes(NorS1,NorS2,label=river, color=log10_DIST_UP_KM), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.dd, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.dd, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.9 Projected Inertia")+ # 0.6805939*33.614 = 22.87748
  ylab("15.9% Projected Inertia")+ # 0.6805939*23.370 = 15.90548
  ggtitle("Congo Synoptic co-inertia.outliers1.9.dd, RV=0.6810")
dev.off()
# pH color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.dd.pH.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.dd,aes(NorS1,NorS2,label=river, color=pH), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="magenta")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.dd, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.dd, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.5 Projected Inertia")+ # 0.691302*32.471 = 22.44727
  ylab("15.7% Projected Inertia")+ # 0.691302*22.684 = 15.68149
  ggtitle("Congo Synoptic co-inertia.outliers1.9.dd, RV=0.6913")
dev.off()
# CH4 color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.dd.CH4.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.dd,aes(NorS1,NorS2,label=river, color=CH4_pct_sat), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="yellow")+
  theme_classic()+
  scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.dd, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.dd, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22.5 Projected Inertia")+ # 0.691302*32.471 = 22.44727
  ylab("15.7% Projected Inertia")+ # 0.691302*22.684 = 15.68149
  ggtitle("Congo Synoptic co-inertia.outliers1.9.dd, RV=0.6913")
dev.off()


# area km2
congo_syn_meta.out1.9.km2 <- congo_syn_meta.out1.9[,c("log10_UPLAND_SKM", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4")]
# PCA
env_pca_congo_syn.out1.9.km2 <- dudi.pca(d=congo_syn_meta.out1.9.km2, row.w=otu_coa_congo_syn.out1.9$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.9.km2 <- coinertia(env_pca_congo_syn.out1.9.km2, otu_coa_congo_syn.out1.9, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.km2 <- as.data.frame(coin_con_syn.out1.9.km2$co)
enviro_loadings_coin.out1.9.km2$variable <- row.names(enviro_loadings_coin.out1.9.km2)

site_scores_coin_con_syn.out1.9.km2_0 <- as.data.frame(coin_con_syn.out1.9.km2$mX)
site_scores_coin_con_syn.out1.9.km2 <- bind_cols(site_scores_coin_con_syn.out1.9.km2_0, congo_syn_meta.out1.9)

# km2 color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.km2.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.km2,aes(NorS1,NorS2,label=river, color=log10_UPLAND_SKM), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.km2, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.km2, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("22/8 Projected Inertia")+ # 0.6802551*33.496 = 22.78582
  ylab("15.8% Projected Inertia")+ # 0.6802551*23.267 = 15.8275
  ggtitle("Congo Synoptic co-inertia.outliers1.9.area.km2, RV=0.6803")
dev.off()

# co-inertia using different landcover metrics
# all continuous landcover metrics
congo_syn_meta.out1.9.allwet <- congo_syn_meta.out1.9[,c("log10_tt", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_cg2", "wet_pc_ug2", "wet_pc_c04", "wet_pc_u04", "wet_pc_c05", "wet_pc_u05")]
# PCA
env_pca_congo_syn.out1.9.allwet <- dudi.pca(d=congo_syn_meta.out1.9.allwet, row.w=otu_coa_congo_syn.out1.9$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.9.allwet <- coinertia(env_pca_congo_syn.out1.9.allwet, otu_coa_congo_syn.out1.9, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.allwet <- as.data.frame(coin_con_syn.out1.9.allwet$co)
enviro_loadings_coin.out1.9.allwet$variable <- row.names(enviro_loadings_coin.out1.9.allwet)

site_scores_coin_con_syn.out1.9.allwet_0 <- as.data.frame(coin_con_syn.out1.9.allwet$mX)
site_scores_coin_con_syn.out1.9.allwet <- bind_cols(site_scores_coin_con_syn.out1.9.allwet_0, congo_syn_meta.out1.9)

# wet.pc.cg2 color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.allwet.wet.pc.c04.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.allwet,aes(NorS1,NorS2,label=river, color=wet_pc_c04), size=4, fontface="bold")+
  scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.allwet, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.allwet, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("26.0Projected Inertia")+ # 0.6588291*39.390 = 25.95128
  ylab("13.0% Projected Inertia")+ # 0.6588291*19.790= 13.03823
  ggtitle("Congo Synoptic co-inertia.outliers1.9.allwet, RV=0.659")
dev.off()

# wet_pc_c05 (one of top vectors from allwet)
congo_syn_meta.out1.9.wet.pc.c05 <- congo_syn_meta.out1.9[,c("log10_tt", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_c05")]
# PCA
env_pca_congo_syn.out1.9.wet.pc.c05 <- dudi.pca(d=congo_syn_meta.out1.9.wet.pc.c05, row.w=otu_coa_congo_syn.out1.9$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.9.wet.pc.c05 <- coinertia(env_pca_congo_syn.out1.9.wet.pc.c05, otu_coa_congo_syn.out1.9, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.wet.pc.c05 <- as.data.frame(coin_con_syn.out1.9.wet.pc.c05$co)
enviro_loadings_coin.out1.9.wet.pc.c05$variable <- row.names(enviro_loadings_coin.out1.9.wet.pc.c05)

site_scores_coin_con_syn.out1.9.wet.pc.c05_0 <- as.data.frame(coin_con_syn.out1.9.wet.pc.c05$mX)
site_scores_coin_con_syn.out1.9.wet.pc.c05 <- bind_cols(site_scores_coin_con_syn.out1.9.wet.pc.c05_0, congo_syn_meta.out1.9)

# wet.pc.c05 color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.wet.pc.c05.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.c05,aes(NorS1,NorS2,label=river, color=wet_pc_c05), size=4, fontface="bold")+
  scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.0% Projected Inertia")+ # 0.6726789*37.159 = 24.99608
  ylab("14.6% Projected Inertia")+ # 0.6726789*21.694= 14.5931
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.c05, RV=0.673")
dev.off()

# pH color in wet.pc.c05 CI
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.wet.pc.c05.pH.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.c05,aes(NorS1,NorS2,label=river, color=pH), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="magenta")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.0% Projected Inertia")+ # 0.6726789*37.159 = 24.99608
  ylab("14.6% Projected Inertia")+ # 0.6726789*21.694= 14.5931
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.c05, RV=0.673")
dev.off()

# DOC color in wet.pc.c05 CI
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.wet.pc.c05.doc.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.c05,aes(NorS1,NorS2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="magenta")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.0% Projected Inertia")+ # 0.6726789*37.159 = 24.99608
  ylab("14.6% Projected Inertia")+ # 0.6726789*21.694= 14.5931
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.c05, RV=0.673")
dev.off()

# CH4 color in wet.pc.c05 CI
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.wet.pc.c05.ch4.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.c05,aes(NorS1,NorS2,label=river, color=log10_CH4_pct_sat), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="yellow")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.0% Projected Inertia")+ # 0.6726789*37.159 = 24.99608
  ylab("14.6% Projected Inertia")+ # 0.6726789*21.694= 14.5931
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.c05, RV=0.673")
dev.off()
```

# set 9 - RA otu conversion
```{r}
#________wet.pc.c05 - indicator species
# use original phyloseq object to eliminate samples from n=23 to n=18
congo_syn.9 <- ps_filter(congo_syn, sample_id!="CONGO_003", sample_id!="CONGO_019", sample_id!="CONGO_017", sample_id!="CONGO_031", sample_id!="CONGO_044") # n=18, matches set 9 here
# convert OTU data to relative abundance
library(phylosmith)
congo_syn.9.ra<-relative_abundance(congo_syn.9, sig_fig=6)
otu_congo_syn.9.ra <- otu_table(congo_syn.9.ra)
write.csv(otu_congo_syn.9.ra, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/otu_congo_syn.9.ra.csv")
otu_congo_syn.9.ra.t <- as.data.frame(t(otu_congo_syn.9.ra)) # transpose OTU table, so rows=samples, columns=OTUs. This is input format for indicator species
write.csv(otu_congo_syn.9.ra.t, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/otu_congo_syn.9.ra.t.csv")
```

# set 9 - wet_pc_c05 indicator
```{r}
# use factoextra to create elbow plot to find optimal number of kmeans clusters; use on CI site scores
library(factoextra)
set.seed(123)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/k.ebbow.coin_con_syn.out1.9.wet.pc.c05.pdf", width=11, height=8)
coords.coin_con_syn.out1.9.wet.pc.c05 <- site_scores_coin_con_syn.out1.9.wet.pc.c05[,c("NorS1", "NorS2")]
fviz_nbclust(coords.coin_con_syn.out1.9.wet.pc.c05, kmeans, method="wss")
dev.off() # on inspection, n=3 looks ideal

# run kmeans = 3 on metadata to prepare for indicator species
kmeans_sites.9.wet.pc.c05 <- kmeans(site_scores_coin_con_syn.out1.9.wet.pc.c05[,c("NorS1", "NorS2")], centers=3, iter.max=10, nstart=25)
site_scores_coin_con_syn.out1.9.wet.pc.c05$wet.pc.c05clus<-as.factor(kmeans_sites.9.wet.pc.c05$cluster)

library(indicspecies)
indic.9.wet.pc.c05 <- multipatt(otu_congo_syn.9.ra.t, site_scores_coin_con_syn.out1.9.wet.pc.c05$wet.pc.u05clus, control=how(nperm=999))
indic.9.wet.pc.c05.results <- as.data.frame(indic.9.wet.pc.c05$sign)
indic.9.wet.pc.c05.results.2 <- subset(indic.9.wet.pc.c05.results, p.value<0.05&!is.na(p.value))
indic.9.wet.pc.c05.results.2$OTU_ID2 <- row.names(indic.9.wet.pc.c05.results.2)

tax <- as.data.frame(tax_mat)
tax$OTU_ID2 <- row.names(tax)
write.csv(tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/tax.csv")

indic.9.wet.pc.c05.results.2.tax <- merge(indic.9.wet.pc.c05.results.2,tax,by="OTU_ID2")

write.csv(indic.9.wet.pc.c05.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.wet.pc.c05.results.3clust.tax.csv")

summary(indic.9.wet.pc.c05, indvalcomp=TRUE)

# cluster color (n=3) for wet.pc.c05
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.wet.pc.c05.3clust.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.c05,aes(NorS1,NorS2,label=river, color=wet.pc.u05clus), size=4, fontface="bold")+
  #scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.0% Projected Inertia")+ # 0.6726789*37.159 = 24.99608
  ylab("14.6% Projected Inertia")+ # 0.6726789*21.694= 14.5931
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.c05, RV=0.673 kmeans=3")
dev.off()

  # pH indicator species on the wet.pc.c05 co-inertia
# standardize pH data (z-score)
pH_all <- as.data.frame(congo_syn_meta.out1.9$pH)
k.pH.scaled <- scale(pH_all)
# use factoextra to create elbow plot to find optimal number of kmeans clusters; use on CI site scores
library(factoextra)
set.seed(132)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/k.ebbow.coin_con_syn.out1.9.pH.pdf", width=11, height=8)
fviz_nbclust(k.pH.scaled, kmeans, method="wss")
dev.off() # on inspection, n=3 or 5 look good

# run kmeans = 3 on metadata to prepare for indicator species
kmeans.pH.wet.pc.c05.3 <- kmeans(k.pH.scaled, centers=3, iter.max=10, nstart=25)
site_scores_coin_con_syn.out1.9.wet.pc.c05$clus.pH.3<-as.factor(kmeans.pH.wet.pc.c05.3$cluster)

library(indicspecies)
indic.9.pH.3.wet.pc.c05 <- multipatt(otu_congo_syn.9.ra.t, site_scores_coin_con_syn.out1.9.wet.pc.c05$clus.pH.3, control=how(nperm=999))
indic.9.pH.3.wet.pc.c05.results <- as.data.frame(indic.9.pH.3.wet.pc.c05$sign)
indic.9.pH.3.wet.pc.c05.results.2 <- subset(indic.9.pH.3.wet.pc.c05.results, p.value<0.05&!is.na(p.value))
indic.9.pH.3.wet.pc.c05.results.2$OTU_ID2 <- row.names(indic.9.pH.3.wet.pc.c05.results.2)

indic.9.pH.3.wet.pc.c05.results.2.tax <- merge(indic.9.pH.3.wet.pc.c05.results.2,tax,by="OTU_ID2")

write.csv(indic.9.pH.3.wet.pc.c05.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.pH.3,wet.pc.c05.results.3clust.tax.csv")

summary(indic.9.pH.3.wet.pc.c05, indvalcomp=TRUE)

# cluster color (n=3) for pH in wet.pc.c05 co-inertia (coin_con_syn.out1.9.wet.pc.c05)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.pH.3.wet.pc.c05.3clust.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.c05,aes(NorS1,NorS2,label=river, color=clus.pH.3), size=4, fontface="bold")+
  #scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.0% Projected Inertia")+ # 0.6726789*37.159 = 24.99608
  ylab("14.6% Projected Inertia")+ # 0.6726789*21.694= 14.5931
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.c05, RV=0.673 kmeans=3 by pH")
dev.off()

# pH kmeans = 5 on metadata to prepare for indicator species
kmeans.pH.wet.pc.c05.5 <- kmeans(k.pH.scaled, centers=5, iter.max=10, nstart=25)
site_scores_coin_con_syn.out1.9.wet.pc.c05$clus.pH.5<-as.factor(kmeans.pH.wet.pc.c05.5$cluster)

library(indicspecies)
indic.9.pH.5.wet.pc.c05 <- multipatt(otu_congo_syn.9.ra.t, site_scores_coin_con_syn.out1.9.wet.pc.c05$clus.pH.5, control=how(nperm=999))
indic.9.pH.5.wet.pc.c05.results <- as.data.frame(indic.9.pH.5.wet.pc.c05$sign)
indic.9.pH.5.wet.pc.c05.results.2 <- subset(indic.9.pH.5.wet.pc.c05.results, p.value<0.05&!is.na(p.value))
indic.9.pH.5.wet.pc.c05.results.2$OTU_ID2 <- row.names(indic.9.pH.5.wet.pc.c05.results.2)

indic.9.pH.5.wet.pc.c05.results.2.tax <- merge(indic.9.pH.5.wet.pc.c05.results.2,tax,by="OTU_ID2")
write.csv(indic.9.pH.5.wet.pc.c05.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.pH.5.wet.pc.c05.results.3clust.tax.csv")

# cluster color (n=5) for pH in wet.pc.c05 co-inertia (coin_con_syn.out1.9.wet.pc.c05)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.pH.5.wet.pc.c05.5clust.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.c05,aes(NorS1,NorS2,label=river, color=clus.pH.5), size=4, fontface="bold")+
  #scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  #scale_y_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.0% Projected Inertia")+ # 0.6726789*37.159 = 24.99608
  ylab("14.6% Projected Inertia")+ # 0.6726789*21.694= 14.5931
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.c05, RV=0.673 kmeans=5 by pH")
dev.off()

# CH4 kmeans = 3 on metadata to prepare for indicator species
CH4_all_congo_syn_meta.out1.9 <- as.data.frame(congo_syn_meta.out1.9$CH4_pct_sat)
k.CH4.scaled.1.9 <- scale(CH4_all_congo_syn_meta.out1.9)

kmeans_sites.9.ch4.3.wet.pc.c05 <- kmeans(k.CH4.scaled.1.9, centers=3, iter.max=10, nstart=25)
site_scores_coin_con_syn.out1.9.wet.pc.c05$clus.ch4.3<-as.factor(kmeans_sites.9.ch4.3.wet.pc.c05$cluster)

library(indicspecies)
indic.9.ch4.3.wet.pc.c05 <- multipatt(otu_congo_syn.9.ra.t, site_scores_coin_con_syn.out1.9.wet.pc.c05$clus.ch4.3, control=how(nperm=999))
indic.9.ch4.3.wet.pc.c05.results <- as.data.frame(indic.9.ch4.3.wet.pc.c05$sign)
indic.9.ch4.3.wet.pc.c05.results.2 <- subset(indic.9.ch4.3.wet.pc.c05.results, p.value<0.05&!is.na(p.value))
indic.9.ch4.3.wet.pc.c05.results.2$OTU_ID2 <- row.names(indic.9.ch4.3.wet.pc.c05.results.2)

indic.9.ch4.3.wet.pc.c05.results.2.tax <- merge(indic.9.ch4.3.wet.pc.c05.results.2,tax,by="OTU_ID2")
write.csv(indic.9.ch4.3.wet.pc.c05.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.ch4.3.wet.pc.c05.results.3clust.tax.csv")

summary(indic.9.ch4.3.wet.pc.c05, indvalcomp=TRUE)

# cluster color (n=3) for ch4 in wet.pc.c05
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.ch4.3.wet.pc.c05.3clust.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.c05,aes(NorS1,NorS2,label=river, color=clus.ch4.3), size=4, fontface="bold")+
  #scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.0% Projected Inertia")+ # 0.6726789*37.159 = 24.99608
  ylab("14.6% Projected Inertia")+ # 0.6726789*21.694= 14.5931
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.c05, RV=0.673 kmeans=3 for CH4")
dev.off()

  # DOC indicator species on the wet.pc.c05 co-inertia
# standardize DOC data (z-score)
doc_all_congo_syn_meta.out1.9 <- as.data.frame(congo_syn_meta.out1.9$doc_mcc_l)
k.doc.scaled.1.9 <- scale(doc_all_congo_syn_meta.out1.9)
# use factoextra to create elbow plot to find optimal number of kmeans clusters; use on CI site scores
library(factoextra)
set.seed(133)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/k.ebbow.coin_con_syn.out1.9.DOC.pdf", width=11, height=8)
fviz_nbclust(k.doc.scaled.1.9, kmeans, method="wss")
dev.off() # on inspection, n=2 or 3 look good

# run kmeans = 3 on metadata to prepare for indicator species
kmeans_sites.9.doc.3.wet.pc.c05 <- kmeans(k.doc.scaled.1.9, centers=3, iter.max=10, nstart=25)
site_scores_coin_con_syn.out1.9.wet.pc.c05$clus.doc.3<-as.factor(kmeans_sites.9.doc.3.wet.pc.c05$cluster)

library(indicspecies)
indic.9.doc.3.wet.pc.c05 <- multipatt(otu_congo_syn.9.ra.t, site_scores_coin_con_syn.out1.9.wet.pc.c05$clus.doc.3, control=how(nperm=999))
indic.9.doc.3.wet.pc.c05.results <- as.data.frame(indic.9.doc.3.wet.pc.c05$sign)
indic.9.doc.3.wet.pc.c05.results.2 <- subset(indic.9.doc.3.wet.pc.c05.results, p.value<0.05&!is.na(p.value))
indic.9.doc.3.wet.pc.c05.results.2$OTU_ID2 <- row.names(indic.9.doc.3.wet.pc.c05.results.2)

indic.9.doc.3.wet.pc.c05.results.2.tax <- merge(indic.9.doc.3.wet.pc.c05.results.2,tax,by="OTU_ID2")
write.csv(indic.9.doc.3.wet.pc.c05.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.doc.3.wet.pc.c05.results.3clust.tax.csv")

summary(indic.9.doc.3.wet.pc.c05, indvalcomp=TRUE)

# cluster color (n=3) for DOC in wet.pc.c05
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.doc.3.wet.pc.c05.3clust.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.c05,aes(NorS1,NorS2,label=river, color=clus.doc.3), size=4, fontface="bold")+
  #scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.c05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("25.0% Projected Inertia")+ # 0.6726789*37.159 = 24.99608
  ylab("14.6% Projected Inertia")+ # 0.6726789*21.694= 14.5931
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.c05, RV=0.673 kmeans=3 for DOC")
dev.off()

# add sample number and site_id to end
site_scores_coin_con_syn.out1.9.wet.pc.c05$sample_nr <- site_scores_coin_con_syn.out1.9.wet.pc.c05$Row.names
site_scores_coin_con_syn.out1.9.wet.pc.c05$site_name <- site_scores_coin_con_syn.out1.9.wet.pc.c05$site_id.1
kmeans_metadata.9.wet.pc.c05 <- site_scores_coin_con_syn.out1.9.wet.pc.c05[, 216:221]
write.csv(kmeans_metadata.9.wet.pc.c05, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/kmeans_metadata.9.wet.pc.c05.csv")
```


# Set 9 - wet_pc_u05 CI + indicator - OLD
```{r}
########################
# wet_pc_u05 co-inertia and indicator species
########################

congo_syn_meta.out1.9.wet.pc.u05 <- congo_syn_meta.out1.9[,c("log10_tt", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_u05")]
# PCA
env_pca_congo_syn.out1.9.wet.pc.u05 <- dudi.pca(d=congo_syn_meta.out1.9.wet.pc.u05, row.w=otu_coa_congo_syn.out1.9$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.9.wet.pc.u05 <- coinertia(env_pca_congo_syn.out1.9.wet.pc.u05, otu_coa_congo_syn.out1.9, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.wet.pc.u05 <- as.data.frame(coin_con_syn.out1.9.wet.pc.u05$co)
enviro_loadings_coin.out1.9.wet.pc.u05$variable <- row.names(enviro_loadings_coin.out1.9.wet.pc.u05)

site_scores_coin_con_syn.out1.9.wet.pc.u05_0 <- as.data.frame(coin_con_syn.out1.9.wet.pc.u05$mX)
site_scores_coin_con_syn.out1.9.wet.pc.u05 <- bind_cols(site_scores_coin_con_syn.out1.9.wet.pc.u05_0, congo_syn_meta.out1.9)

# wet.pc.u05 color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.wet.pc.u05.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.u05,aes(NorS1,NorS2,label=river, color=wet_pc_u05), size=4, fontface="bold")+
  scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.6% Projected Inertia")+ # 0.6531755*36.20 = 23.64495
  ylab("15.2% Projected Inertia")+ # 0.6531755*23.33= 15.23858
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.u05, RV=0.653")
dev.off()

# DOC color in CI for wet.pc.u05
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.wet.pc.u05.DOC.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.u05,aes(NorS1,NorS2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="magenta")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.6% Projected Inertia")+ # 0.6531755*36.20 = 23.64495
  ylab("15.2% Projected Inertia")+ # 0.6531755*23.33= 15.23858
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.u05, RV=0.653")
dev.off()

    #________wet.pc.u05 - indicator species
# use factoextra to create elbow plot to find optimal number of kmeans clusters; use on CI site scores
library(factoextra)
set.seed(130)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/k.ebbow.coin_con_syn.out1.9.wet.pc.u05.pdf", width=11, height=8)
coords.coin_con_syn.out1.9.wet.pc.u05 <- site_scores_coin_con_syn.out1.9.wet.pc.u05[,c("NorS1", "NorS2")]
fviz_nbclust(coords.coin_con_syn.out1.9.wet.pc.u05, kmeans, method="wss")
dev.off() # on inspection, n=3 looks ideal

# run kmeans = 3 on metadata to prepare for indicator species
kmeans_sites.9.wet.pc.u05 <- kmeans(site_scores_coin_con_syn.out1.9.wet.pc.u05[,c("NorS1", "NorS2")], centers=3, iter.max=10, nstart=25)
site_scores_coin_con_syn.out1.9.wet.pc.u05$clus.wet.pc.u05<-as.factor(kmeans_sites.9.wet.pc.u05$cluster)

library(indicspecies)
indic.9.wet.pc.u05 <- multipatt(otu_congo_syn.9.ra.t, site_scores_coin_con_syn.out1.9.wet.pc.u05$clus, control=how(nperm=999))
indic.9.wet.pc.u05.results <- as.data.frame(indic.9.wet.pc.u05$sign)
indic.9.wet.pc.u05.results.2 <- subset(indic.9.wet.pc.u05.results, p.value<0.05&!is.na(p.value))
indic.9.wet.pc.u05.results.2$OTU_ID2 <- row.names(indic.9.wet.pc.u05.results.2)

indic.9.wet.pc.u05.results.2.tax <- merge(indic.9.wet.pc.u05.results.2,tax,by="OTU_ID2")
write.csv(indic.9.wet.pc.u05.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.wet.pc.u05.results.3clust.tax.csv")

summary(indic.9.wet.pc.u05, indvalcomp=TRUE)

# cluster color (n=3) for wet.pc.u05
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.wet.pc.u05.3clust.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.u05,aes(NorS1,NorS2,label=river, color=clus.wet.pc.u05), size=4, fontface="bold")+
  #scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.6% Projected Inertia")+ # 0.6726789*37.159 = 24.99608
  ylab("15.2% Projected Inertia")+ # 0.6726789*21.694= 14.5931
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.u05, RV=0.653 kmeans=3")
dev.off()

  # DOC indicator species on the wet.pc.u05 co-inertia
# standardize DOC data (z-score)
doc_all_congo_syn_meta.out1.9 <- as.data.frame(congo_syn_meta.out1.9$doc_mcc_l)
k.doc.scaled.1.9 <- scale(doc_all_congo_syn_meta.out1.9)
# use factoextra to create elbow plot to find optimal number of kmeans clusters; use on CI site scores
library(factoextra)
set.seed(133)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/k.ebbow.coin_con_syn.out1.9.DOC.pdf", width=11, height=8)
fviz_nbclust(k.doc.scaled.1.9, kmeans, method="wss")
dev.off() # on inspection, n=2 or 3 look good

# run kmeans = 3 on metadata to prepare for indicator species
kmeans_sites.9.doc.3.wet.pc.u05 <- kmeans(k.doc.scaled.1.9, centers=3, iter.max=10, nstart=25)
site_scores_coin_con_syn.out1.9.wet.pc.u05$clus.doc.3<-as.factor(kmeans_sites.9.doc.3.wet.pc.u05$cluster)

library(indicspecies)
indic.9.doc.3.wet.pc.u05 <- multipatt(otu_congo_syn.9.ra.t, site_scores_coin_con_syn.out1.9.wet.pc.u05$clus.doc.3, control=how(nperm=999))
indic.9.doc.3.wet.pc.u05.results <- as.data.frame(indic.9.doc.3.wet.pc.u05$sign)
indic.9.doc.3.wet.pc.u05.results.2 <- subset(indic.9.doc.3.wet.pc.u05.results, p.value<0.05&!is.na(p.value))
indic.9.doc.3.wet.pc.u05.results.2$OTU_ID2 <- row.names(indic.9.doc.3.wet.pc.u05.results.2)

indic.9.doc.3.wet.pc.u05.results.2.tax <- merge(indic.9.doc.3.wet.pc.u05.results.2,tax,by="OTU_ID2")
write.csv(indic.9.doc.3.wet.pc.u05.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.doc.3.wet.pc.u05.results.3clust.tax.csv")

summary(indic.9.doc.3.wet.pc.u05, indvalcomp=TRUE)

# cluster color (n=3) for doc in wet.pc.u05
#pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.doc.3.wet.pc.u05.3clust.pdf", width=9, height=8)
svg("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.doc.3.wet.pc.u05.3clust.svg", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.u05,aes(NorS1,NorS2,label=river, color=clus.doc.3), size=4, fontface="bold")+
  #scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.6% Projected Inertia")+ # 0.6531755*36.20 = 23.64495
  ylab("15.2% Projected Inertia")+ # 0.6531755*23.334= 15.23858
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.u05, RV=0.653 kmeans=3 for DOC")
dev.off()

  #  # CH4 indicator species on the wet.pc.u05 co-inertia
# standardize CH4 data (z-score)
CH4_all_congo_syn_meta.out1.9 <- as.data.frame(congo_syn_meta.out1.9$CH4_pct_sat)
k.CH4.scaled.1.9 <- scale(CH4_all_congo_syn_meta.out1.9)
# use factoextra to create elbow plot to find optimal number of kmeans clusters; use on CI site scores
library(factoextra)
set.seed(134)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/k.ebbow.coin_con_syn.out1.9.CH4.pdf", width=11, height=8)
fviz_nbclust(k.CH4.scaled.1.9, kmeans, method="wss")
dev.off() # on inspection, n=3 look good

# run kmeans = 3 on metadata to prepare for indicator species
kmeans_sites.9.ch4.3.wet.pc.u05 <- kmeans(k.CH4.scaled.1.9, centers=3, iter.max=10, nstart=25)
site_scores_coin_con_syn.out1.9.wet.pc.u05$clus.ch4.3<-as.factor(kmeans_sites.9.ch4.3.wet.pc.u05$cluster)

library(indicspecies)
indic.9.ch4.3.wet.pc.u05 <- multipatt(otu_congo_syn.9.ra.t, site_scores_coin_con_syn.out1.9.wet.pc.u05$clus.ch4.3, control=how(nperm=999))
indic.9.ch4.3.wet.pc.u05.results <- as.data.frame(indic.9.ch4.3.wet.pc.u05$sign)
indic.9.ch4.3.wet.pc.u05.results.2 <- subset(indic.9.ch4.3.wet.pc.u05.results, p.value<0.05&!is.na(p.value))
indic.9.ch4.3.wet.pc.u05.results.2$OTU_ID2 <- row.names(indic.9.ch4.3.wet.pc.u05.results.2)

indic.9.ch4.3.wet.pc.u05.results.2.tax <- merge(indic.9.ch4.3.wet.pc.u05.results.2,tax,by="OTU_ID2")
write.csv(indic.9.ch4.3.wet.pc.u05.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.ch4.3.wet.pc.u05.results.3clust.tax.csv")

summary(indic.9.ch4.3.wet.pc.u05, indvalcomp=TRUE)

# cluster color (n=3) for ch4 in wet.pc.u05
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.ch4.3.wet.pc.u05.3clust.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.u05,aes(NorS1,NorS2,label=river, color=clus.ch4.3), size=4, fontface="bold")+
  #scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.6% Projected Inertia")+ #  0.6531755*36.20 = 23.64495
  ylab("15.2% Projected Inertia")+ # 0.6531755*23.334= 15.23858
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.u05, RV=0.653 kmeans=3 for CH4")
dev.off()

  # pH indicator on wet.pc.u05
# run kmeans = 3 on metadata to prepare for indicator species
kmeans.pH.wet.pc.u05.3 <- kmeans(k.pH.scaled, centers=3, iter.max=10, nstart=25)
site_scores_coin_con_syn.out1.9.wet.pc.u05$clus.pH.3<-as.factor(kmeans.pH.wet.pc.u05.3$cluster)

library(indicspecies)
indic.9.pH.3.wet.pc.u05 <- multipatt(otu_congo_syn.9.ra.t, site_scores_coin_con_syn.out1.9.wet.pc.u05$clus.pH.3, control=how(nperm=999))
indic.9.pH.3.wet.pc.u05.results <- as.data.frame(indic.9.pH.3.wet.pc.u05$sign)
indic.9.pH.3.wet.pc.u05.results.2 <- subset(indic.9.pH.3.wet.pc.u05.results, p.value<0.05&!is.na(p.value))
indic.9.pH.3.wet.pc.u05.results.2$OTU_ID2 <- row.names(indic.9.pH.3.wet.pc.u05.results.2)

indic.9.pH.3.wet.pc.u05.results.2.tax <- merge(indic.9.pH.3.wet.pc.u05.results.2,tax,by="OTU_ID2")

write.csv(indic.9.pH.3.wet.pc.u05.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.pH.3.wet.pc.u05.results.3clust.tax.csv")

summary(indic.9.pH.3.wet.pc.u05, indvalcomp=TRUE)

# add sample number and site_id to end
site_scores_coin_con_syn.out1.9.wet.pc.u05$sample_nr <- site_scores_coin_con_syn.out1.9.wet.pc.u05$Row.names
site_scores_coin_con_syn.out1.9.wet.pc.u05$site_name <- site_scores_coin_con_syn.out1.9.wet.pc.u05$site_id.1

# cluster color (n=3) for pH in wet.pc.u05 co-inertia (coin_con_syn.out1.9.wet.pc.u05)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.pH.3.wet.pc.u05.3clust.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.u05,aes(NorS1,NorS2,label=river, color=clus.pH.3), size=4, fontface="bold")+
  #scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.6% Projected Inertia")+ #  0.6531755*36.20 = 23.64495
  ylab("15.2% Projected Inertia")+ # 0.6531755*23.334= 15.23858
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.u05, RV=0.653 kmeans=3 by pH")
dev.off()

  # TT indicator on wet.pc.u05
# run k-means on metadata
tt_congo_syn_meta.out1.9 <- as.data.frame(congo_syn_meta.out1.9$tt_res_hr)
k.tt.scaled.1.9 <- scale(tt_congo_syn_meta.out1.9)
# use factoextra to create elbow plot to find optimal number of kmeans clusters; use on CI site scores
library(factoextra)
set.seed(135)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/k.elbow.coin_con_syn.out1.9.tt.pdf", width=11, height=8)
fviz_nbclust(k.tt.scaled.1.9, kmeans, method="wss")
dev.off() # on inspection, n=3 look good

kmeans.tt.wet.pc.u05.3 <- kmeans(k.tt.scaled.1.9, centers=3, iter.max=10, nstart=25)
site_scores_coin_con_syn.out1.9.wet.pc.u05$clus.tt.3<-as.factor(kmeans.tt.wet.pc.u05.3$cluster)

library(indicspecies)
indic.9.tt.3.wet.pc.u05 <- multipatt(otu_congo_syn.9.ra.t, site_scores_coin_con_syn.out1.9.wet.pc.u05$clus.tt.3, control=how(nperm=999))
indic.9.tt.3.wet.pc.u05.results <- as.data.frame(indic.9.tt.3.wet.pc.u05$sign)
indic.9.tt.3.wet.pc.u05.results.2 <- subset(indic.9.tt.3.wet.pc.u05.results, p.value<0.05&!is.na(p.value))
indic.9.tt.3.wet.pc.u05.results.2$OTU_ID2 <- row.names(indic.9.tt.3.wet.pc.u05.results.2)

indic.9.tt.3.wet.pc.u05.results.2.tax <- merge(indic.9.tt.3.wet.pc.u05.results.2,tax,by="OTU_ID2")

write.csv(indic.9.tt.3.wet.pc.u05.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.tt.3,wet.pc.u05.results.3clust.tax.csv")

summary(indic.9.tt.3.wet.pc.u05)

# cluster color (n=3) for TT in wet.pc.u05 co-inertia (coin_con_syn.out1.9.wet.pc.u05)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.tt.3.wet.pc.u05.3clust.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.u05,aes(NorS1,NorS2,label=river, color=clus.tt.3), size=4, fontface="bold")+
  #scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.u05, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.6% Projected Inertia")+ #  0.6531755*36.20 = 23.64495
  ylab("15.2% Projected Inertia")+ # 0.6531755*23.334= 15.23858
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.u05, RV=0.653 kmeans=3 by TT")
dev.off()

# make csv of the metadata variables kmeans groupings and site names
# add sample number and site_id to end
site_scores_coin_con_syn.out1.9.wet.pc.u05$sample_nr <- site_scores_coin_con_syn.out1.9.wet.pc.u05$Row.names
site_scores_coin_con_syn.out1.9.wet.pc.u05$site_name <- site_scores_coin_con_syn.out1.9.wet.pc.u05$site_id.1
kmeans_metadata.9.wet.pc.u05 <- site_scores_coin_con_syn.out1.9.wet.pc.u05[, 216:221]
write.csv(kmeans_metadata.9.wet.pc.u05, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/kmeans_metadata.9.wet.pc.u05.csv")




#############################
# wet_pc_ug2
#############################
congo_syn_meta.out1.9.wet.pc.ug2 <- congo_syn_meta.out1.9[,c("log10_tt", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_ug2")]
# PCA
env_pca_congo_syn.out1.9.wet.pc.ug2 <- dudi.pca(d=congo_syn_meta.out1.9.wet.pc.ug2, row.w=otu_coa_congo_syn.out1.9$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.9.wet.pc.ug2 <- coinertia(env_pca_congo_syn.out1.9.wet.pc.ug2, otu_coa_congo_syn.out1.9, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.wet.pc.ug2 <- as.data.frame(coin_con_syn.out1.9.wet.pc.ug2$co)
enviro_loadings_coin.out1.9.wet.pc.ug2$variable <- row.names(enviro_loadings_coin.out1.9.wet.pc.ug2)

site_scores_coin_con_syn.out1.9.wet.pc.ug2_0 <- as.data.frame(coin_con_syn.out1.9.wet.pc.ug2$mX)
site_scores_coin_con_syn.out1.9.wet.pc.ug2 <- bind_cols(site_scores_coin_con_syn.out1.9.wet.pc.ug2_0, congo_syn_meta.out1.9)

# wet.pc.ug2 color
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.wet.pc.ug2.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.ug2,aes(NorS1,NorS2,label=river, color=wet_pc_ug2), size=4, fontface="bold")+
  scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.ug2, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.ug2, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.5% Projected Inertia")+ # 0.6515844*36.03 = 23.47659
  ylab("15.3% Projected Inertia")+ # 0.6515844*23.55= 15.34481
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.ug2, RV=0.652")
dev.off()

    #________wet.pc.ug2 - indicator species
# use factoextra to create elbow plot to find optimal number of kmeans clusters; use on CI site scores
library(factoextra)
set.seed(131)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/k.ebbow.coin_con_syn.out1.9.wet.pc.ug2.pdf", width=11, height=8)
coords.coin_con_syn.out1.9.wet.pc.ug2 <- site_scores_coin_con_syn.out1.9.wet.pc.ug2[,c("NorS1", "NorS2")]
fviz_nbclust(coords.coin_con_syn.out1.9.wet.pc.ug2, kmeans, method="wss")
dev.off() # on inspection, n=3 looks ideal

# run kmeans = 3 on metadata to prepare for indicator species
kmeans_sites.9.wet.pc.ug2 <- kmeans(site_scores_coin_con_syn.out1.9.wet.pc.ug2[,c("NorS1", "NorS2")], centers=3, iter.max=10, nstart=25)
site_scores_coin_con_syn.out1.9.wet.pc.ug2$clus<-as.factor(kmeans_sites.9.wet.pc.ug2$cluster)

library(indicspecies)
indic.9.wet.pc.ug2 <- multipatt(otu_congo_syn.9.ra.t, site_scores_coin_con_syn.out1.9.wet.pc.ug2$clus, control=how(nperm=999))
indic.9.wet.pc.ug2.results <- as.data.frame(indic.9.wet.pc.ug2$sign)
indic.9.wet.pc.ug2.results.2 <- subset(indic.9.wet.pc.ug2.results, p.value<0.05&!is.na(p.value))
indic.9.wet.pc.ug2.results.2$OTU_ID2 <- row.names(indic.9.wet.pc.ug2.results.2)

indic.9.wet.pc.ug2.results.2.tax <- merge(indic.9.wet.pc.ug2.results.2,tax,by="OTU_ID2")
write.csv(indic.9.wet.pc.ug2.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.wet.pc.ug2.results.3clust.tax.csv")

# cluster color (n=3) for wet.pc.ug2
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/coin.con.syn.out1.9.wet.pc.ug2.3clust.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.ug2,aes(NorS1,NorS2,label=river, color=clus), size=4, fontface="bold")+
  #scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.ug2, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.ug2, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.5% Projected Inertia")+ # 0.6726789*37.159 = 24.99608
  ylab("15.3% Projected Inertia")+ # 0.6726789*21.694= 14.5931
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.ug2, RV=0.673 kmeans=3")
dev.off()
```

# Set 9 - wet_pc_uall clustering - new
```{r}
########################
# wet_pc_uall
########################
# metadata
congo_syn_meta.out1.9.wet_pc_uall <- congo_syn_meta.out1.9[,c("Row.names", "river", "log10_tt", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_uall")]

# coa_coords
coa_coords.9 <- as.data.frame(otu_coa_congo_syn.out1.9$li)
coa_coords.9$Row.names <- congo_syn_meta.out1.9$Row.names
coa_coords.9.meta <- merge(congo_syn_meta.out1.9.wet_pc_uall, coa_coords.9, by="Row.names", all.x=TRUE) 

# optimal wet_pc_uall bins
# n=2
coa_coords.9.meta$qcut_2_wet_pc_uall <-gtools::quantcut(coa_coords.9.meta$wet_pc_uall, q=2, na.rm=TRUE)
uall_centroid_qcut_2 <- coa_coords.9.meta[,c("Axis1","Axis2","qcut_2_wet_pc_uall")] %>% group_by(qcut_2_wet_pc_uall) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut2 <- merge(uall_centroid_qcut_2,coa_coords.9.meta[,c("Axis1","Axis2","qcut_2_wet_pc_uall")],by="qcut_2_wet_pc_uall",wc.y=TRUE)
coa_coords.9.qcut2$distance_to_centroid<-sqrt((coa_coords.9.qcut2$Axis1-coa_coords.9.qcut2$Axis1_mean)^2+(coa_coords.9.qcut2$Axis2-coa_coords.9.qcut2$Axis2_mean)^2)
bin2_centroid_dist_uall<-mean(coa_coords.9.qcut2$distance_to_centroid)

# n=3
coa_coords.9.meta$qcut_3_wet_pc_uall <-gtools::quantcut(coa_coords.9.meta$wet_pc_uall, q=3, na.rm=TRUE)
uall_centroid_qcut_3 <- coa_coords.9.meta[,c("Axis1","Axis2","qcut_3_wet_pc_uall")] %>% group_by(qcut_3_wet_pc_uall) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut3 <- merge(uall_centroid_qcut_3,coa_coords.9.meta[,c("Axis1","Axis2","qcut_3_wet_pc_uall")],by="qcut_3_wet_pc_uall",wc.y=TRUE)
coa_coords.9.qcut3$distance_to_centroid<-sqrt((coa_coords.9.qcut3$Axis1-coa_coords.9.qcut3$Axis1_mean)^2+(coa_coords.9.qcut3$Axis2-coa_coords.9.qcut3$Axis2_mean)^2)
bin3_centroid_dist_uall<-mean(coa_coords.9.qcut3$distance_to_centroid)

#n=4
coa_coords.9.meta$qcut_4_wet_pc_uall <-gtools::quantcut(coa_coords.9.meta$wet_pc_uall, q=4, na.rm=TRUE)
uall_centroid_qcut_4 <- coa_coords.9.meta[,c("Axis1","Axis2","qcut_4_wet_pc_uall")] %>% group_by(qcut_4_wet_pc_uall) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut4 <- merge(uall_centroid_qcut_4,coa_coords.9.meta[,c("Axis1","Axis2","qcut_4_wet_pc_uall")],by="qcut_4_wet_pc_uall",wc.y=TRUE)
coa_coords.9.qcut4$distance_to_centroid<-sqrt((coa_coords.9.qcut4$Axis1-coa_coords.9.qcut4$Axis1_mean)^2+(coa_coords.9.qcut4$Axis2-coa_coords.9.qcut4$Axis2_mean)^2)
bin4_centroid_dist_uall<-mean(coa_coords.9.qcut4$distance_to_centroid)

#n=5
coa_coords.9.meta$qcut_5_wet_pc_uall <-gtools::quantcut(coa_coords.9.meta$wet_pc_uall, q=5, na.rm=TRUE)
uall_centroid_qcut_5 <- coa_coords.9.meta[,c("Axis1","Axis2","qcut_5_wet_pc_uall")] %>% group_by(qcut_5_wet_pc_uall) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut5 <- merge(uall_centroid_qcut_5,coa_coords.9.meta[,c("Axis1","Axis2","qcut_5_wet_pc_uall")],by="qcut_5_wet_pc_uall",wc.y=TRUE)
coa_coords.9.qcut5$distance_to_centroid<-sqrt((coa_coords.9.qcut5$Axis1-coa_coords.9.qcut5$Axis1_mean)^2+(coa_coords.9.qcut5$Axis2-coa_coords.9.qcut5$Axis2_mean)^2)
bin5_centroid_dist_uall<-mean(coa_coords.9.qcut5$distance_to_centroid)

#n=6
coa_coords.9.meta$qcut_6_wet_pc_uall <-gtools::quantcut(coa_coords.9.meta$wet_pc_uall, q=6, na.rm=TRUE)
uall_centroid_qcut_6 <- coa_coords.9.meta[,c("Axis1","Axis2","qcut_6_wet_pc_uall")] %>% group_by(qcut_6_wet_pc_uall) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut6 <- merge(uall_centroid_qcut_6,coa_coords.9.meta[,c("Axis1","Axis2","qcut_6_wet_pc_uall")],by="qcut_6_wet_pc_uall",wc.y=TRUE)
coa_coords.9.qcut6$distance_to_centroid<-sqrt((coa_coords.9.qcut6$Axis1-coa_coords.9.qcut6$Axis1_mean)^2+(coa_coords.9.qcut6$Axis2-coa_coords.9.qcut6$Axis2_mean)^2)
bin6_centroid_dist_uall<-mean(coa_coords.9.qcut6$distance_to_centroid)

#n=7
coa_coords.9.meta$qcut_7_wet_pc_uall <-gtools::quantcut(coa_coords.9.meta$wet_pc_uall, q=7, na.rm=TRUE)
uall_centroid_qcut_7 <- coa_coords.9.meta[,c("Axis1","Axis2","qcut_7_wet_pc_uall")] %>% group_by(qcut_7_wet_pc_uall) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut7 <- merge(uall_centroid_qcut_7,coa_coords.9.meta[,c("Axis1","Axis2","qcut_7_wet_pc_uall")],by="qcut_7_wet_pc_uall",wc.y=TRUE)
coa_coords.9.qcut7$distance_to_centroid<-sqrt((coa_coords.9.qcut7$Axis1-coa_coords.9.qcut7$Axis1_mean)^2+(coa_coords.9.qcut7$Axis2-coa_coords.9.qcut7$Axis2_mean)^2)
bin7_centroid_dist_uall<-mean(coa_coords.9.qcut7$distance_to_centroid)

#n=8
coa_coords.9.meta$qcut_8_wet_pc_uall <-gtools::quantcut(coa_coords.9.meta$wet_pc_uall, q=8, na.rm=TRUE)
uall_centroid_qcut_8 <- coa_coords.9.meta[,c("Axis1","Axis2","qcut_8_wet_pc_uall")] %>% group_by(qcut_8_wet_pc_uall) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut8 <- merge(uall_centroid_qcut_8,coa_coords.9.meta[,c("Axis1","Axis2","qcut_8_wet_pc_uall")],by="qcut_8_wet_pc_uall",wc.y=TRUE)
coa_coords.9.qcut8$distance_to_centroid<-sqrt((coa_coords.9.qcut8$Axis1-coa_coords.9.qcut8$Axis1_mean)^2+(coa_coords.9.qcut8$Axis2-coa_coords.9.qcut8$Axis2_mean)^2)
bin8_centroid_dist_uall<-mean(coa_coords.9.qcut8$distance_to_centroid)

#n=9
coa_coords.9.meta$qcut_9_wet_pc_uall <-gtools::quantcut(coa_coords.9.meta$wet_pc_uall, q=9, na.rm=TRUE)
uall_centroid_qcut_9 <- coa_coords.9.meta[,c("Axis1","Axis2","qcut_9_wet_pc_uall")] %>% group_by(qcut_9_wet_pc_uall) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut9 <- merge(uall_centroid_qcut_9,coa_coords.9.meta[,c("Axis1","Axis2","qcut_9_wet_pc_uall")],by="qcut_9_wet_pc_uall",wc.y=TRUE)
coa_coords.9.qcut9$distance_to_centroid<-sqrt((coa_coords.9.qcut9$Axis1-coa_coords.9.qcut9$Axis1_mean)^2+(coa_coords.9.qcut9$Axis2-coa_coords.9.qcut9$Axis2_mean)^2)
bin9_centroid_dist_uall<-mean(coa_coords.9.qcut9$distance_to_centroid)

#n=10 -- does not work, says "breaks are not unique" - prob too many bins for the small sample size
coa_coords.9.meta$qcut_10_wet_pc_uall <-gtools::quantcut(coa_coords.9.meta$wet_pc_uall, q=10, na.rm=TRUE)
uall_centroid_qcut_10 <- coa_coords.9.meta[,c("Axis1","Axis2","qcut_10_wet_pc_uall")] %>% group_by(qcut_10_wet_pc_uall) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut10 <- merge(uall_centroid_qcut_10,coa_coords.9.meta[,c("Axis1","Axis2","qcut_10_wet_pc_uall")],by="qcut_10_wet_pc_uall",wc.y=TRUE)
coa_coords.9.qcut10$distance_to_centroid<-sqrt((coa_coords.9.qcut10$Axis1-coa_coords.9.qcut10$Axis1_mean)^2+(coa_coords.9.qcut10$Axis2-coa_coords.9.qcut10$Axis2_mean)^2)
bin10_centroid_dist_uall<-mean(coa_coords.9.qcut10$distance_to_centroid)

# elbow plot of distance to group centroid
elbow_data_wet_pc_uall<-data.frame(bins=c(2,3,4,5,6,7,8,9),distances=c(bin2_centroid_dist_uall,bin3_centroid_dist_uall,bin4_centroid_dist_uall,bin5_centroid_dist_uall,bin6_centroid_dist_uall,bin7_centroid_dist_uall,bin8_centroid_dist_uall,bin9_centroid_dist_uall))

# histogram of wet_pc_uall distribution across samples
hist(coa_coords.9.meta$wet_pc_uall, main="Histogram: wet.pc.uall sample distribution", xlab="Percent upstream wetland", col="blue", las=1, breaks=20)
# color using 2 groups of wet_pc_uall based on histogram: high and low
coa_coords.9.meta$wet_pc_uall_hilo <- ifelse(coa_coords.9.meta$wet_pc_uall <=58, "low",
                                             ifelse(coa_coords.9.meta$wet_pc_uall >=58, "high", "null"))

pdf("output/elbow_wet_pc_uall.pdf") # 5 bins looks best
#svglite("elbow_wet_pc_uall.svg")
ggplot(elbow_data_wet_pc_uall, aes(x=bins, y=distances))+
  geom_point(size=5, color="black")+
  theme_classic()+
  scale_x_continuous(breaks=seq(2, 9, by=1))+ #("2", "3", "4", "5", "6", "7", "8", "9"))+
  xlab("Number of bins")+
  ylab("Mean distance to centroid")+
  ggtitle("Eblow plot - wet_pc_uall mean distance to centroid")
dev.off()

#make baseplot
pdf("output/qint_coa_ellip.wet.pc.uall.9.pdf")
qint_coa_ellip.wet.pc.uall.9 <- gg_ordiplot(otu_coa_congo_syn.out1.9, groups=coa_coords.9.meta$qcut_5_wet_pc_uall, kind="se", conf=0.95)
dev.off()

# extract info to customize in ggplot
qint_coa_ellip.wet.pc.uall.9.plot <- qint_coa_ellip.wet.pc.uall.9$plot

# extract plotting values manually from the quint_coa_ellip ordination object, to fully customize using ggplot
# add x and y coordinates
coa_coords.9.meta$quint_wet_pc_uall_ord_x<-qint_coa_ellip.wet.pc.uall.9$df_ord$x
coa_coords.9.meta$quint_wet_pc_uall_ord_y<-qint_coa_ellip.wet.pc.uall.9$df_ord$y

# main figure, no ellipses, 5 bins wet_pc_uall
pdf("output/qint_coa_ellip.wet.pc.uall.9.v2.pdf", width=11, height=8)
#svglite("output/qint_coa_ellip.wet.pc.uall.9.v2.svg", width=11, height=8)
ggplot(coa_coords.9.meta, aes(x=quint_wet_pc_uall_ord_x,y=quint_wet_pc_uall_ord_y, label=river, color=qcut_5_wet_pc_uall))+
  geom_point(size=5)+
  theme_classic()+
  #geom_path(data = quint_coa_ellip$df_ellipse, aes(x=x, y=y, color=Group), show.legend=TRUE, inherit.aes=FALSE)+
#scale_fill_viridis(discrete=TRUE,option="C",name="Travel Time Group (Days)")+
#scale_shape_manual(values=c(23,21))+
#  scale_x_reverse()+
#  scale_y_reverse()+
#guides(fill = guide_legend(override.aes=list(shape=21)))+
  xlab("14.2% projected inertia")+
  ylab("12.0% projected inertia")+
ggtitle("COA quint Color-Coded by wet_pc_uall Groups - using ordiplot")
dev.off()

# add ellipses
pdf("output/qint_coa_ellip.wet.pc.uall.9.v3.pdf", width=11, height=8)
#svglite("quart_coa_ellip_v4.r2.svg", width=11, height=8)
ggplot(coa_coords.9.meta, aes(x=quint_wet_pc_uall_ord_x,y=quint_wet_pc_uall_ord_y, label=river, color=qcut_5_wet_pc_uall))+
  geom_point(size=5)+
  theme_classic()+
  geom_path(data = qint_coa_ellip.wet.pc.uall.9$df_ellipse, aes(x=x, y=y, color=Group), linewidth=1, show.legend=TRUE, inherit.aes=FALSE)+
#scale_fill_viridis(discrete=TRUE,option="C",name="Travel Time Group (Days)")+
#scale_shape_manual(values=c(23,21))+
#  scale_x_reverse()+
#  scale_y_reverse()+
#guides(fill = guide_legend(override.aes=list(shape=21)))+
  xlab("14.2% projected inertia")+
  ylab("12.0% projected inertia")+
ggtitle("COA quint Color-Coded by wet_pc_uall Groups - using ordiplot")
dev.off()

### k-means multivariate clustering based on top CI variables
#subset to top 3 vectors in each axis from coin_con_syn.out1.9.wet.pc.uall (below)
kmeans.coin.1.9.wet.pc.uall.topvars <- congo_syn_meta.out1.9.wet.pc.uall[,c("pH", "log10_DOC", "wet_pc_uall", "log10_CO2_pct_sat", "log10_PO4", "log10_do_mg_l")]
library(factoextra)
set.seed(100)
pdf("output/k.elbow.kmeans.coin.1.9.wet.pc.uall.topvars.pdf", width=11, height=8)
fviz_nbclust(kmeans.coin.1.9.wet.pc.uall.topvars, kmeans, method="wss")
dev.off()

kmeans.coin.1.9.wet.pc.uall.topvars.cluster <-kmeans(kmeans.coin.1.9.wet.pc.uall.topvars, centers=2, iter.max=10, nstart=25)
congo_syn_meta.out1.9.wet_pc_uall$clus.topvars <- as.factor(kmeans.coin.1.9.wet.pc.uall.topvars.cluster$cluster)


```

# Set 9 - wet_pc_uall CI with all hydro variables - to see which hydros dominate small streams
```{r}
congo_syn_meta.out1.9.allhydro <- congo_syn_meta.out1.9[,c("log10_tt", "log10_q_daily_mean", "log10_DIST_UP_KM", "log10_UPLAND_SKM", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_uall")]
# PCA
env_pca_congo_syn.out1.9.allhydro <- dudi.pca(d=congo_syn_meta.out1.9.allhydro, row.w=otu_coa_congo_syn.out1.9$lw, scale = TRUE, scannf = FALSE, nf = 10)
# CI
coin_con_syn.out1.9.allhydro <- coinertia(env_pca_congo_syn.out1.9.allhydro, otu_coa_congo_syn.out1.9, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.allhydro <- as.data.frame(coin_con_syn.out1.9.allhydro$co)
enviro_loadings_coin.out1.9.allhydro$variable <- row.names(enviro_loadings_coin.out1.9.allhydro)

site_scores_coin_con_syn.out1.9.allhydro_0 <- as.data.frame(coin_con_syn.out1.9.allhydro$mX)
site_scores_coin_con_syn.out1.9.allhydro <- bind_cols(site_scores_coin_con_syn.out1.9.allhydro_0, congo_syn_meta.out1.9)

# color by dendritic distance (strongest hydro var)
pdf("output/coin.con.syn.out1.9.allhydro.dd.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.allhydro,aes(NorS1,NorS2,label=river, color=log10_DIST_UP_KM), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  #scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.allhydro, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.allhydro, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("20.9% Projected Inertia")+ # 0.6411369*32.673 = 20.94787
  ylab("18.4% Projected Inertia")+ # 0.6411369*28.689= 18.39358
  ggtitle("Congo Synoptic Set 9 allhydro-coin.con.syn.out1.9.allhydro.dd.pdf, RV=0.641")
dev.off()


```



# Set 9 - wet_pc_uall-based CI + indicator - using TT
```{r}
congo_syn_meta.out1.9.wet.pc.uall <- congo_syn_meta.out1.9[,c("log10_tt_nores", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_uall")]
# PCA
env_pca_congo_syn.out1.9.wet.pc.uall <- dudi.pca(d=congo_syn_meta.out1.9.wet.pc.uall, row.w=otu_coa_congo_syn.out1.9$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.9.wet.pc.uall <- coinertia(env_pca_congo_syn.out1.9.wet.pc.uall, otu_coa_congo_syn.out1.9, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.wet.pc.uall <- as.data.frame(coin_con_syn.out1.9.wet.pc.uall$co)
enviro_loadings_coin.out1.9.wet.pc.uall$variable <- row.names(enviro_loadings_coin.out1.9.wet.pc.uall)

site_scores_coin_con_syn.out1.9.wet.pc.uall_0 <- as.data.frame(coin_con_syn.out1.9.wet.pc.uall$mX)
site_scores_coin_con_syn.out1.9.wet.pc.uall <- bind_cols(site_scores_coin_con_syn.out1.9.wet.pc.uall_0, coa_coords.9.meta)

# wet_pc_uall - 5 clusters from elbow_wet_pc_uall.pdf
pdf("output/coin.con.syn.out1.9.wet.pc.uall.5clust.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.uall,aes(NorS1,NorS2,label=river, color=qcut_5_wet_pc_uall), size=4, fontface="bold")+
  #scale_color_continuous(low="green", high="blue")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.uall, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.uall, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.5% Projected Inertia")+ # 0.6507404*36.065 = 23.46895
  ylab("15.3% Projected Inertia")+ # 0.6507404*23.498= 15.2911
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.uall, RV=0.651 - 5 bins wet.pc.uall")
dev.off()

# wet_pc_uall continuous color
pdf("output/coin.con.syn.out1.9.wet.pc.uall.2.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.uall,aes(NorS1,NorS2,label=river, color=wet_pc_uall), size=5)+
  #scale_color_continuous(low="lightgreen", high="blue")+
  scale_color_continuous(low="khaki", high="maroon") +
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.uall, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.uall, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.5% Projected Inertia")+ # 0.6507404*36.065 = 23.46895
  ylab("15.3% Projected Inertia")+ # 0.6507404*23.498= 15.2911
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.uall.2, RV=0.651")
dev.off()

wet_hilo_color <- c("low"="lightgreen", "high"="blue")
# wet_pc_uall hi-lo color
pdf("output/coin.con.syn.out1.9.wet.pc.uall.hilo.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.uall,aes(NorS1,NorS2,label=river, color=wet_pc_uall_hilo), size=4, fontface="bold")+
  scale_color_manual(values=wet_hilo_color)+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.uall, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.uall, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.5% Projected Inertia")+ # 0.6507404*36.065 = 23.46895
  ylab("15.3% Projected Inertia")+ # 0.6507404*23.498= 15.2911
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.uall, RV=0.651 - 5 bins wet.pc.uall")
dev.off()

# log10_DOC color
pdf("output/coin.con.syn.out1.9.wet.pc.uall.log10DOC.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.uall,aes(NorS1,NorS2,label=river, color=log10_DOC), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="magenta")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.uall, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.uall, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.5% Projected Inertia")+ # 0.6507404*36.065 = 23.46895
  ylab("15.3% Projected Inertia")+ # 0.6507404*23.498= 15.2911
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.uall, RV=0.651 - 5 bins wet.pc.uall")
dev.off()

# CH4 color
pdf("output/coin.con.syn.out1.9.wet.pc.uall.log10CH4.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.uall,aes(NorS1,NorS2,label=river, color=log10_CH4_pct_sat), size=4, fontface="bold")+
  scale_color_gradient(low="blue", high="yellow")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.uall, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.uall, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.5% Projected Inertia")+ # 0.6507404*36.065 = 23.46895
  ylab("15.3% Projected Inertia")+ # 0.6507404*23.498= 15.2911
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.uall, RV=0.651 - 5 bins wet.pc.uall")
dev.off()

# pH color
pdf("output/coin.con.syn.out1.9.wet.pc.uall.pH.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.uall,aes(NorS1,NorS2,label=river, color=pH), size=4, fontface="bold")+
  scale_color_gradient(low="yellow", high="magenta")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.uall, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.uall, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.5% Projected Inertia")+ # 0.6507404*36.065 = 23.46895
  ylab("15.3% Projected Inertia")+ # 0.6507404*23.498= 15.2911
  ggtitle("Congo Synoptic co-inertia.outliers1.9.wet.pc.uall, RV=0.651 - 5 bins wet.pc.uall")
dev.off()

##############
# try CI using PCoA as input
## tt - 9.1 - wuni using dudi.pco w/cailliez correction on unifrac matrix first 
# original otu table: otu_con_syn.out1.9
otu_con_syn.out1.9.t <- t(otu_con_syn.out1.9)
otu.9.wuni <- beta.div(otu_con_syn.out1.9.t, "unifrac", weighted=TRUE, tree=tree)
# test if weighted unifrac distance matrix is euclidean (required for PCoA)
is.euclid(otu.9.wuni) # false
# convert wuni to euclidean via transformation
otu.9.wuni.euc <- cailliez(otu.9.wuni)
is.euclid(otu.9.wuni.euc) # true

pcoa.9.wuni.dudi <- dudi.pco(otu.9.wuni.euc, scannf=FALSE)

# pull out coordinates
pcoa.9.wuni.dudi.li <- as.data.frame(pcoa.9.wuni.dudi$li)
pcoa.9.wuni.dudi.li.1 <- -1*pcoa.9.wuni.dudi.li[,1]
pcoa.9.wuni.dudi.li.2 <- -2*pcoa.9.wuni.dudi.li[,2]

congo_syn_meta.out1.9.pcoa <- congo_syn_meta.out1.9

congo_syn_meta.out1.9.pcoa$pcoa.9.wuni.dudi.li.1 <- as.numeric(pcoa.9.wuni.dudi.li.1)
congo_syn_meta.out1.9.pcoa$pcoa.9.wuni.dudi.li.2 <- as.numeric(pcoa.9.wuni.dudi.li.2)

# wet_pc_uall - wuni dudi.pco cailliez
pdf("output/pcoa.9.wuni.wet.pc.uall.dudi.euc.pdf", width=9, height=8)
ggplot()+
  geom_text(data=congo_syn_meta.out1.9.pcoa, aes(pcoa.9.wuni.dudi.li.1, pcoa.9.wuni.dudi.li.2, label=site_id.1, color=wet_pc_uall), size=4, position=position_nudge(y=0.003), fontface="bold")+
              theme_classic()+
  #scale_fill_brewer(palette="Set1")+
  scale_color_continuous(low="lightgreen", high="blue")+
  #scale_x_reverse()+
  #scale_y_reverse()+
  xlab("35.5% Projected Inertia")+ 
  ylab("17.3% Projected Inertia")+ 
  ggtitle("PCoA set 9 wuni ASV - dudi.pco cailliez trans")
dev.off()

print(pcoa.9.wuni.dudi$eig)

# co-inertia using PCoA as input - 9.1
# make new PCA with weighted PCoA row weights
env_pca_congo_syn.out1.9.1.wet.pc.uall <- dudi.pca(d=congo_syn_meta.out1.9.wet.pc.uall, row.w=pcoa.9.wuni.dudi$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.9.1.wet.pc.uall.pcoa <- coinertia(env_pca_congo_syn.out1.9.1.wet.pc.uall, pcoa.9.wuni.dudi, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.1.wet.pc.uall <- as.data.frame(coin_con_syn.out1.9.1.wet.pc.uall.pcoa$co)
enviro_loadings_coin.out1.9.1.wet.pc.uall$variable <- row.names(enviro_loadings_coin.out1.9.1.wet.pc.uall)

site_scores_coin_con_syn.out1.9.1.wet.pc.uall_0 <- as.data.frame(coin_con_syn.out1.9.1.wet.pc.uall.pcoa$mX)
site_scores_coin_con_syn.out1.9.1.wet.pc.uall <- bind_cols(site_scores_coin_con_syn.out1.9.1.wet.pc.uall_0, coa_coords.9.meta)

pdf("output/coin.con.syn.out1.9.1.wet.pc.uall.pcoa.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.1.wet.pc.uall,aes(NorS1,NorS2,label=river, color=wet_pc_uall), size=4, fontface="bold")+
  scale_color_continuous(low="lightgreen", high="blue")+
  theme_classic()+
  #scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.1.wet.pc.uall, aes(x=0, y=0, xend=Comp1*30, yend=Comp2*30), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.1.wet.pc.uall, aes(Comp1*30, Comp2*30, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("13.8% Projected Inertia")+ # 0.3963724*34.786 = 13.78821
  ylab("12.2% Projected Inertia")+ # 0.3963724*30.704= 12.17022
  ggtitle("Congo Synoptic co-inertia.outliers1.9.1.wet.pc.uall w/PCoA, RV=0.397")
dev.off()


#############################
#### indicator species
#############################

# wet.pc.uall - indicator species
# bins hased on wet_pc_uall_hilo analysis above, from histogram and our manual cutoff bsaed on ecological knowledge (no kmeans, no quantcut, etc)
library(indicspecies)
indic.9.wet.pc.uall.hilo <- multipatt(otu_congo_syn.9.ra.t, site_scores_coin_con_syn.out1.9.wet.pc.uall$wet_pc_uall_hilo, control=how(nperm=999))
indic.9.wet.pc.uall.hilo.results <- as.data.frame(indic.9.wet.pc.uall.hilo$sign)
indic.9.wet.pc.uall.hilo.results.2 <- subset(indic.9.wet.pc.uall.hilo.results, p.value<0.05&!is.na(p.value))
indic.9.wet.pc.uall.hilo.results.2$OTU_ID2 <- row.names(indic.9.wet.pc.uall.hilo.results.2)

indic.9.wet.pc.uall.hilo.results.2.tax <- merge(indic.9.wet.pc.uall.hilo.results.2,tax,by="OTU_ID2")
write.csv(indic.9.wet.pc.uall.hilo.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.wet.pc.uall.hilo.results.2.tax.csv")

summary(indic.9.wet.pc.uall.hilo, indvalcomp=TRUE)

# CH4 - Byron wants 2 bins
  # histograms
hist(coa_coords.9.meta$log10_CH4_pct_sat, main="Histogram: log10_CH4 sample distribution", xlab="log10(CH4)", col="blue", las=1, breaks=20)
hist(congo_syn_meta.out1.9$CH4_pct_sat, main="Histogram: CH4_pct_sat sample distribution", xlab="CH4_pct_sat", col="blue", las=1, breaks=20)
ch4.zscore.1.9 <- scale(congo_syn_meta.out1.9$CH4_pct_sat)
hist(ch4.zscore.1.9, main="Histogram: CH4_pct_sat z-score sample distribution", xlab="scaled CH4_pct_sat", col="blue", las=1, breaks=20)
  # hi-low
congo_syn_meta.out1.9.indic <- congo_syn_meta.out1.9
congo_syn_meta.out1.9.indic$CH4_high_low <- ifelse(congo_syn_meta.out1.9.indic$CH4_pct_sat <=100000, "low",
                                      ifelse(congo_syn_meta.out1.9.indic >=100000, "high", "null"))
  # indicator
indic.9.CH4.hilo <- multipatt(otu_congo_syn.9.ra.t, congo_syn_meta.out1.9.indic$CH4_high_low, control=how(nperm=999))
indic.9.CH4.hilo.results <- as.data.frame(indic.9.CH4.hilo$sign)
indic.9.CH4.hilo.results.2 <- subset(indic.9.CH4.hilo.results, p.value<0.05&!is.na(p.value))
indic.9.CH4.hilo.results.2$OTU_ID2 <- row.names(indic.9.CH4.hilo.results.2)

indic.9.CH4.hilo.results.2.tax <- merge(indic.9.CH4.hilo.results.2,tax,by="OTU_ID2")
write.csv(indic.9.CH4.hilo.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.CH4.hilo.results.2.tax.csv")

summary(indic.9.CH4.hilo, indvalcomp=TRUE)

# DO
  # histograms
hist(congo_syn_meta.out1.9.indic$do_mg_l, main="Histogram: DO_mg_l sample distribution", xlab="DO (mg/L)", col="blue", las=1, breaks=30)
hist(congo_syn_meta.out1.9.indic$log10_do_mg_l, main="Histogram: log10_DO_mg_l sample distribution", xlab="log10_DO (mg/L)", col="blue", las=1, breaks=20)
  # hi-low
congo_syn_meta.out1.9.indic$do_mg_l_high_low <- ifelse(congo_syn_meta.out1.9.indic$do_mg_l <=4, "low",
                                      ifelse(congo_syn_meta.out1.9.indic$do_mg_l >=4, "high", "null"))
  # indicator
  # indicator
indic.9.DO.hilo <- multipatt(otu_congo_syn.9.ra.t, congo_syn_meta.out1.9.indic$do_mg_l_high_low, control=how(nperm=999))
indic.9.DO.hilo.results <- as.data.frame(indic.9.DO.hilo$sign)
indic.9.DO.hilo.results.2 <- subset(indic.9.DO.hilo.results, p.value<0.05&!is.na(p.value))
indic.9.DO.hilo.results.2$OTU_ID2 <- row.names(indic.9.DO.hilo.results.2)

indic.9.DO.hilo.results.2.tax <- merge(indic.9.DO.hilo.results.2,tax,by="OTU_ID2")
write.csv(indic.9.DO.hilo.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.9.DO.hilo.results.2.tax.csv")

summary(indic.9.DO.hilo, indvalcomp=TRUE)

# adapt this? : coa_coords.9.meta$qcut_2_wet_pc_uall <-gtools::quantcut(coa_coords.9.meta$wet_pc_uall, q=6, na.rm=TRUE)

## OLD CODE BELOW
  #  # CH4 indicator species on the wet.pc.u05 co-inertia
# standardize CH4 data (z-score)
CH4_all_congo_syn_meta.out1.9 <- as.data.frame(congo_syn_meta.out1.9$CH4_pct_sat)
k.CH4.scaled.1.9 <- scale(CH4_all_congo_syn_meta.out1.9)
# use factoextra to create elbow plot to find optimal number of kmeans clusters; use on CI site scores
library(factoextra)
set.seed(134)
pdf("/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/k.ebbow.coin_con_syn.out1.9.CH4.pdf", width=11, height=8)
fviz_nbclust(k.CH4.scaled.1.9, kmeans, method="wss")
dev.off() # on inspection, n=3 look good


```

#Set 9 - labdsv indicator spp
```{r}
# indicator spp using labdsv (alt R package, used in Kellog et al)
library(labdsv)
otu_con_syn.out1.9.labdsv <- meta_otu_congo_syn.out1.9[,c((otu_congo_syn_start-1):otu_congo_syn_end)] # n=18; same OTU table we've been using, but keep a column for sample names
rownames(otu_con_syn.out1.9.labdsv) <- otu_con_syn.out1.9.labdsv[,1] # assign first column as row names
otu_con_syn.out1.9.labdsv <- otu_con_syn.out1.9.labdsv[,-1] # remove column 1 containing row names; otu table matches otu table prev extracted from rarified phyloseq object above; n=18
# check rowsums
otu_con_syn.out1.9.labdsv$row.sums <- rowSums(otu_con_syn.out1.9.labdsv) # all add to 9384
otu_con_syn.out1.9.labdsv <- otu_con_syn.out1.9.labdsv[, -ncol(otu_con_syn.out1.9.labdsv)] # remove row sums

# metadata file from above: congo_syn_meta.out1.9
congo_syn_meta.out1.9.labdsv <- congo_syn_meta.out1.9
rownames(congo_syn_meta.out1.9.labdsv) <- congo_syn_meta.out1.9.labdsv[,1]
congo_syn_meta.out1.9.labdsv <- congo_syn_meta.out1.9.labdsv[,-1]

# assign groups
congo_syn_meta.out1.9.labdsv$wet_pc_uall_hilo <- ifelse(congo_syn_meta.out1.9.labdsv$wet_pc_uall <=58, "low",
                                             ifelse(congo_syn_meta.out1.9.labdsv$wet_pc_uall >=58, "high", "null"))

congo_syn_meta.out1.9.labdsv$CH4_high_low <- ifelse(congo_syn_meta.out1.9.labdsv$CH4_pct_sat <=100000, "low",
                                      ifelse(congo_syn_meta.out1.9.labdsv >=100000, "high", "null"))

congo_syn_meta.out1.9.labdsv$do_mg_l_high_low <- ifelse(congo_syn_meta.out1.9.labdsv$do_mg_l <=4, "low",
                                      ifelse(congo_syn_meta.out1.9.labdsv$do_mg_l >=4, "high", "null"))
# make numeric vectors
library(dplyr)
wet.9.groups <- recode(congo_syn_meta.out1.9.labdsv$wet_pc_uall_hilo, "low"=1, "high"=2)
ch4.9.groups <- recode(congo_syn_meta.out1.9.labdsv$CH4_high_low, "low"=1, "high"=2)
do.9.groups <- recode(congo_syn_meta.out1.9.labdsv$do_mg_l_high_low, "low"=1, "high"=2)

# checks
length(wet.9.groups) == nrow(otu_con_syn.out1.9.labdsv) # dimensions TRUE
length(ch4.9.groups) == nrow(otu_con_syn.out1.9.labdsv) # dimensions TRUE
length(do.9.groups) == nrow(otu_con_syn.out1.9.labdsv) # dimensions TRUE

zero_occurances.9 <- colSums(otu_con_syn.out1.9.labdsv) == 0
otu_con_syn.out1.9.labdsv <- otu_con_syn.out1.9.labdsv[, !zero_occurances.9] # from 3114 to 2211 columns - so lots of species that are absent from these samples; should have used microViz!  but this zero_occurrances approach works too

otu_con_syn.out1.9.labdsv$row.sums <- rowSums(otu_con_syn.out1.9.labdsv) # all add to 9384
otu_con_syn.out1.9.labdsv <- otu_con_syn.out1.9.labdsv[, -ncol(otu_con_syn.out1.9.labdsv)] # remove row sums

########
# wet indicator
########
indic.labdsv.9.wet<- indval(otu_con_syn.out1.9.labdsv, wet.9.groups, numitr=999)

summary(indic.labdsv.9.wet, type="short") # 26 sig indicators; group 1(low) = 6 ASVs, group 2(high) = 20 ASVs

# save results
indic.labdsv.9.wet.res <- data.frame(
 # ASV = rownames(indic.labdsv.8.tt$indval),
  relfrq = indic.labdsv.9.wet$relfrq,
  relabu = indic.labdsv.9.wet$relabu,
  indval = indic.labdsv.9.wet$indval,
  pval = indic.labdsv.9.wet$pval
)
write.csv(indic.labdsv.9.wet.res, "output/indic.labdsv.9.wet.res.csv")

# subset dfs of OTUs that are in high vs low wetlands
# wetland-high
sel.ind.wet.hi <- which(congo_syn_meta.out1.9.labdsv$wet_pc_uall_hilo == "high") # 4, 14, 16, 17. 19
wet.otu.hi.sam.9 <- otu_con_syn.out1.9.labdsv[sel.ind.wet.hi, ] # 2211 ASVs
zero.occ.wet.hi.sam.9 <- colSums(wet.otu.hi.sam.9) == 0
wet.otu.hi.sam.9 <- wet.otu.hi.sam.9[, !zero.occ.wet.hi.sam.9] # from 2211 to 701 ASVs
# wetland-low
sel.ind.wet.low <- which(congo_syn_meta.out1.9.labdsv$wet_pc_uall_hilo == "low") # 1  2  3  5  6  7  8  9 10 11 12 13 15
wet.otu.low.sam.9 <- otu_con_syn.out1.9.labdsv[sel.ind.wet.low, ] # 2211 ASVs
zero.occ.wet.otu.low.sam.9 <- colSums(wet.otu.low.sam.9) == 0
wet.otu.low.sam.9 <- wet.otu.low.sam.9[, !zero.occ.wet.otu.low.sam.9] # from 2211 to 1669 ASVs
# compare OTUs
library(dplyr)
wet.hi.otus <- colnames(wet.otu.hi.sam.9) # get headers for list of OTUs
wet.low.otus <- colnames(wet.otu.low.sam.9) # get headers for list of OTUs
wet.hi.unique <- setdiff(wet.hi.otus, wet.low.otus) # get unique OTUs in wet-high # 542
wet.low.unique <- setdiff(wet.low.otus, wet.hi.otus) # get unique OTUs in wet-low # 1510
wet.common <- intersect(wet.hi.otus, wet.low.otus) # 159

# assign tax + output
wet.hi.unique.df <- data.frame(OTU_ID2 = wet.hi.unique) # convert list to df
wet.hi.unique.df.tax <- merge(wet.hi.unique.df, tax, by="OTU_ID2")
write.csv(wet.hi.unique.df.tax, "output/wet.hi.unique.df.tax.csv")

wet.low.unique.df <- data.frame(OTU_ID2 = wet.low.unique)
wet.low.unique.df.tax <- merge(wet.low.unique.df, tax, by="OTU_ID2")
write.csv(wet.low.unique.df.tax, "output/wet.low.unique.df.tax.csv")

wet.common.df <- data.frame(OTU_ID2 = wet.common)
wet.common.df.tax <- merge(wet.common.df, tax, by="OTU_ID2")
write.csv(wet.common.df.tax, "output/wet.common.df.tax.csv")


## match sig ASVs with DNA sequence, create new fata file
# dna-sequences.fasta is a qiime2 export from ct_mek_con_merged_rep_seqs_bac_arc_20230908.qza
library(seqinr)
#fasta.file <-  "../../ct_qiime_100/dna-sequences.fasta"
#dna.seq <- read.fasta(file = fasta.file)
# otu list for set 8.tt
set.9.indic.wet.ASVlist.file <- "output/set.9.indic.wet.ASVlist.csv"
set.9.indic.wet.ASVlist <- readLines(set.9.indic.wet.ASVlist.file)
# Extract sequences for specific OTUs
set.9.indic.wet.ASV.seq <- lapply(set.9.indic.wet.ASVlist, function(otu) {
  if (otu %in% names(dna.seq)) {
    seq <- dna.seq[[otu]]
    return(toupper(seq))
  } else {
    return(NULL)
  }
})
# Remove NULL entries
set.9.indic.wet.ASV.seq <- set.9.indic.wet.ASV.seq[!sapply(set.9.indic.wet.ASV.seq, is.null)]
# write extracted set8.tt sequences into a new fasta file
set.9.wet.out.fasta <- "output/set.9.indic.wet.ASV.seq.fasta"
write.fasta(sequences=set.9.indic.wet.ASV.seq, names=set.9.indic.wet.ASVlist, file.out=set.9.wet.out.fasta)

#######
# CH4 indicator
#######
indic.labdsv.9.ch4<- indval(otu_con_syn.out1.9.labdsv, ch4.9.groups, numitr=999)

summary(indic.labdsv.9.ch4, type="short") # 50 sig indicators; group 1(low) = 3 ASVs, group 2(high) = 47 ASVs

# save results
indic.labdsv.9.ch4.res <- data.frame(
 # ASV = rownames(indic.labdsv.8.tt$indval),
  relfrq = indic.labdsv.9.ch4$relfrq,
  relabu = indic.labdsv.9.ch4$relabu,
  indval = indic.labdsv.9.ch4$indval,
  pval = indic.labdsv.9.ch4$pval
)
write.csv(indic.labdsv.9.ch4.res, "output/indic.labdsv.9.ch4.res.csv")

# subset dfs of OTUs that are in high vs low CH4
# CH4-high
sel.ind.ch4.hi <- which(congo_syn_meta.out1.9.labdsv$CH4_high_low == "high") # rows 8 10 12
ch4.otu.hi.sam.9 <- otu_con_syn.out1.9.labdsv[sel.ind.ch4.hi, ] # 2211 ASVs
zero.occ.ch4.otu.hi.sam.9 <- colSums(ch4.otu.hi.sam.9) == 0
ch4.otu.hi.sam.9 <- ch4.otu.hi.sam.9[, !zero.occ.ch4.otu.hi.sam.9] # 529 ASVs
# ch4-low
sel.ind.ch4.low <- which(congo_syn_meta.out1.9.labdsv$CH4_high_low == "low") # 1  2  3  4  5  6  7  9 11 13 14 15 16 17 18
ch4.otu.low.sam.9 <- otu_con_syn.out1.9.labdsv[sel.ind.ch4.low, ] # 2211 ASVs
zero.occ.ch4.otu.low.sam.9 <- colSums(ch4.otu.low.sam.9) == 0
ch4.otu.low.sam.9 <- ch4.otu.low.sam.9[, !zero.occ.ch4.otu.low.sam.9] # 1815 ASVS
# compare CH4 ASVs
ch4.hi.otus <- colnames(ch4.otu.hi.sam.9) # get headers for list of high OTUs
ch4.low.otus <- colnames(ch4.otu.low.sam.9) # get headers for list of low OTUs
ch4.hi.uniqe <- setdiff(ch4.hi.otus, ch4.low.otus) # get unique OTUs in CH4-high # 396
ch4.low.unique <- setdiff(ch4.low.otus, ch4.hi.otus) # get unique OTUs in CH4-low # 1682
ch4.common <- intersect(ch4.hi.otus, ch4.low.otus) # 133 OTUs

# assign tax + output
ch4.hi.uniqe.df <- data.frame(OTU_ID2 = ch4.hi.uniqe) # convert list to df
ch4.hi.uniqe.df.tax <- merge(ch4.hi.uniqe.df, tax, by="OTU_ID2")
write.csv(ch4.hi.uniqe.df.tax, "output/ch4.hi.uniqe.df.tax.csv")

ch4.low.unique.df <- data.frame(OTU_ID2 = ch4.low.unique)
ch4.low.unique.df.tax <- merge(ch4.low.unique.df, tax, by="OTU_ID2")
write.csv(ch4.low.unique.df.tax, "output/ch4.low.unique.df.tax.csv")

ch4.common.df <- data.frame(OTU_ID2 = ch4.common)
ch4.common.df.tax <- merge(ch4.common.df, tax, by="OTU_ID2")
write.csv(ch4.common.df.tax, "output/ch4.common.df.tax.csv")


library(seqinr)
#fasta.file <-  "../../ct_qiime_100/dna-sequences.fasta"
#dna.seq <- read.fasta(file = fasta.file)
# otu list for set 8.tt
set.9.indic.ch4.ASVlist.file <- "output/set.9.indic.ch4.ASVlist.csv"
set.9.indic.ch4.ASVlist <- readLines(set.9.indic.ch4.ASVlist.file)
# Extract sequences for specific OTUs
set.9.indic.ch4.ASV.seq <- lapply(set.9.indic.ch4.ASVlist, function(otu) {
  if (otu %in% names(dna.seq)) {
    seq <- dna.seq[[otu]]
    return(toupper(seq))
  } else {
    return(NULL)
  }
})
# Remove NULL entries
set.9.indic.ch4.ASV.seq <- set.9.indic.ch4.ASV.seq[!sapply(set.9.indic.ch4.ASV.seq, is.null)]
# write extracted set8.tt sequences into a new fasta file
set.9.ch4.out.fasta <- "output/set.9.indic.ch4.ASV.seq.fasta"
write.fasta(sequences=set.9.indic.ch4.ASV.seq, names=set.9.indic.ch4.ASVlist, file.out=set.9.ch4.out.fasta)

#######
# DO indicator at 4 mg/l cutoff
######
indic.labdsv.9.DO<- indval(otu_con_syn.out1.9.labdsv, do.9.groups, numitr=999)

summary(indic.labdsv.9.DO, type="short") # 14 sig indicators; group 1(low) = 3 ASVs, group 2(high) = 11 ASVs

# save results
indic.labdsv.9.DO.res <- data.frame(
 # ASV = rownames(indic.labdsv.8.tt$indval),
  relfrq = indic.labdsv.9.DO$relfrq,
  relabu = indic.labdsv.9.DO$relabu,
  indval = indic.labdsv.9.DO$indval,
  pval = indic.labdsv.9.DO$pval
)
write.csv(indic.labdsv.9.DO.res, "output/indic.labdsv.9.DO.res.csv")

library(seqinr)
#fasta.file <-  "../../ct_qiime_100/dna-sequences.fasta"
#dna.seq <- read.fasta(file = fasta.file)
# otu list for set 8.tt
set.9.indic.DO.ASVlist.file <- "output/set.9.indic.DO.ASVlist.csv"
set.9.indic.DO.ASVlist <- readLines(set.9.indic.DO.ASVlist.file)
# Extract sequences for specific OTUs
set.9.indic.DO.ASV.seq <- lapply(set.9.indic.DO.ASVlist, function(otu) {
  if (otu %in% names(dna.seq)) {
    seq <- dna.seq[[otu]]
    return(toupper(seq))
  } else {
    return(NULL)
  }
})
# Remove NULL entries
set.9.indic.DO.ASV.seq <- set.9.indic.DO.ASV.seq[!sapply(set.9.indic.DO.ASV.seq, is.null)]
# write extracted set8.tt sequences into a new fasta file
set.9.DO.out.fasta <- "output/set.9.indic.DO.ASV.seq.fasta"
write.fasta(sequences=set.9.indic.DO.ASV.seq, names=set.9.indic.DO.ASVlist, file.out=set.9.DO.out.fasta)

#######
# DO indicator at 2 mg/l cutoff
######
congo_syn_meta.out1.9.labdsv$do_mg_l_high_low.2 <- ifelse(congo_syn_meta.out1.9.labdsv$do_mg_l <2, "low",
                                      ifelse(congo_syn_meta.out1.9.labdsv$do_mg_l >=2, "high", "null"))

do.9.groups.2 <- recode(congo_syn_meta.out1.9.labdsv$do_mg_l_high_low.2, "low"=1, "high"=2)

length(do.9.groups.2) == nrow(otu_con_syn.out1.9.labdsv) # dimensions TRUE

indic.labdsv.9.2.DO<- indval(otu_con_syn.out1.9.labdsv, do.9.groups.2, numitr=999)

summary(indic.labdsv.9.2.DO, type="short") # 10 sig indicators; group 1(low) = 5 ASVs, group 2(high) = 5 ASVs

# save results
indic.labdsv.9.2.DO.res <- data.frame(
 # ASV = rownames(indic.labdsv.8.tt$indval),
  relfrq = indic.labdsv.9.2.DO$relfrq,
  relabu = indic.labdsv.9.2.DO$relabu,
  indval = indic.labdsv.9.2.DO$indval,
  pval = indic.labdsv.9.2.DO$pval
)
write.csv(indic.labdsv.9.2.DO.res, "output/indic.labdsv.9.2.DO.res.csv")

# subset dfs of OTUs that are in high vs low DO
# DO-high
sel.ind.DO.hi <- which(congo_syn_meta.out1.9.labdsv$do_mg_l_high_low.2 == "high") # rows 1  2  3  5  6  7  8  9 11 13
DO.otu.hi.sam.9 <- otu_con_syn.out1.9.labdsv[sel.ind.DO.hi, ] # 2211 ASVs
zero.occ.DO.otu.hi.sam.9 <- colSums(DO.otu.hi.sam.9) == 0
DO.otu.hi.sam.9 <- DO.otu.hi.sam.9[, !zero.occ.DO.otu.hi.sam.9] # 1277 ASVs
# DO-low
sel.ind.DO.low <- which(congo_syn_meta.out1.9.labdsv$do_mg_l_high_low.2 == "low") # 4 10 12 14 15 16 17 18
DO.otu.low.sam.9 <- otu_con_syn.out1.9.labdsv[sel.ind.DO.low, ] # 2211 ASVs
zero.occ.DO.otu.low.sam.9 <- colSums(DO.otu.low.sam.9) == 0
DO.otu.low.sam.9 <- DO.otu.low.sam.9[, !zero.occ.DO.otu.low.sam.9] # 1178
# compare DO ASVs
DO.hi.otus <- colnames(DO.otu.hi.sam.9) # headers for high OTUs
DO.low.otus <- colnames(DO.otu.low.sam.9) # headers for low OTUs
DO.hi.unique <- setdiff(DO.hi.otus, DO.low.otus) # unique OTUs in high-DO # 1033
DO.low.unique <- setdiff(DO.low.otus, DO.hi.otus) # unique OTUs in low-DO # 934
DO.common <- intersect(DO.hi.otus, DO.low.otus) # 244 common OTUs

# assign tax + output
DO.hi.unique.df <- data.frame(OTU_ID2 = DO.hi.unique) # convert list to df
DO.hi.unique.df.tax <- merge(DO.hi.unique.df, tax, by="OTU_ID2")
write.csv(DO.hi.unique.df.tax, "output/DO.hi.unique.df.tax.csv")

DO.low.unique.df <- data.frame(OTU_ID2 = DO.low.unique)
DO.low.unique.df.tax <- merge(DO.low.unique.df, tax, by="OTU_ID2")
write.csv(DO.low.unique.df.tax, "output/DO.low.unique.df.tax.csv")

DO.common.df <- data.frame(OTU_ID2 = DO.common)
DO.common.df.tax <- merge(DO.common.df, tax, by="OTU_ID2")
write.csv(DO.common.df.tax, "output/DO.common.df.tax.csv")



```



# set 9 - Differential abundance w/ ALDEx2 or DESeq2
```{r}
library(ALDEx2)

# ASV should be unrarified but filtered

#### test 1 - no rarify, no filter
# go back to original phyloseq object, before rarefying: community3 <-ps_filter(community, sample_id!="ME060025", s
# remove all samples with reads sample.size < 9383
sort(sample_sums(community3))
com_4_unrare <- prune_samples(sample_sums(community3) > 9383, community3)

# get congo synoptics
com_4_unrare <- ps_filter(com_4_unrare, project=="congo", sam_type == "synoptic") # n=23
sort(sample_sums(com_4_unrare))
# remove large rivers + outliers
com_9_unrare <- ps_filter(com_4_unrare, site_id.1!="Sangha",  site_id.1!="Oubangui", site_id.1!="Congo", site_id.1!="Lefini", site_id.1!="Mopanzi") # n=18, samples match set 9


# pull out OTU
otu_com_9 <- otu_table(com_9_unrare)
otu_com_9 <- as.data.frame(otu_com_9) # rows=taxa, columns=samples

# pull out groups
meta_com_9 <- data.frame(sample_data(com_9_unrare))
meta_com_9$sample_number <- row.names(meta_com_9)

meta_com_9$wet_pc_u04 <- as.numeric(meta_com_9$wet_pc_u04)
meta_com_9$wet_pc_u05 <- as.numeric(meta_com_9$wet_pc_u05)
meta_com_9$wet_pc_uall <- meta_com_9$wet_pc_u04 + meta_com_9$wet_pc_u05

########
# ALDEx2 - wet_pc_uall
########
# assign hi-lo for wet_pc_uall as in code above
meta_com_9$wet_pc_uall_hilo <- ifelse(meta_com_9$wet_pc_uall <=58, "low",
                                             ifelse(meta_com_9$wet_pc_uall >=58, "high", "null"))

cond_com_9_wet_pc_uall <- meta_com_9$wet_pc_uall_hilo

# make sure your conditions/group order in metadata matches those in the OTU table
ald.9.wet.pc.uall <- aldex(reads=otu_com_9, conditions = cond_com_9_wet_pc_uall, mc.samples = 128, test="t", effect=TRUE,
             	include.sample.summary = FALSE, verbose=T, denom="all", paired.test=FALSE)


aldex.plot(ald.9.wet.pc.uall, type="volcano", test="welch", xlab="Difference",
    ylab="-1(log10(q))", main='Volcano plot') 


# modular, step by step approach
ald.9.wet.pc.uall.clr <- aldex.clr(otu_com_9, cond_com_9_wet_pc_uall, mc.samples=128, verbose=TRUE, denom="all", gamma=1e-5)
# t-test - Welch's and Wilcoxon
ald.9.wet.pc.uall.tt <- aldex.ttest(ald.9.wet.pc.uall.clr, hist.plot=TRUE, paired.test=FALSE, verbose=TRUE)
# effect sizes for within and between groups
ald.9.wet.pc.uall.eff <- aldex.effect(ald.9.wet.pc.uall.clr, CI=TRUE, verbose=TRUE, include.sample.summary=TRUE, paired.test=TRUE)
# aldex plot
ald.9.wet.pc.uall.all <- data.frame(ald.9.wet.pc.uall.tt, ald.9.wet.pc.uall.eff)
# plot  -- all not significant
par(mfrow=c(1,3))
aldex.plot(ald.9.wet.pc.uall.all, type="MA", test="welch", main="MA plot")
aldex.plot(ald.9.wet.pc.uall.all, type="MW", test="welch", main="effect plot")
aldex.plot(ald.9.wet.pc.uall.all, type="volcano", test="welch", main="volcano plot")


# try filtering the samples
## PREVALENCE based filtering (ASV must appear in a minimum number of samples; prob does not make sense for a study like this, where rare taxa can be interesting
#Nearing et al. 2022 recommends filtering out ASVs found in fewer than 10% of samples; https://doi.org/10.1038/s41467-022-28034-z
  # from their script: https://github.com/nearinj/Comparison_of_DA_microbiome_methods/blob/main/Pipeline_scripts/Tool_scripts/Filter_samples_and_features.R


# 10%
remove_rare_features <- function( table , cutoff_pro) {
  if(cutoff_pro==0){
    message("No filtering will be done due to cutoff_pro set to 0")
    return(table)
  }
  row2keep <- c()
  cutoff <- ceiling( cutoff_pro * ncol(table) ) # otu_com_9 has 18 columns
  for ( i in 1:nrow(table) ) {
    row_nonzero <- length( which( table[ i , ]  > 0 ) )
    if ( row_nonzero > cutoff ) {
      row2keep <- c( row2keep , i)
    }
  }
  return( table [ row2keep , , drop=FALSE ])
}
# apply function to ASV table
cutoff_pro_value = 0.10
otu_com_9_filt.10 <- remove_rare_features(otu_com_9, cutoff_pro_value) # goes from 2298 ASVs to 203 ASVs!
# 10% filter ALDEx2
ald.9.10pct.wet.pc.uall <- aldex(reads=otu_com_9_filt.10, conditions = cond_com_9_wet_pc_uall, mc.samples = 1000, test="t", effect=TRUE,
             	include.sample.summary = FALSE, verbose=T, denom="all")
aldex.plot(ald.9.10pct.wet.pc.uall, type="volcano", test="welch", xlab="Difference",
    ylab="-1(log10(q))", main='Volcano plot')  # 2 samples DE; 10% too stringent?

# 5%
cutoff_pro_value = 0.05
otu_com_9_filt.05 <- remove_rare_features(otu_com_9, cutoff_pro_value) # goes from 2298 ASVs to 399 ASVs
# 5% filter ALDEx2
ald.9.05pct.wet.pc.uall <- aldex(reads=otu_com_9_filt.05, conditions = cond_com_9_wet_pc_uall, mc.samples = 1000, test="t", effect=TRUE,
             	include.sample.summary = FALSE, verbose=T, denom="all")
aldex.plot(ald.9.05pct.wet.pc.uall, type="volcano", test="welch", xlab="Difference",
# ALDeX
    ylab="-1(log10(q))", main='Volcano plot')  # 2 samples DE; 5% too stringent?

# 1%
cutoff_pro_value = 0.01
otu_com_9_filt.01 <- remove_rare_features(otu_com_9, cutoff_pro_value) # goes from 2298 ASVs to 399 ASVs
# 1% filter ALDEx2
ald.9.01pct.wet.pc.uall <- aldex(reads=otu_com_9_filt.01, conditions = cond_com_9_wet_pc_uall, mc.samples = 1000, test="t", effect=TRUE,
             	include.sample.summary = FALSE, verbose=T, denom="all")
aldex.plot(ald.9.01pct.wet.pc.uall, type="volcano", test="welch", xlab="Difference",
    ylab="-1(log10(q))", main='Volcano plot')  # 2 samples DE; 1% too stringent?

# 0.1%
cutoff_pro_value = 0.0000001
otu_com_9_filt.0.1 <- remove_rare_features(otu_com_9, cutoff_pro_value) # goes from 2298 ASVs to 399 ASVs
# 1% filter ALDEx2
ald.9.01pct.wet.pc.uall <- aldex(reads=otu_com_9_filt.01, conditions = cond_com_9_wet_pc_uall, mc.samples = 1000, test="t", effect=TRUE,
             	include.sample.summary = FALSE, verbose=T, denom="all")
aldex.plot(ald.9.01pct.wet.pc.uall, type="volcano", test="welch", xlab="Difference",
    ylab="-1(log10(q))", main='Volcano plot')  # 2 samples DE; 1% too stringent?


## FREQUENCY
# examine row sums
otu_com_9$rowsums <- rowSums(otu_com_9, na.rm=TRUE)
#row.sums.otu.com.9 <- rowSums(otu_com_9, na.rm=TRUE)
#row.sums.otu.com.9 <- as.data.frame(setNames(row.sums.otu.com.9, rownames(otu_com_9)))
#colnames(row.sums.otu.com.9)[1] <- "rowsum"
#summary(row.sums.otu.com.9$rowsum)

# filter out ASVs with freqs =< 2
library(dplyr)
freq.3 <- 3
freq.filt.3.otu.com.9 <- otu_com_9 %>%
  filter(rowsums >= freq.3) # removes all ASVs with 2 or fewer samples; goes from 2198 to 2102 samples
# remove rowsum column
freq.filt.3.otu.com.9 <- freq.filt.3.otu.com.9[, -ncol(freq.filt.3.otu.com.9)]
ald.9.3freq.wet.pc.uall <- aldex(reads=freq.filt.3.otu.com.9, conditions = cond_com_9_wet_pc_uall, mc.samples = 1000, test="t", effect=TRUE,
             	include.sample.summary = FALSE, verbose=T, denom="all", paired.test=FALSE)
aldex.plot(ald.9.3freq.wet.pc.uall, type="volcano", test="welch", xlab="Difference",
    ylab="-1(log10(q))", main='Volcano plot')  # 0 samples DE; too many samples dropped?

####################
# DESeq2 wet_pc_uall -- via phyloseq ---- https://www.bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html
####################
# add high-low TT variable in phyloseq object
sample_data(com_9_unrare) <- sample_data(meta_com_9)
# convert to deseq2 object
library(DESeq2)
dsq.com.9.wet.pc.uall <- phyloseq_to_deseq2(com_9_unrare, ~ wet_pc_uall_hilo)

###########
# 3 approaches for calc geometric means prior to estimating size factors
###########

## 1) standard estimateSizeFactrs (default) = geometric means calculated from the data
dsq.com.9.wet.def = estimateSizeFactors(dsq.com.9.wet.pc.uall) # no separate geometric mean is calculated first
dsq.com.9.wet.def = DESeq(dsq.com.9.wet.def, fitType="local")
# investigate results
res.dsq.com.9.wet.def <- results(dsq.com.9.wet.def) # results object
res.dsq.com.9.wet.def <- res.dsq.com.9.wet.def[order(res.dsq.com.9.wet.def$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.9.wet.def <- res.dsq.com.9.wet.def[(res.dsq.com.9.wet.def$padj < alpha), ] # only sig results
sigtab.dsq.com.9.wet.def <- cbind(as(sigtab.dsq.com.9.wet.def, "data.frame"), as(tax_table(com_9_unrare)[rownames(sigtab.dsq.com.9.wet.def), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.9.wet.def, "output/sigtab.dsq.com.9.wet.def.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.9.wet.def = sigtab.dsq.com.9.wet.def[sigtab.dsq.com.9.wet.def[, "log2FoldChange"] > 0, ]
posigtab.dsq.com.9.wet.def = posigtab.dsq.com.9.wet.def[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.9.wet.def.genus = subset(sigtab.dsq.com.9.wet.def, !is.na(Genus))
# Class order
x.9.wet.def = tapply(sigtab.dsq.com.9.wet.def.genus$log2FoldChange, sigtab.dsq.com.9.wet.def.genus$Class, function(x.9.wet.def) max(x.9.wet.def))
x.9.wet.def=sort(x.9.wet.def, TRUE)
sigtab.dsq.com.9.wet.def.genus$Class = factor(as.character(sigtab.dsq.com.9.wet.def.genus$Class), levels=names(x.9.wet.def))
# Genus order
x.9.wet.def = tapply(sigtab.dsq.com.9.wet.def.genus$log2FoldChange, sigtab.dsq.com.9.wet.def.genus$Genus, function(x.9.wet.def) max(x.9.wet.def))
x.9.wet.def=sort(x.9.wet.def, TRUE)
sigtab.dsq.com.9.wet.def.genus$Genus = factor(as.character(sigtab.dsq.com.9.wet.def.genus$Genus), levels=names(x.9.wet.def))
# plot with x=log2FC, y=genus, color=Class
ggplot(sigtab.dsq.com.9.wet.def.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5))

## 2) geoMeans = custom phyloseq approach developed by McMurdie, which deals with lots of zeroes in 16S data
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans.9.wet = apply(counts(dsq.com.9.wet.pc.uall), 1, gm_mean) # custom gm_mean function applied to all rows
dsq.com.9.wet.gm = estimateSizeFactors(dsq.com.9.wet.pc.uall, geoMeans = geoMeans.9.wet) # estimate size factors based on that computed gm_mean
dsq.com.9.wet.gm = DESeq(dsq.com.9.wet.gm, fitType="local")
# investigate results
res.dsq.com.9.wet.gm <- results(dsq.com.9.wet.gm) # results object
res.dsq.com.9.wet.gm <- res.dsq.com.9.wet.gm[order(res.dsq.com.9.wet.gm$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.9.wet.gm <- res.dsq.com.9.wet.gm[(res.dsq.com.9.wet.gm$padj < alpha), ] # only sig results
sigtab.dsq.com.9.wet.gm <- cbind(as(sigtab.dsq.com.9.wet.gm, "data.frame"), as(tax_table(com_9_unrare)[rownames(sigtab.dsq.com.9.wet.gm), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.9.wet.gm, "output/sigtab.dsq.com.9.wet.gm.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.9.wet.gm = sigtab.dsq.com.9.wet.gm[sigtab.dsq.com.9.wet.gm[, "log2FoldChange"] > 0, ] # only shows ASVs with positive FC
posigtab.dsq.com.9.wet.gm = posigtab.dsq.com.9.wet.gm[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.9.wet.gm.genus = subset(sigtab.dsq.com.9.wet.gm, !is.na(Genus))
# Class order
x.9.wet.gm = tapply(sigtab.dsq.com.9.wet.gm.genus$log2FoldChange, sigtab.dsq.com.9.wet.gm.genus$Class, function(x.9.wet.gm) max(x.9.wet.gm))
x.9.wet.gm=sort(x.9.wet.gm, TRUE)
sigtab.dsq.com.9.wet.gm.genus$Class = factor(as.character(sigtab.dsq.com.9.wet.gm.genus$Class), levels=names(x.9.wet.gm))
# Genus order
x.9.wet.gm = tapply(sigtab.dsq.com.9.wet.gm.genus$log2FoldChange, sigtab.dsq.com.9.wet.gm.genus$Genus, function(x.9.wet.gm) max(x.9.wet.gm))
x.9.wet.gm=sort(x.9.wet.gm, TRUE)
sigtab.dsq.com.9.wet.gm.genus$Genus = factor(as.character(sigtab.dsq.com.9.wet.gm.genus$Genus), levels=names(x.9.wet.gm))
# plot with x=log2FC, y=genus, color=Class
pdf("output/sigtab.dsq.com.9.wet.gm.genus.pdf")
ggplot(sigtab.dsq.com.9.wet.gm.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5)) +
  ggtitle("Wetlands high vs low - DESeq2 (sigtab.dsq.com.9.wet.gm.genus)")
dev.off()

## 3) poscounts = "deals with a gene with some zeros, by calculating a modified geometric mean by taking the n-th root of the product of the non-zero counts. This evolved out of use cases with Paul McMurdie’s phyloseq package for metagenomic (16S, sic) samples"
dsq.com.9.wet.ps = estimateSizeFactors(dsq.com.9.wet.pc.uall, type="poscounts")
dsq.com.9.wet.ps = DESeq(dsq.com.9.wet.ps, fitType="local")
# investigate results
res.dsq.com.9.wet.ps <- results(dsq.com.9.wet.ps) # results object
res.dsq.com.9.wet.ps <- res.dsq.com.9.wet.ps[order(res.dsq.com.9.wet.ps$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.9.wet.ps <- res.dsq.com.9.wet.ps[(res.dsq.com.9.wet.ps$padj < alpha), ] # only sig results
sigtab.dsq.com.9.wet.ps <- cbind(as(sigtab.dsq.com.9.wet.ps, "data.frame"), as(tax_table(com_9_unrare)[rownames(sigtab.dsq.com.9.wet.ps), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.9.wet.ps, "output/sigtab.dsq.com.9.wet.ps.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.9.wet.ps = sigtab.dsq.com.9.wet.ps[sigtab.dsq.com.9.wet.ps[, "log2FoldChange"] > 0, ] # only includes ASVs with positive FC
posigtab.dsq.com.9.wet.ps = posigtab.dsq.com.9.wet.ps[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.9.wet.ps.genus = subset(sigtab.dsq.com.9.wet.ps, !is.na(Genus))
# Class order
x.9.wet.ps = tapply(sigtab.dsq.com.9.wet.ps.genus$log2FoldChange, sigtab.dsq.com.9.wet.ps.genus$Class, function(x.9.wet.ps) max(x.9.wet.ps))
x.9.wet.ps=sort(x.9.wet.ps, TRUE)
sigtab.dsq.com.9.wet.ps.genus$Class = factor(as.character(sigtab.dsq.com.9.wet.ps.genus$Class), levels=names(x.9.wet.ps))
# Genus order
x.9.wet.ps = tapply(sigtab.dsq.com.9.wet.ps.genus$log2FoldChange, sigtab.dsq.com.9.wet.ps.genus$Genus, function(x.9.wet.ps) max(x.9.wet.ps))
x.9.wet.ps=sort(x.9.wet.ps, TRUE)
sigtab.dsq.com.9.wet.ps.genus$Genus = factor(as.character(sigtab.dsq.com.9.wet.ps.genus$Genus), levels=names(x.9.wet.ps))
# plot with x=log2FC, y=genus, color=Class
ggplot(sigtab.dsq.com.9.wet.ps.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5))

########
# ALDEx2 - CH4
########
# assign hi-lo for wet_pc_uall as in code above
meta_com_9$CH4_high_low <- ifelse(meta_com_9$CH4_pct_sat <=100000, "low",
                                             ifelse(meta_com_9$CH4_pct_sat >=100000, "high", "null"))

cond_com_9_CH4 <- meta_com_9$CH4_high_low

# modular, step by step approach
ald.9.CH4.clr <- aldex.clr(otu_com_9, cond_com_9_CH4, mc.samples=128, verbose=TRUE, denom="all", gamma=1e-3)
# t-test - Welch's and Wilcoxon
ald.9.CH4.tt <- aldex.ttest(ald.9.CH4.clr, hist.plot=TRUE, paired.test=FALSE, verbose=TRUE)
# effect sizes for within and between groups
ald.9.CH4.eff <- aldex.effect(ald.9.CH4.clr, CI=TRUE, verbose=TRUE, include.sample.summary=TRUE, paired.test=TRUE)
# aldex plot
ald.9.CH4.all <- data.frame(ald.9.CH4.tt, ald.9.CH4.eff)
# plot  -- all not significant
pdf("output/ald.9.CH4.all.pdf")
par(mfrow=c(1,3))
aldex.plot(ald.9.CH4.all, type="MA", test="welch", main="MA plot")
aldex.plot(ald.9.CH4.all, type="MW", test="welch", main="effect plot")
aldex.plot(ald.9.CH4.all, type="volcano", test="welch", main="volcano plot")
dev.off()

# save results
ald.9.CH4.all.results <- ald.9.CH4.all
ald.9.CH4.all.results$OTU_ID2 <- row.names(ald.9.CH4.all.results)
ald.9.CH4.all.results.tax <- merge (ald.9.CH4.all.results, tax, by="OTU_ID2")
write.csv(ald.9.CH4.all.results.tax, "output/ald.9.CH4.all.results.tax.csv")

####################
# DESeq2 CH4 -- via phyloseq ---- https://www.bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html
####################
# add high-low TT variable in phyloseq object
sample_data(com_9_unrare) <- sample_data(meta_com_9)
# convert to deseq2 object
library(DESeq2)
dsq.com.9.CH4 <- phyloseq_to_deseq2(com_9_unrare, ~ CH4_high_low)

###########
# 3 approaches for calc geometric means prior to estimating size factors
###########

## 1) standard estimateSizeFactrs (default) = geometric means calculated from the data
dsq.com.9.CH4.def = estimateSizeFactors(dsq.com.9.CH4) # no separate geometric mean is calculated first
dsq.com.9.CH4.def = DESeq(dsq.com.9.CH4.def, fitType="local")
# investigate results
res.dsq.com.9.CH4.def <- results(dsq.com.9.CH4.def) # results object
res.dsq.com.9.CH4.def <- res.dsq.com.9.CH4.def[order(res.dsq.com.9.CH4.def$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.9.CH4.def <- res.dsq.com.9.CH4.def[(res.dsq.com.9.CH4.def$padj < alpha), ] # only sig results
sigtab.dsq.com.9.CH4.def <- cbind(as(sigtab.dsq.com.9.CH4.def, "data.frame"), as(tax_table(com_9_unrare)[rownames(sigtab.dsq.com.9.CH4.def), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.9.CH4.def, "output/sigtab.dsq.com.9.CH4.def.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.9.CH4.def = sigtab.dsq.com.9.CH4.def[sigtab.dsq.com.9.CH4.def[, "log2FoldChange"] > 0, ]
posigtab.dsq.com.9.CH4.def = posigtab.dsq.com.9.CH4.def[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.9.CH4.def.genus = subset(sigtab.dsq.com.9.CH4.def, !is.na(Genus))
# Class order
x.9.CH4.def = tapply(sigtab.dsq.com.9.CH4.def.genus$log2FoldChange, sigtab.dsq.com.9.CH4.def.genus$Class, function(x.9.CH4.def) max(x.9.CH4.def))
x.9.CH4.def=sort(x.9.CH4.def, TRUE)
sigtab.dsq.com.9.CH4.def.genus$Class = factor(as.character(sigtab.dsq.com.9.CH4.def.genus$Class), levels=names(x.9.CH4.def))
# Genus order
x.9.CH4.def = tapply(sigtab.dsq.com.9.CH4.def.genus$log2FoldChange, sigtab.dsq.com.9.CH4.def.genus$Genus, function(x.9.CH4.def) max(x.9.CH4.def))
x.9.CH4.def=sort(x.9.CH4.def, TRUE)
sigtab.dsq.com.9.CH4.def.genus$Genus = factor(as.character(sigtab.dsq.com.9.CH4.def.genus$Genus), levels=names(x.9.CH4.def))
# plot with x=log2FC, y=genus, color=Class
ggplot(sigtab.dsq.com.9.CH4.def.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5))

## 2) geoMeans = custom phyloseq approach developed by McMurdie, which deals with lots of zeroes in 16S data
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans.9.CH4 = apply(counts(dsq.com.9.CH4), 1, gm_mean) # custom gm_mean function applied to all rows
dsq.com.9.CH4.gm = estimateSizeFactors(dsq.com.9.CH4, geoMeans = geoMeans.9.CH4) # estimate size factors based on that computed gm_mean
dsq.com.9.CH4.gm = DESeq(dsq.com.9.CH4.gm, fitType="local")
# investigate results
res.dsq.com.9.CH4.gm <- results(dsq.com.9.CH4.gm) # results object
res.dsq.com.9.CH4.gm <- res.dsq.com.9.CH4.gm[order(res.dsq.com.9.CH4.gm$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.9.CH4.gm <- res.dsq.com.9.CH4.gm[(res.dsq.com.9.CH4.gm$padj < alpha), ] # only sig results
sigtab.dsq.com.9.CH4.gm <- cbind(as(sigtab.dsq.com.9.CH4.gm, "data.frame"), as(tax_table(com_9_unrare)[rownames(sigtab.dsq.com.9.CH4.gm), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.9.CH4.gm, "output/sigtab.dsq.com.9.CH4.gm.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.9.CH4.gm = sigtab.dsq.com.9.CH4.gm[sigtab.dsq.com.9.CH4.gm[, "log2FoldChange"] > 0, ] # only shows ASVs with positive FC
posigtab.dsq.com.9.CH4.gm = posigtab.dsq.com.9.CH4.gm[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.9.CH4.gm.genus = subset(sigtab.dsq.com.9.CH4.gm, !is.na(Genus))
# Class order
x.9.CH4.gm = tapply(sigtab.dsq.com.9.CH4.gm.genus$log2FoldChange, sigtab.dsq.com.9.CH4.gm.genus$Class, function(x.9.CH4.gm) max(x.9.CH4.gm))
x.9.CH4.gm=sort(x.9.CH4.gm, TRUE)
sigtab.dsq.com.9.CH4.gm.genus$Class = factor(as.character(sigtab.dsq.com.9.CH4.gm.genus$Class), levels=names(x.9.CH4.gm))
# Genus order
x.9.CH4.gm = tapply(sigtab.dsq.com.9.CH4.gm.genus$log2FoldChange, sigtab.dsq.com.9.CH4.gm.genus$Genus, function(x.9.CH4.gm) max(x.9.CH4.gm))
x.9.CH4.gm=sort(x.9.CH4.gm, TRUE)
sigtab.dsq.com.9.CH4.gm.genus$Genus = factor(as.character(sigtab.dsq.com.9.CH4.gm.genus$Genus), levels=names(x.9.CH4.gm))
# plot with x=log2FC, y=genus, color=Class
pdf("output/sigtab.dsq.com.9.CH4.gm.genus.pdf")
ggplot(sigtab.dsq.com.9.CH4.gm.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5)) +
  ggtitle("CH4 high vs low set 9 - DESeq2 (sigtab.dsq.com.9.CH4.gm.genus)")

## 3) poscounts = "deals with a gene with some zeros, by calculating a modified geometric mean by taking the n-th root of the product of the non-zero counts. This evolved out of use cases with Paul McMurdie’s phyloseq package for metagenomic (16S, sic) samples"
dsq.com.9.CH4.ps = estimateSizeFactors(dsq.com.9.CH4, type="poscounts")
dsq.com.9.CH4.ps = DESeq(dsq.com.9.CH4.ps, fitType="local")
# investigate results
res.dsq.com.9.CH4.ps <- results(dsq.com.9.CH4.ps) # results object
res.dsq.com.9.CH4.ps <- res.dsq.com.9.CH4.ps[order(res.dsq.com.9.CH4.ps$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.9.CH4.ps <- res.dsq.com.9.CH4.ps[(res.dsq.com.9.CH4.ps$padj < alpha), ] # only sig results
sigtab.dsq.com.9.CH4.ps <- cbind(as(sigtab.dsq.com.9.CH4.ps, "data.frame"), as(tax_table(com_9_unrare)[rownames(sigtab.dsq.com.9.CH4.ps), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.9.CH4.ps, "output/sigtab.dsq.com.9.CH4.ps.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.9.CH4.ps = sigtab.dsq.com.9.CH4.ps[sigtab.dsq.com.9.CH4.ps[, "log2FoldChange"] > 0, ] # only includes ASVs with positive FC
posigtab.dsq.com.9.CH4.ps = posigtab.dsq.com.9.CH4.ps[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.9.CH4.ps.genus = subset(sigtab.dsq.com.9.CH4.ps, !is.na(Genus))
# Class order
x.9.CH4.ps = tapply(sigtab.dsq.com.9.CH4.ps.genus$log2FoldChange, sigtab.dsq.com.9.CH4.ps.genus$Class, function(x.9.CH4.ps) max(x.9.CH4.ps))
x.9.CH4.ps=sort(x.9.CH4.ps, TRUE)
sigtab.dsq.com.9.CH4.ps.genus$Class = factor(as.character(sigtab.dsq.com.9.CH4.ps.genus$Class), levels=names(x.9.CH4.ps))
# Genus order
x.9.CH4.ps = tapply(sigtab.dsq.com.9.CH4.ps.genus$log2FoldChange, sigtab.dsq.com.9.CH4.ps.genus$Genus, function(x.9.CH4.ps) max(x.9.CH4.ps))
x.9.CH4.ps=sort(x.9.CH4.ps, TRUE)
sigtab.dsq.com.9.CH4.ps.genus$Genus = factor(as.character(sigtab.dsq.com.9.CH4.ps.genus$Genus), levels=names(x.9.CH4.ps))
# plot with x=log2FC, y=genus, color=Class
ggplot(sigtab.dsq.com.9.CH4.ps.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5))



########
# DO
########
meta_com_9$do_mg_l_high_low <- ifelse(meta_com_9$do_mg_l <=4, "low",
                                      ifelse(meta_com_9$do_mg_l >=4, "high", "null"))

cond_com_9_DO <- meta_com_9$do_mg_l_high_low

# modular, step by step approach
ald.9.DO.clr <- aldex.clr(otu_com_9, cond_com_9_DO, mc.samples=128, verbose=TRUE, denom="all", gamma=1e-3)
# t-test - Welch's and Wilcoxon
ald.9.DO.tt <- aldex.ttest(ald.9.DO.clr, hist.plot=TRUE, paired.test=FALSE, verbose=TRUE)
# effect sizes for within and between groups
ald.9.DO.eff <- aldex.effect(ald.9.DO.clr, CI=TRUE, verbose=TRUE, include.sample.summary=TRUE, paired.test=TRUE)
# aldex plot
ald.9.DO.all <- data.frame(ald.9.DO.tt, ald.9.DO.eff)
# plot  -- all not significant
#pdf("output/ald.9.CH4.all.pdf")
par(mfrow=c(1,3))
aldex.plot(ald.9.DO.all, type="MA", test="welch", main="MA plot")
aldex.plot(ald.9.DO.all, type="MW", test="welch", main="effect plot")
aldex.plot(ald.9.DO.all, type="volcano", test="welch", main="volcano plot")
#dev.off()


####################
# DESeq2 - DO -- via phyloseq ---- https://www.bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html
####################
# add high-low TT variable in phyloseq object
sample_data(com_9_unrare) <- sample_data(meta_com_9)
# convert to deseq2 object
library(DESeq2)
dsq.com.9.DO <- phyloseq_to_deseq2(com_9_unrare, ~ do_mg_l_high_low)
###########
# 3 approaches for calc geometric means prior to estimating size factors
###########

## 1) standard estimateSizeFactrs (default) = geometric means calculated from the data
dsq.com.9.DO.def = estimateSizeFactors(dsq.com.9.DO) # no separate geometric mean is calculated first
dsq.com.9.DO.def = DESeq(dsq.com.9.DO.def, fitType="local")
# investigate results
res.dsq.com.9.DO.def <- results(dsq.com.9.DO.def) # results object
res.dsq.com.9.DO.def <- res.dsq.com.9.DO.def[order(res.dsq.com.9.DO.def$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.9.DO.def <- res.dsq.com.9.DO.def[(res.dsq.com.9.DO.def$padj < alpha), ] # only sig results
sigtab.dsq.com.9.DO.def <- cbind(as(sigtab.dsq.com.9.DO.def, "data.frame"), as(tax_table(com_9_unrare)[rownames(sigtab.dsq.com.9.DO.def), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.9.DO.def, "output/sigtab.dsq.com.9.DO.def.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.9.DO.def = sigtab.dsq.com.9.DO.def[sigtab.dsq.com.9.DO.def[, "log2FoldChange"] > 0, ]
posigtab.dsq.com.9.DO.def = posigtab.dsq.com.9.DO.def[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.9.DO.def.genus = subset(sigtab.dsq.com.9.DO.def, !is.na(Genus))
# Class order
x.9.DO.def = tapply(sigtab.dsq.com.9.DO.def.genus$log2FoldChange, sigtab.dsq.com.9.DO.def.genus$Class, function(x.9.DO.def) max(x.9.DO.def))
x.9.DO.def=sort(x.9.DO.def, TRUE)
sigtab.dsq.com.9.DO.def.genus$Class = factor(as.character(sigtab.dsq.com.9.DO.def.genus$Class), levels=names(x.9.DO.def))
# Genus order
x.9.DO.def = tapply(sigtab.dsq.com.9.DO.def.genus$log2FoldChange, sigtab.dsq.com.9.DO.def.genus$Genus, function(x.9.DO.def) max(x.9.DO.def))
x.9.DO.def=sort(x.9.DO.def, TRUE)
sigtab.dsq.com.9.DO.def.genus$Genus = factor(as.character(sigtab.dsq.com.9.DO.def.genus$Genus), levels=names(x.9.DO.def))
# plot with x=log2FC, y=genus, color=Class
ggplot(sigtab.dsq.com.9.DO.def.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5))

## 2) geoMeans = custom phyloseq approach developed by McMurdie, which deals with lots of zeroes in 16S data
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans.9.DO = apply(counts(dsq.com.9.DO), 1, gm_mean) # custom gm_mean function applied to all rows
dsq.com.9.DO.gm = estimateSizeFactors(dsq.com.9.DO, geoMeans = geoMeans.9.DO) # estimate size factors based on that computed gm_mean
dsq.com.9.DO.gm = DESeq(dsq.com.9.DO.gm, fitType="local")
# investigate results
res.dsq.com.9.DO.gm <- results(dsq.com.9.DO.gm) # results object
res.dsq.com.9.DO.gm <- res.dsq.com.9.DO.gm[order(res.dsq.com.9.DO.gm$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.9.DO.gm <- res.dsq.com.9.DO.gm[(res.dsq.com.9.DO.gm$padj < alpha), ] # only sig results
sigtab.dsq.com.9.DO.gm <- cbind(as(sigtab.dsq.com.9.DO.gm, "data.frame"), as(tax_table(com_9_unrare)[rownames(sigtab.dsq.com.9.DO.gm), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.9.DO.gm, "output/sigtab.dsq.com.9.DO.gm.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.9.DO.gm = sigtab.dsq.com.9.DO.gm[sigtab.dsq.com.9.DO.gm[, "log2FoldChange"] > 0, ] # only shows ASVs with positive FC
posigtab.dsq.com.9.DO.gm = posigtab.dsq.com.9.DO.gm[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.9.DO.gm.genus = subset(sigtab.dsq.com.9.DO.gm, !is.na(Genus))
# Class order
x.9.DO.gm = tapply(sigtab.dsq.com.9.DO.gm.genus$log2FoldChange, sigtab.dsq.com.9.DO.gm.genus$Class, function(x.9.DO.gm) max(x.9.DO.gm))
x.9.DO.gm=sort(x.9.DO.gm, TRUE)
sigtab.dsq.com.9.DO.gm.genus$Class = factor(as.character(sigtab.dsq.com.9.DO.gm.genus$Class), levels=names(x.9.DO.gm))
# Genus order
x.9.DO.gm = tapply(sigtab.dsq.com.9.DO.gm.genus$log2FoldChange, sigtab.dsq.com.9.DO.gm.genus$Genus, function(x.9.DO.gm) max(x.9.DO.gm))
x.9.DO.gm=sort(x.9.DO.gm, TRUE)
sigtab.dsq.com.9.DO.gm.genus$Genus = factor(as.character(sigtab.dsq.com.9.DO.gm.genus$Genus), levels=names(x.9.DO.gm))
# plot with x=log2FC, y=genus, color=Class
pdf("output/sigtab.dsq.com.9.DO.gm.genus.pdf")
ggplot(sigtab.dsq.com.9.DO.gm.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5)) +
  ggtitle("DO high vs low - set 9 (sigtab.dsq.com.9.DO.gm.genus)")
dev.off()

## 3) poscounts = "deals with a gene with some zeros, by calculating a modified geometric mean by taking the n-th root of the product of the non-zero counts. This evolved out of use cases with Paul McMurdie’s phyloseq package for metagenomic (16S, sic) samples"
dsq.com.9.DO.ps = estimateSizeFactors(dsq.com.9.DO, type="poscounts")
dsq.com.9.DO.ps = DESeq(dsq.com.9.DO.ps, fitType="local")
# investigate results
res.dsq.com.9.DO.ps <- results(dsq.com.9.DO.ps) # results object
res.dsq.com.9.DO.ps <- res.dsq.com.9.DO.ps[order(res.dsq.com.9.DO.ps$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.9.DO.ps <- res.dsq.com.9.DO.ps[(res.dsq.com.9.DO.ps$padj < alpha), ] # only sig results
sigtab.dsq.com.9.DO.ps <- cbind(as(sigtab.dsq.com.9.DO.ps, "data.frame"), as(tax_table(com_9_unrare)[rownames(sigtab.dsq.com.9.DO.ps), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.9.DO.ps, "output/sigtab.dsq.com.9.DO.ps.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.9.DO.ps = sigtab.dsq.com.9.DO.ps[sigtab.dsq.com.9.DO.ps[, "log2FoldChange"] > 0, ] # only includes ASVs with positive FC
posigtab.dsq.com.9.DO.ps = posigtab.dsq.com.9.DO.ps[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.9.DO.ps.genus = subset(sigtab.dsq.com.9.DO.ps, !is.na(Genus))
# Class order
x.9.DO.ps = tapply(sigtab.dsq.com.9.DO.ps.genus$log2FoldChange, sigtab.dsq.com.9.DO.ps.genus$Class, function(x.9.DO.ps) max(x.9.DO.ps))
x.9.DO.ps=sort(x.9.DO.ps, TRUE)
sigtab.dsq.com.9.DO.ps.genus$Class = factor(as.character(sigtab.dsq.com.9.DO.ps.genus$Class), levels=names(x.9.DO.ps))
# Genus order
x.9.DO.ps = tapply(sigtab.dsq.com.9.DO.ps.genus$log2FoldChange, sigtab.dsq.com.9.DO.ps.genus$Genus, function(x.9.DO.ps) max(x.9.DO.ps))
x.9.DO.ps=sort(x.9.DO.ps, TRUE)
sigtab.dsq.com.9.DO.ps.genus$Genus = factor(as.character(sigtab.dsq.com.9.DO.ps.genus$Genus), levels=names(x.9.DO.ps))
# plot with x=log2FC, y=genus, color=Class
ggplot(sigtab.dsq.com.9.DO.ps.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5))


```


# Set 9 - wet_pc_uall-based CI + indicator - using log10_DIST_UP_KM
```{r}
congo_syn_meta.out1.9.wet.pc.uall.dd <- congo_syn_meta.out1.9[,c("log10_DIST_UP_KM", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_uall")]
# PCA
env_pca_congo_syn.out1.9.wet.pc.uall.dd <- dudi.pca(d=congo_syn_meta.out1.9.wet.pc.uall.dd, row.w=otu_coa_congo_syn.out1.9$lw, scale = TRUE, scannf = FALSE, nf = 10)

# co-inertia
coin_con_syn.out1.9.wet.pc.uall.dd <- coinertia(env_pca_congo_syn.out1.9.wet.pc.uall.dd, otu_coa_congo_syn.out1.9, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.9.wet.pc.uall.dd <- as.data.frame(coin_con_syn.out1.9.wet.pc.uall.dd$co)
enviro_loadings_coin.out1.9.wet.pc.uall.dd$variable <- row.names(enviro_loadings_coin.out1.9.wet.pc.uall.dd)

site_scores_coin_con_syn.out1.9.wet.pc.uall.dd_0 <- as.data.frame(coin_con_syn.out1.9.wet.pc.uall.dd$mX)
site_scores_coin_con_syn.out1.9.wet.pc.uall.dd <- bind_cols(site_scores_coin_con_syn.out1.9.wet.pc.uall.dd_0, coa_coords.9.meta)

# wet_pc_uall continuous color
pdf("output/coin.con.syn.out1.9.wet.pc.uall.dd.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.uall.dd,aes(NorS1,NorS2,label=river, color=wet_pc_uall), size=4, fontface="bold")+
  scale_color_continuous(low="lightgreen", high="blue")+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.uall.dd, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.uall.dd, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.5% Projected Inertia")+ # 0.6614409*35.52 = 23.49438
  ylab("15.5% Projected Inertia")+ # 0.6614409*23.43= 15.49756
  ggtitle("Congo Synoptic coin.con.syn.out1.9.wet.pc.uall.dd, RV=0.661")
dev.off()

wet_hilo_color <- c("low"="lightgreen", "high"="blue")
# wet_pc_uall hi-lo color
pdf("output/coin.con.syn.out1.9.wet.pc.uall.dd.hilo.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.9.wet.pc.uall.dd,aes(NorS1,NorS2,label=river, color=wet_pc_uall_hilo), size=4, fontface="bold")+
  scale_color_manual(values=wet_hilo_color)+
  theme_classic()+
  scale_y_reverse()+
  scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.9.wet.pc.uall.dd, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.9.wet.pc.uall.dd, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("23.5% Projected Inertia")+ # 0.6614409*35.52 = 23.49438
  ylab("15.5% Projected Inertia")+ # 0.6614409*23.43= 15.49756
  ggtitle("Congo Synoptic coin.con.syn.out1.9.wet.pc.uall.dd.hilo, RV=0.661")
dev.off()

```


# Set 9 - DOC clustering - new
```{r}
# metadata
congo_syn_meta.out1.9.doc <- congo_syn_meta.out1.9[,c("Row.names", "river", "log10_tt", "temp", "pH", "log10_suva_254", "doc_mcc_l", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_uall")]

# coa_coords
coa_coords.9.meta.doc <- merge(congo_syn_meta.out1.9.doc, coa_coords.9, by="Row.names", all.x=TRUE) 

# optimal log10_DOC bins
# n=2
coa_coords.9.meta.doc$qcut_2_log10_DOC <-gtools::quantcut(coa_coords.9.meta.doc$log10_DOC, q=2, na.rm=TRUE)
uall_centroid_qcut_2.doc <- coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_2_log10_DOC")] %>% group_by(qcut_2_log10_DOC) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut2.log10doc <- merge(uall_centroid_qcut_2.doc,coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_2_log10_DOC")],by="qcut_2_log10_DOC",wc.y=TRUE)
coa_coords.9.qcut2.log10doc$distance_to_centroid<-sqrt((coa_coords.9.qcut2.log10doc$Axis1-coa_coords.9.qcut2.log10doc$Axis1_mean)^2+(coa_coords.9.qcut2.log10doc$Axis2-coa_coords.9.qcut2.log10doc$Axis2_mean)^2)
bin2_centroid_dist_log10doc<-mean(coa_coords.9.qcut2.log10doc$distance_to_centroid)

# n=3
coa_coords.9.meta.doc$qcut_3_log10_DOC <-gtools::quantcut(coa_coords.9.meta.doc$log10_DOC, q=3, na.rm=TRUE)
uall_centroid_qcut_3.doc <- coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_3_log10_DOC")] %>% group_by(qcut_3_log10_DOC) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut3.log10doc <- merge(uall_centroid_qcut_3.doc,coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_3_log10_DOC")],by="qcut_3_log10_DOC",wc.y=TRUE)
coa_coords.9.qcut3.log10doc$distance_to_centroid<-sqrt((coa_coords.9.qcut3.log10doc$Axis1-coa_coords.9.qcut3.log10doc$Axis1_mean)^2+(coa_coords.9.qcut3.log10doc$Axis2-coa_coords.9.qcut3.log10doc$Axis2_mean)^2)
bin3_centroid_dist_log10doc<-mean(coa_coords.9.qcut3.log10doc$distance_to_centroid)

#n=4
coa_coords.9.meta.doc$qcut_4_log10_DOC <-gtools::quantcut(coa_coords.9.meta.doc$log10_DOC, q=4, na.rm=TRUE)
uall_centroid_qcut_4.doc <- coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_4_log10_DOC")] %>% group_by(qcut_4_log10_DOC) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut4.log10doc <- merge(uall_centroid_qcut_4.doc,coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_4_log10_DOC")],by="qcut_4_log10_DOC",wc.y=TRUE)
coa_coords.9.qcut4.log10doc$distance_to_centroid<-sqrt((coa_coords.9.qcut4.log10doc$Axis1-coa_coords.9.qcut4.log10doc$Axis1_mean)^2+(coa_coords.9.qcut4.log10doc$Axis2-coa_coords.9.qcut4.log10doc$Axis2_mean)^2)
bin4_centroid_dist_log10doc<-mean(coa_coords.9.qcut4.log10doc$distance_to_centroid)

#n=5
coa_coords.9.meta.doc$qcut_5_log10_DOC <-gtools::quantcut(coa_coords.9.meta.doc$log10_DOC, q=5, na.rm=TRUE)
uall_centroid_qcut_5.doc <- coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_5_log10_DOC")] %>% group_by(qcut_5_log10_DOC) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut5.log10doc <- merge(uall_centroid_qcut_5.doc,coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_5_log10_DOC")],by="qcut_5_log10_DOC",wc.y=TRUE)
coa_coords.9.qcut5.log10doc$distance_to_centroid<-sqrt((coa_coords.9.qcut5.log10doc$Axis1-coa_coords.9.qcut5.log10doc$Axis1_mean)^2+(coa_coords.9.qcut5.log10doc$Axis2-coa_coords.9.qcut5.log10doc$Axis2_mean)^2)
bin5_centroid_dist_log10doc<-mean(coa_coords.9.qcut5.log10doc$distance_to_centroid)

#n=6
coa_coords.9.meta.doc$qcut_6_log10_DOC <-gtools::quantcut(coa_coords.9.meta.doc$log10_DOC, q=6, na.rm=TRUE)
uall_centroid_qcut_6.doc <- coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_6_log10_DOC")] %>% group_by(qcut_6_log10_DOC) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut6.log10doc <- merge(uall_centroid_qcut_6.doc,coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_6_log10_DOC")],by="qcut_6_log10_DOC",wc.y=TRUE)
coa_coords.9.qcut6.log10doc$distance_to_centroid<-sqrt((coa_coords.9.qcut6.log10doc$Axis1-coa_coords.9.qcut6.log10doc$Axis1_mean)^2+(coa_coords.9.qcut6.log10doc$Axis2-coa_coords.9.qcut6.log10doc$Axis2_mean)^2)
bin6_centroid_dist_log10doc<-mean(coa_coords.9.qcut6.log10doc$distance_to_centroid)

#n=7
coa_coords.9.meta.doc$qcut_7_log10_DOC <-gtools::quantcut(coa_coords.9.meta.doc$log10_DOC, q=7, na.rm=TRUE)
uall_centroid_qcut_7.doc <- coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_7_log10_DOC")] %>% group_by(qcut_7_log10_DOC) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut7.log10doc <- merge(uall_centroid_qcut_7.doc,coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_7_log10_DOC")],by="qcut_7_log10_DOC",wc.y=TRUE)
coa_coords.9.qcut7.log10doc$distance_to_centroid<-sqrt((coa_coords.9.qcut7.log10doc$Axis1-coa_coords.9.qcut7.log10doc$Axis1_mean)^2+(coa_coords.9.qcut7.log10doc$Axis2-coa_coords.9.qcut7.log10doc$Axis2_mean)^2)
bin7_centroid_dist_log10doc<-mean(coa_coords.9.qcut7.log10doc$distance_to_centroid)

#n=8
coa_coords.9.meta.doc$qcut_8_log10_DOC <-gtools::quantcut(coa_coords.9.meta.doc$log10_DOC, q=8, na.rm=TRUE)
uall_centroid_qcut_8.doc <- coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_8_log10_DOC")] %>% group_by(qcut_8_log10_DOC) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut8.log10doc <- merge(uall_centroid_qcut_8.doc,coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_8_log10_DOC")],by="qcut_8_log10_DOC",wc.y=TRUE)
coa_coords.9.qcut8.log10doc$distance_to_centroid<-sqrt((coa_coords.9.qcut8.log10doc$Axis1-coa_coords.9.qcut8.log10doc$Axis1_mean)^2+(coa_coords.9.qcut8.log10doc$Axis2-coa_coords.9.qcut8.log10doc$Axis2_mean)^2)
bin8_centroid_dist_log10doc<-mean(coa_coords.9.qcut8.log10doc$distance_to_centroid)

#n=9
coa_coords.9.meta.doc$qcut_9_log10_DOC <-gtools::quantcut(coa_coords.9.meta.doc$log10_DOC, q=9, na.rm=TRUE)
uall_centroid_qcut_9.doc <- coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_9_log10_DOC")] %>% group_by(qcut_9_log10_DOC) %>% summarise(across(.cols = everything(), list(mean=mean, sd=sd, se=~sd(.)/sqrt(n()))))
coa_coords.9.qcut9.log10doc <- merge(uall_centroid_qcut_9.doc,coa_coords.9.meta.doc[,c("Axis1","Axis2","qcut_9_log10_DOC")],by="qcut_9_log10_DOC",wc.y=TRUE)
coa_coords.9.qcut9.log10doc$distance_to_centroid<-sqrt((coa_coords.9.qcut9.log10doc$Axis1-coa_coords.9.qcut9.log10doc$Axis1_mean)^2+(coa_coords.9.qcut9.log10doc$Axis2-coa_coords.9.qcut9.log10doc$Axis2_mean)^2)
bin9_centroid_dist_log10doc<-mean(coa_coords.9.qcut9.log10doc$distance_to_centroid)


# elbow plot of distance to group centroid
elbow_data_log10_DOC<-data.frame(bins=c(2,3,4,5,6,7,8,9),distances=c(bin2_centroid_dist_log10doc,bin3_centroid_dist_log10doc,bin4_centroid_dist_log10doc,bin5_centroid_dist_log10doc,bin6_centroid_dist_log10doc,bin7_centroid_dist_log10doc,bin8_centroid_dist_log10doc,bin9_centroid_dist_log10doc))

pdf("output/elbow_log10_DOC.pdf") # 5 bins looks best
#svglite("elbow_wet_pc_uall.svg")
ggplot(elbow_data_log10_DOC, aes(x=bins, y=distances))+
  geom_point(size=5, color="black")+
  theme_classic()+
  scale_x_continuous(breaks=seq(2, 9, by=1))+ #("2", "3", "4", "5", "6", "7", "8", "9"))+
  xlab("Number of bins")+
  ylab("Mean distance to centroid")+
  ggtitle("Eblow plot - log10_DOC mean distance to centroid")
dev.off()

# histogram of wet_pc_uall distribution across samples
hist(coa_coords.9.meta.doc$log10_DOC, main="Histogram: log10_DOC sample distribution", xlab="log10(DOC)", col="blue", las=1, breaks=20) # maybe ~ 3 bins?
# color using 2 groups of wet_pc_uall based on histogram: high and low
#coa_coords.9.meta$wet_pc_uall_hilo <- ifelse(coa_coords.9.meta$wet_pc_uall <=58, "low",
#                                             ifelse(coa_coords.9.meta$wet_pc_uall >=58, "high", "null"))


```



# Set 10 - synoptic small stream, high wetland sites removed
```{r}
# remove the "high" wet_pc_uall samples; should go from n=18 to n=13
meta_otu_congo_syn_out1.10 <- subset (meta_otu_congo_syn.out1.9, river!="Tanga" & river!="Eholola" & river!="Moundoungoma" & river!="Likoula Aux Herbes" & river!="Ngoko-South")

congo_syn_meta.out1.10 <- meta_otu_congo_syn_out1.10[,c(1:congo_syn_meta_col)]
otu_con_syn.out1.10 <- meta_otu_congo_syn_out1.10[,c(otu_congo_syn_start:otu_congo_syn_end)] # 3114 ASVs

# check zero occurrences
zero.otu_con_syn.out1.10 <- colSums(otu_con_syn.out1.10) == 0
otu_con_syn.out1.10.nozero <- otu_con_syn.out1.10[, !zero.otu_con_syn.out1.10] # from 3114 to 1669 ASVs

# COA
otu_coa_congo_syn.out1.10 <- dudi.coa(otu_con_syn.out1.10.nozero, scannf=FALSE, nf=8)

# subset for CI using dendritic distance
congo_syn_meta.out1.10.nowet <- congo_syn_meta.out1.10[,c("log10_DIST_UP_KM", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_uall")]
# PCA
env_pca_congo_syn_meta.out1.10.nowet <- dudi.pca(d=congo_syn_meta.out1.10.nowet, row.w=otu_coa_congo_syn.out1.10$lw, scale = TRUE, scannf = FALSE, nf = 10)
# CI
coin_con_syn.out1.10.nowet <- coinertia(env_pca_congo_syn_meta.out1.10.nowet, otu_coa_congo_syn.out1.10, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.10.nowet <- as.data.frame(coin_con_syn.out1.10.nowet$co)
enviro_loadings_coin.out1.10.nowet$variable <- row.names(enviro_loadings_coin.out1.10.nowet)

site_scores_coin_con_syn.out1.10.nowet_0 <- as.data.frame(coin_con_syn.out1.10.nowet$mX)
site_scores_coin_con_syn.out1.10.nowet <- bind_cols(site_scores_coin_con_syn.out1.10.nowet_0, congo_syn_meta.out1.10)

# color by wet_pc_uall
pdf("output/coin.con.syn.out1.10.nowet.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.10.nowet,aes(NorS1,NorS2,label=river, color=wet_pc_uall), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  #scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.10.nowet, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.10.nowet, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("31.0% Projected Inertia")+ # 0.7874987*39.324 = 30.9676
  ylab("22.6% Projected Inertia")+ # 0.7874987*28.689= 22.59255
  ggtitle("Congo Synoptic Set 10 coin_con_syn.out1.10.nowet, RV=0.788")
dev.off()

# color by CH4 (highest cumulative vector loading)
pdf("output/coin.con.syn.out1.10.CH4.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.10.nowet,aes(NorS1,NorS2,label=river, color=log10_CH4_pct_sat), size=4, fontface="bold")+
  scale_color_continuous(low="yellow", high="magenta")+
  theme_classic()+
  scale_y_reverse()+
  #scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.10.nowet, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.10.nowet, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("31.0% Projected Inertia")+ # 0.7874987*39.324 = 30.9676
  ylab("22.6% Projected Inertia")+ # 0.7874987*28.689= 22.59255
  ggtitle("Congo Synoptic Set 10 coin_con_syn.out1.10.CH4, RV=0.788")
dev.off()

# subset for CI using TT
congo_syn_meta.out1.10.nowet.tt <- congo_syn_meta.out1.10[,c("log10_tt_nores", "temp", "pH", "log10_suva_254", "log10_DOC", "Sr","log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "log10_NOx", "log10_PO4", "wet_pc_uall")]
# PCA
env_pca_congo_syn_meta.out1.10.nowet.tt <- dudi.pca(d=congo_syn_meta.out1.10.nowet.tt, row.w=otu_coa_congo_syn.out1.10$lw, scale = TRUE, scannf = FALSE, nf = 10)
# CI
coin_con_syn.out1.10.nowet.tt <- coinertia(env_pca_congo_syn_meta.out1.10.nowet.tt, otu_coa_congo_syn.out1.10, scan = FALSE, nf = 5)

enviro_loadings_coin.out1.10.nowet.tt <- as.data.frame(coin_con_syn.out1.10.nowet.tt$co)
enviro_loadings_coin.out1.10.nowet.tt$variable <- row.names(enviro_loadings_coin.out1.10.nowet.tt)

site_scores_coin_con_syn.out1.10.nowet.tt_0 <- as.data.frame(coin_con_syn.out1.10.nowet.tt$mX)
site_scores_coin_con_syn.out1.10.nowet.tt <- bind_cols(site_scores_coin_con_syn.out1.10.nowet.tt_0, congo_syn_meta.out1.10)

# color by wet_pc_uall
pdf("output/coin.con.syn.out1.10.nowet.tt.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.10.nowet.tt,aes(NorS1,NorS2,label=river, color=wet_pc_uall), size=4, fontface="bold")+
  scale_color_continuous(low="blue", high="green")+
  theme_classic()+
  scale_y_reverse()+
  #scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.10.nowet.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.10.nowet.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("31.2% Projected Inertia")+ # 0.7841689*39.775 = 31.19032
  ylab("19.0% Projected Inertia")+ # 0.7841689*24.195= 18.97297
  ggtitle("Congo Synoptic Set 10 coin_con_syn.out1.10.nowet.tt, RV=0.784")
dev.off()

# color by CH4 (highest cumulative vector loading?)
pdf("output/coin.con.syn.out1.10.nowet.CH4.pdf", width=9, height=8)
ggplot()+
  geom_text(data=site_scores_coin_con_syn.out1.10.nowet.tt,aes(NorS1,NorS2,label=river, color=log10_CH4_pct_sat), size=5)+
  scale_color_continuous(low="yellow", high="magenta")+
  theme_classic()+
  scale_y_reverse()+
  #scale_x_reverse()+
  geom_segment(data=enviro_loadings_coin.out1.10.nowet.tt, aes(x=0, y=0, xend=Comp1*3, yend=Comp2*3), arrow=arrow(length=unit(0.5,"cm")), color="black", linewidth=0.75)+
  geom_text_repel(data=enviro_loadings_coin.out1.10.nowet.tt, aes(Comp1*3, Comp2*3, label=variable), force=2, min.segment.length=Inf, color="black", size=5)+
  xlab("31.2% Projected Inertia")+ # 0.7841689*39.775 = 31.19032
  ylab("19.0% Projected Inertia")+ # 0.7841689*24.195= 18.97297
  ggtitle("Congo Synoptic Set 10 coin_con_syn.out1.10.nowet.tt, RV=0.784")
dev.off()

# indicator species - set 10
congo_syn_meta.out1.10.indic <- congo_syn_meta.out1.10
  # histograms
hist(congo_syn_meta.out1.10.indic$log10_CH4_pct_sat, main="Histogram: log10_CH4 sample distribution", xlab="log10(CH4)", col="blue", las=1, breaks=20)
hist(congo_syn_meta.out1.10.indic$CH4_pct_sat, main="Histogram: CH4_pct_sat sample distribution", xlab="CH4_pct_sat", col="blue", las=1, breaks=20)
  # hi-low
congo_syn_meta.out1.10.indic$CH4_high_low <- ifelse(congo_syn_meta.out1.10.indic$CH4_pct_sat <=100000, "low",
                                      ifelse(congo_syn_meta.out1.10.indic >=100000, "high", "null"))
# must set OTU table to RA data
# use original phyloseq object to eliminate samples from n=23 to n=18
library(microViz) # 17, 31, 44, 22, 37, 36, 40, 008
congo_syn.10 <- ps_filter(congo_syn, sample_id!="CONGO_003", sample_id!="CONGO_019", sample_id!="CONGO_017", sample_id!="CONGO_031", sample_id!="CONGO_044", sample_id!="CONGO_022", sample_id!="CONGO_037", sample_id!="CONGO_036", sample_id!="CONGO_040", sample_id!="CONGO_008") # elim Lefini, Mopanzi (Djiri already eliminated), large river sites (Sangha, Oubangui, Congo), and wetland sites (Mundoungoma, Eholola, Tanga, Likoula Aux Herbes, Kgoko-South)  ; get n=13
# convert OTU data to relative abundance
library(phylosmith)
congo_syn.10.ra<-relative_abundance(congo_syn.10, sig_fig=6)
otu_congo_syn.10.ra <- otu_table(congo_syn.10.ra)
write.csv(otu_congo_syn.10.ra, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/otu_congo_syn.10.ra.csv")
otu_congo_syn.10.ra.t <- as.data.frame(t(otu_congo_syn.10.ra)) # transpose OTU table, so rows=samples, columns=OTUs. This is input format for indicator species
write.csv(otu_congo_syn.10.ra.t, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/otu_congo_syn.10.ra.t.csv")

  # indicator
indic.10.CH4.hilo <- multipatt(otu_congo_syn.10.ra.t, congo_syn_meta.out1.10.indic$CH4_high_low, control=how(nperm=999))
indic.10.CH4.hilo.results <- as.data.frame(indic.10.CH4.hilo$sign)
indic.10.CH4.hilo.results.2 <- subset(indic.10.CH4.hilo.results, p.value<0.05&!is.na(p.value))
indic.10.CH4.hilo.results.2$OTU_ID2 <- row.names(indic.10.CH4.hilo.results.2)

indic.10.CH4.hilo.results.2.tax <- merge(indic.10.CH4.hilo.results.2,tax,by="OTU_ID2")
write.csv(indic.10.CH4.hilo.results.2.tax, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/indic.10.CH4.hilo.results.2.tax.csv")

summary(indic.10.CH4.hilo, indvalcomp=TRUE) # only one indiactor spp; next biplot CH4 vs DO, see where it cuts. Maybe move Motaba to other group

#######
# ALDEx2
#######
com_4_unrare <- ps_filter(com_4_unrare, project=="congo", sam_type == "synoptic") # n=23
sort(sample_sums(com_4_unrare))
# remove large rivers + outliers
com_10_unrare <- ps_filter(com_4_unrare, site_id.1!="Sangha",  site_id.1!="Oubangui", site_id.1!="Congo", site_id.1!="Lefini", site_id.1!="Mopanzi", site_id.1!="Moundoungoma", site_id.1!="Eholola", site_id.1!="Tanga", site_id.1!="Likoula Aux Herbes", site_id.1!="Ngoko-South") # n=18, samples match set 9


# pull out OTU
otu_com_10 <- otu_table(com_10_unrare)
otu_com_10 <- as.data.frame(otu_com_10) # rows=taxa, columns=samples

# pull out groups
meta_com_10 <- data.frame(sample_data(com_10_unrare))
meta_com_10$sample_number <- row.names(meta_com_10)


# assign hi-lo for wet_pc_uall as in code above
meta_com_10$CH4_grp <- ifelse(meta_com_10$CH4_pct_sat <=100000, "low",
                                             ifelse(meta_com_10$CH4_pct_sat >=100000, "high", "null"))

cond_com_10_CH4_DO <- meta_com_10$CH4_high_low

# need absolute abundance
# modular, step by step approach
ald.10.CH4_DO.clr <- aldex.clr(otu_com_10, cond_com_10_CH4_DO, mc.samples=128, verbose=TRUE, denom="all", gamma=1e-3)
# t-test - Welch's and Wilcoxon
ald.10.CH4_DO.tt <- aldex.ttest(ald.10.CH4_DO.clr, hist.plot=TRUE, paired.test=FALSE, verbose=TRUE)
# effect sizes for within and between groups
ald.10.CH4_DO.eff <- aldex.effect(ald.10.CH4_DO.clr, CI=TRUE, verbose=TRUE, include.sample.summary=TRUE, paired.test=TRUE)
# aldex plot
ald.10.CH4_DO.all <- data.frame(ald.10.CH4_DO.tt, ald.10.CH4_DO.eff)
# plot  -- all not significant
pdf("output/ald.10.CH4.DO.all.pdf")
par(mfrow=c(1,3))
aldex.plot(ald.10.CH4_DO.all, type="MA", test="welch", main="MA plot")
aldex.plot(ald.10.CH4_DO.all, type="MW", test="welch", main="effect plot")
aldex.plot(ald.10.CH4_DO.all, type="volcano", test="welch", main="volcano plot")
dev.off()

# save results
ald.10.CH4.DO.all.results <- ald.10.CH4_DO.all
ald.10.CH4.DO.all.results$OTU_ID2 <- row.names(ald.10.CH4.DO.all.results)
ald.10.CH4.DO.all.results.tax <- merge (ald.10.CH4.DO.all.results, tax, by="OTU_ID2")
write.csv(ald.10.CH4.DO.all.results.tax, "output/ald.10.CH4.DO.all.results.tax.csv")

########
# DESeq2 via phyloseq ---- https://www.bioconductor.org/packages/devel/bioc/vignettes/phyloseq/inst/doc/phyloseq-mixture-models.html
#######
# add high-low CH4 variable in phyloseq object
meta_com_10$CH4_grp <- as.factor(meta_com_10$CH4_grp)
sample_data(com_10_unrare) <- sample_data(meta_com_10)
# convert to deseq2 object
library(DESeq2)
dsq.com.10.CH4 <- phyloseq_to_deseq2(com_10_unrare, ~ CH4_high_low)
dsq.com.10.CH4$CH4_high_low <- factor(dsq.com.10.CH4$CH4_high_low, levels = c("low", "high")) # manually set low=reference, high=treatment
###########
# 3 approaches for calc geometric means prior to estimating size factors
###########

## 1) standard estimateSizeFactrs (default) = geometric means calculated from the data
dsq.com.10.CH4.def = estimateSizeFactors(dsq.com.10.CH4) # no separate geometric mean is calculated first
dsq.com.10.CH4.def = DESeq(dsq.com.10.CH4.def, fitType="local")
# investigate results
res.dsq.com.10.CH4.def <- results(dsq.com.10.CH4.def) # results object
res.dsq.com.10.CH4.def <- res.dsq.com.10.CH4.def[order(res.dsq.com.10.CH4.def$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.10.CH4.def <- res.dsq.com.10.CH4.def[(res.dsq.com.10.CH4.def$padj < alpha), ] # only sig results
sigtab.dsq.com.10.CH4.def <- cbind(as(sigtab.dsq.com.10.CH4.def, "data.frame"), as(tax_table(com_10_unrare)[rownames(sigtab.dsq.com.10.CH4.def), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.10.CH4.def, "output/sigtab.dsq.com.10.CH4.def.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.10.CH4.def = sigtab.dsq.com.10.CH4.def[sigtab.dsq.com.10.CH4.def[, "log2FoldChange"] > 0, ]
posigtab.dsq.com.10.CH4.def = posigtab.dsq.com.10.CH4.def[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.10.CH4.def.genus = subset(sigtab.dsq.com.10.CH4.def, !is.na(Genus))
# Class order
x.10.CH4.def = tapply(sigtab.dsq.com.10.CH4.def.genus$log2FoldChange, sigtab.dsq.com.10.CH4.def.genus$Class, function(x.10.CH4.def) max(x.10.CH4.def))
x.10.CH4.def=sort(x.10.CH4.def, TRUE)
sigtab.dsq.com.10.CH4.def.genus$Class = factor(as.character(sigtab.dsq.com.10.CH4.def.genus$Class), levels=names(x.10.CH4.def))
# Genus order
x.10.CH4.def = tapply(sigtab.dsq.com.10.CH4.def.genus$log2FoldChange, sigtab.dsq.com.10.CH4.def.genus$Genus, function(x.10.CH4.def) max(x.10.CH4.def))
x.10.CH4.def=sort(x.10.CH4.def, TRUE)
sigtab.dsq.com.10.CH4.def.genus$Genus = factor(as.character(sigtab.dsq.com.10.CH4.def.genus$Genus), levels=names(x.10.CH4.def))
# plot with x=log2FC, y=genus, color=Class
ggplot(sigtab.dsq.com.10.CH4.def.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -80, hjust = 0, vjust = 0.5))

## 2) geoMeans = custom phyloseq approach developed by McMurdie, which deals with lots of zeroes in 16S data
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans.10.CH4 = apply(counts(dsq.com.10.CH4), 1, gm_mean) # custom gm_mean function applied to all rows
dsq.com.10.CH4.gm = estimateSizeFactors(dsq.com.10.CH4, geoMeans = geoMeans.10.CH4) # estimate size factors based on that computed gm_mean
dsq.com.10.CH4.gm = DESeq(dsq.com.10.CH4.gm, fitType="local")
# investigate results
res.dsq.com.10.CH4.gm <- results(dsq.com.10.CH4.gm) # results object
res.dsq.com.10.CH4.gm <- res.dsq.com.10.CH4.gm[order(res.dsq.com.10.CH4.gm$padj, na.last=NA), ] # order by adjusted p-value

alpha = 0.05
sigtab.dsq.com.10.CH4.gm <- res.dsq.com.10.CH4.gm[(res.dsq.com.10.CH4.gm$padj < alpha), ] # only sig results
sigtab.dsq.com.10.CH4.gm <- cbind(as(sigtab.dsq.com.10.CH4.gm, "data.frame"), as(tax_table(com_10_unrare)[rownames(sigtab.dsq.com.10.CH4.gm), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.10.CH4.gm, "output/sigtab.dsq.com.10.CH4.gm.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.10.CH4.gm = sigtab.dsq.com.10.CH4.gm[sigtab.dsq.com.10.CH4.gm[, "log2FoldChange"] > 0, ] # only shows ASVs with positive FC
posigtab.dsq.com.10.CH4.gm = posigtab.dsq.com.10.CH4.gm[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.10.CH4.gm.genus = subset(sigtab.dsq.com.10.CH4.gm, !is.na(Genus))
# Class order
x.10.CH4.gm = tapply(sigtab.dsq.com.10.CH4.gm.genus$log2FoldChange, sigtab.dsq.com.10.CH4.gm.genus$Class, function(x.10.CH4.gm) max(x.10.CH4.gm))
x.10.CH4.gm=sort(x.10.CH4.gm, TRUE)
sigtab.dsq.com.10.CH4.gm.genus$Class = factor(as.character(sigtab.dsq.com.10.CH4.gm.genus$Class), levels=names(x.10.CH4.gm))
# Genus order
x.10.CH4.gm = tapply(sigtab.dsq.com.10.CH4.gm.genus$log2FoldChange, sigtab.dsq.com.10.CH4.gm.genus$Genus, function(x.10.CH4.gm) max(x.10.CH4.gm))
x.10.CH4.gm=sort(x.10.CH4.gm, TRUE)
sigtab.dsq.com.10.CH4.gm.genus$Genus = factor(as.character(sigtab.dsq.com.10.CH4.gm.genus$Genus), levels=names(x.10.CH4.gm))
# plot with x=log2FC, y=genus, color=Class
pdf("output/sigtab.dsq.com.10.CH4.gm.genus.pdf")
ggplot(sigtab.dsq.com.10.CH4.gm.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5)) +
  ggtitle("CH4 high vs low - set 10 (sigtab.dsq.com.10.CH4.gm.genus)")
dev.off()

## 3) poscounts = "deals with a gene with some zeros, by calculating a modified geometric mean by taking the n-th root of the product of the non-zero counts. This evolved out of use cases with Paul McMurdie’s phyloseq package for metagenomic (16S, sic) samples"
dsq.com.10.CH4.ps = estimateSizeFactors(dsq.com.10.CH4, type="poscounts")
dsq.com.10.CH4.ps = DESeq(dsq.com.10.CH4.ps, fitType="local")
# investigate results
res.dsq.com.10.CH4.ps <- results(dsq.com.10.CH4.ps) # results object
res.dsq.com.10.CH4.ps <- res.dsq.com.10.CH4.ps[order(res.dsq.com.10.CH4.ps$padj, na.last=NA), ] # order by adjusted p-value
alpha = 0.05
sigtab.dsq.com.10.CH4.ps <- res.dsq.com.10.CH4.ps[(res.dsq.com.10.CH4.ps$padj < alpha), ] # only sig results
sigtab.dsq.com.10.CH4.ps <- cbind(as(sigtab.dsq.com.10.CH4.ps, "data.frame"), as(tax_table(com_10_unrare)[rownames(sigtab.dsq.com.10.CH4.ps), ], "matrix")) # table of sig enriched ASVs
write.csv(sigtab.dsq.com.10.CH4.ps, "output/sigtab.dsq.com.10.CH4.ps.csv")
# examine OTUs that were significantly enriched in high CH4
# first clean table, just keep our columns of interest
posigtab.dsq.com.10.CH4.ps = sigtab.dsq.com.10.CH4.ps[sigtab.dsq.com.10.CH4.ps[, "log2FoldChange"] > 0, ] # only includes ASVs with positive FC
posigtab.dsq.com.10.CH4.ps = posigtab.dsq.com.10.CH4.ps[, c("baseMean", "log2FoldChange", "lfcSE", "padj", "Phylum", "Class", "Family", "Genus")] 
# plot results - Class + Genus
library("ggplot2")
theme_set(theme_bw())
sigtab.dsq.com.10.CH4.ps.genus = subset(sigtab.dsq.com.10.CH4.ps, !is.na(Genus))
# Class order
x.10.CH4.ps = tapply(sigtab.dsq.com.10.CH4.ps.genus$log2FoldChange, sigtab.dsq.com.10.CH4.ps.genus$Class, function(x.10.CH4.ps) max(x.10.CH4.ps))
x.10.CH4.ps=sort(x.10.CH4.ps, TRUE)
sigtab.dsq.com.10.CH4.ps.genus$Class = factor(as.character(sigtab.dsq.com.10.CH4.ps.genus$Class), levels=names(x.10.CH4.ps))
# Genus order
x.10.CH4.ps = tapply(sigtab.dsq.com.10.CH4.ps.genus$log2FoldChange, sigtab.dsq.com.10.CH4.ps.genus$Genus, function(x.10.CH4.ps) max(x.10.CH4.ps))
x.10.CH4.ps=sort(x.10.CH4.ps, TRUE)
sigtab.dsq.com.10.CH4.ps.genus$Genus = factor(as.character(sigtab.dsq.com.10.CH4.ps.genus$Genus), levels=names(x.10.CH4.ps))
# plot with x=log2FC, y=genus, color=Class
ggplot(sigtab.dsq.com.10.CH4.ps.genus, aes(y=Genus, x=log2FoldChange, color=Class)) +
  geom_vline(xintercept = 0.0, color = "gray", size=0.5) +
  geom_point(size=6) +
  theme(axis.text.x. = element_text(angle = -90, hjust = 0, vjust = 0.5))


#########################################################################################################################
# troubleshooting scratch / demoing DESeq2 for outliers, proper LFC (+/- directions), etc
## 2) geoMeans = custom phyloseq approach developed by McMurdie, which deals with lots of zeroes in 16S data
dsq.com.10.CH4 <- phyloseq_to_deseq2(com_10_unrare, ~ CH4_grp)# 1735 ASVs
#dsq.com.10.CH4$CH4_high_low <- factor(dsq.com.10.CH4$CH4_high_low, levels = c("low", "high")) # manually set low=reference, high=treatment
# pre-filter
dsq.com.10.CH4.keep1 <- rowSums(counts(dsq.com.10.CH4) >= 5) >=2  # 275 ASVs
dsq.com.10.CH4.kept1 <- dsq.com.10.CH4[dsq.com.10.CH4.keep1, ]
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans.10.CH4 = apply(counts(dsq.com.10.CH4), 1, gm_mean) # custom gm_mean function applied to all rows
dsq.com.10.CH4.gm.out = estimateSizeFactors(dsq.com.10.CH4, geoMeans = geoMeans.10.CH4) # estimate size factors based on that computed gm_mean
dsq.com.10.CH4.gm.out = DESeq(dsq.com.10.CH4.gm.out, fitType="local")

# check number of replicates per group (sanity check)
sam_grp.10.CH4.out <- colData(dsq.com.10.CH4.gm.out)
sam_grp.10.CH4.out.reps <- table(sam_grp.10.CH4.out$CH4_high_low)
print(sam_grp.10.CH4.out.reps) # shows high=3, low=10

# examine outliers?????????"

cooks.dsq.com.10.CH4.gm <- assays(dsq.com.10.CH4.gm)[["cooks"]]
par(mar=c(8,5,2,2))
boxplot(log10(assays(dsq.com.10.CH4.gm)[["cooks"]]), range=0, las=2)

replaced.dsq.com.10.CH4.gm <- assays(dsq.com.10.CH4)[["replaceCounts"]] # none replaced
# absolute value LFC + MA plot
res.dsq.com.10.CH4.gm.plot.cons <- results(dsq.com.10.CH4.gm.out, contrast = c("CH4_grp", "high", "low")) # low comes last, set as reference
res.dsq.com.10.CH4.gm.plot.cons <- res.dsq.com.10.CH4.gm.plot.cons[order(res.dsq.com.10.CH4.gm.plot.cons$padj, na.last=NA), ]

sigtab.dsq.com.10.CH4.gm.plot.cons <- res.dsq.com.10.CH4.gm.plot.cons[(res.dsq.com.10.CH4.gm.plot.cons$padj < alpha), ] # only sig; old good code
sigtab.dsq.com.10.CH4.gm.plot.cons <- cbind(as(sigtab.dsq.com.10.CH4.gm.plot.cons, "data.frame"), as(tax_table(com_10_unrare)[rownames(sigtab.dsq.com.10.CH4.gm.plot.cons), ], "matrix")) 
write.csv(sigtab.dsq.com.10.CH4.gm.plot.cons, "output/sigtab.dsq.com.10.CH4.gm.plot.cons.csv")

res.dsq.com.10.CH4.gm.plot.cons.df <- as.data.frame(res.dsq.com.10.CH4.gm.plot.cons)
res.dsq.com.10.CH4.gm.plot <- results(dsq.com.10.CH4.gm.out, lfcThreshold=.5, altHypothesis="greaterAbs")
res.dsq.com.10.CH4.gm.plot <- res.dsq.com.10.CH4.gm.plot[order(res.dsq.com.10.CH4.gm.plot$padj, na.last=NA), ] # order by adjusted p-value
res.dsq.com.10.CH4.gm.plot.df <- as.data.frame(res.dsq.com.10.CH4.gm.plot)


drawLines <- function() abline(h=c(-.5,.5),col="dodgerblue",lwd=2)
ylim <- c(-25,25)
pdf("output/res.dsq.com.10.CH4.gm.plot.pdf")
plotMA(res.dsq.com.10.CH4.gm.plot, ylim=ylim); drawLines()
dev.off()

sigtab.dsq.com.10.CH4.gm.plot <- res.dsq.com.10.CH4.gm.plot[(res.dsq.com.10.CH4.gm.plot$padj < alpha), ] # only sig; old good code
sigtab.dsq.com.10.CH4.gm.plot <- cbind(as(sigtab.dsq.com.10.CH4.gm.plot, "data.frame"), as(tax_table(com_10_unrare)[rownames(sigtab.dsq.com.10.CH4.gm.plot), ], "matrix")) 
write.csv(res.dsq.com.10.CH4.gm.plot, file="output/res.dsq.com.10.CH4.gm.plot.csv")

# lfcShrink
library(ashr)
library(apeglm)
resultsNames(dsq.com.10.CH4.gm.out)
res.dsq.com.10.CH4.gm.norm <- lfcShrink(dsq.com.10.CH4.gm.out, coef=2, type="normal")
res.dsq.com.10.CH4.gm.ape <- lfcShrink(dsq.com.10.CH4.gm.out, coef=2, type="apeglm")
res.dsq.com.10.CH4.gm.ashr <- lfcShrink(dsq.com.10.CH4.gm.out, coef=2, type="ashr")

par(mfrow=c(1,3), mar=c(4,4,2,1))
xlim <- c(1,1e5); ylim <- c(-3,3)
plotMA(res.dsq.com.10.CH4.gm.norm, xlim=xlim, ylim=ylim, main="normal")
plotMA(res.dsq.com.10.CH4.gm.ape, xlim=xlim, ylim=ylim, main="apeglm")
plotMA(res.dsq.com.10.CH4.gm.ashr, xlim=xlim, ylim=ylim, main="ashr")

res.dsq.com.10.CH4.gm.norm <- res.dsq.com.10.CH4.gm.norm[order(res.dsq.com.10.CH4.gm.norm$padj, na.last=NA), ] # order by adjusted p-value
res.dsq.com.10.CH4.gm.norm.df <- as.data.frame(res.dsq.com.10.CH4.gm.norm)

res.dsq.com.10.CH4.gm.ape <- res.dsq.com.10.CH4.gm.ape[order(res.dsq.com.10.CH4.gm.ape$padj, na.last=NA), ] # order by adjusted p-value
res.dsq.com.10.CH4.gm.ape.df <- as.data.frame(res.dsq.com.10.CH4.gm.ape)

res.dsq.com.10.CH4.gm.ashr <- res.dsq.com.10.CH4.gm.ashr[order(res.dsq.com.10.CH4.gm.ashr$padj, na.last=NA), ] # order by adjusted p-value
res.dsq.com.10.CH4.gm.ashr.df <- as.data.frame(res.dsq.com.10.CH4.gm.ashr)

```

# biplots/Spearman tests etc of environmental variables that co-vary
```{r}
# set 9 data (Small rivers)
plot(congo_syn_meta.out1.9$wet_pc_uall, congo_syn_meta.out1.9$doc_mcc_l, main="Wetland % vs. DOC concentration", xlab="wet_pc_uall (%)", ylab="DOC (mg/L)", pch=19)
abline(lm(congo_syn_meta.out1.9$wet_pc_uall ~ congo_syn_meta.out1.9$doc_mcc_l), col="red")

library(ggpubr)
F = as.formula("y~x")
# wet_pc_uall vs doc_mcc_l (linear)
pdf("output/biplot_wet_pc_uall_doc.9.pdf")
ggplot(congo_syn_meta.out1.9, aes(x=wet_pc_uall, y=doc_mcc_l))+
  geom_point()+
  geom_smooth(method=lm, se=FALSE)+
  stat_regline_equation(
    label.x=15, label.y=65,
  aes(label = paste(..eq.label.., ..adj.rr.label.., sep="~~~~")),
    formula=F, size=5,
  )+
  theme_classic()+
  ggtitle("Wetland coverage vs DOC conc.")
dev.off()

#pH and Sr
pdf("output/biplot_Sr_pH.9.pdf")
ggplot(congo_syn_meta.out1.9, aes(x=Sr, y=pH))+
  geom_point()+
  geom_smooth(method=lm, se=FALSE)+
  stat_regline_equation(
    label.x=0.7, label.y=5.5,
  aes(label = paste(..eq.label.., ..adj.rr.label.., sep="~~~~")),
    formula=F, size=5,
  )+
  theme_classic()+
    ggtitle("Sr vs pH")
dev.off()

# GHG
pdf("output/biplot_CH4_CO2.9.pdf")
ggplot(congo_syn_meta.out1.9, aes(x=log10_CH4_pct_sat, y=log10_CO2_pct_sat))+
  geom_point()+
  geom_smooth(method=lm, se=FALSE)+
  stat_regline_equation(
    label.x=4.75, label.y=9,
  aes(label = paste(..eq.label.., ..adj.rr.label.., sep="~~~~")),
    formula=F, size=5,
  )+
  theme_classic()+
  ggtitle("CH4 vs CO2 percent sat (log)")
dev.off()

# pH DOC antiplot
pdf("output/biplot_DOC_pH.9.pdf")
ggplot(congo_syn_meta.out1.9, aes(x=doc_mcc_l, y=pH))+
  geom_point()+
  geom_smooth(method=lm, se=FALSE)+
  stat_regline_equation(
    label.x=20, label.y=5.5,
  aes(label = paste(..eq.label.., ..adj.rr.label.., sep="~~~~")),
    formula=F, size=5,
  )+
  theme_classic()+
  ggtitle("DOC conc. vs pH")
dev.off()

# CO2 DO antiplot
pdf("output/biplot_DOC_DO.9.pdf")
ggplot(congo_syn_meta.out1.9, aes(x=doc_mcc_l, y=do_mg_l))+
  geom_point()+
  geom_smooth(method=lm, se=FALSE)+
  stat_regline_equation(
    label.x=20, label.y=6,
  aes(label = paste(..eq.label.., ..adj.rr.label.., sep="~~~~")),
    formula=F, size=5,
  )+
  theme_classic()+
  ggtitle("DOC vs. DO concs.")
dev.off()

# Byron's NASQAN correlation code
library(corrplot)
# This chunk reads environmental data into tables and calculates Spearman correlation matrices.  Note that "use = complete.obs" means that rows with NA in any of the cells are excluded for all pairwise comparisons.  If you switch to "use = pairwise.complete.obs" then rows are eliminated individually for each pairwise comparison.
# get variables
#BasicVars=c("q_daily_mean", "log10_q_daily_mean", "DIST_UP_KM", "log10_DIST_UP_KM", "UPLAND_SKM", "log10_UPLAND_SKM", "doc_mcc_l", "log10_DOC", "pH", "Nox", "log10_NOx", "TDN_mgN_l", "log10_TDN", "PO4", "log10_PO4", "Sr", "suva245_mgc_l_m", "log10_suva_254", "do_mg_l", "log10_do_mg_l", "CO2_pct_sat", "log10_CO2_pct_sat", "CH4_pct_sat", "log10_CH4_pct_sat", "temp", "tt_res_hr", "log10_tt_nores", "wet_pc_uall")
#designfileBasic = subset(congo_syn_meta.out1.9, select=BasicVars)

BasicVars.9_linear=c("tt_nores_hr","temp", "doc_mcc_l", "pH", "Nox", "PO4", "Sr", "suva245_mgc_l_m", "do_mg_l", "CO2_pct_sat", "CH4_pct_sat", "wet_pc_uall")
designfileBasic.9.linear = subset(congo_syn_meta.out1.9, select=BasicVars.9_linear)

BasicVars.9_log10=c("log10_tt_nores", "log10_DOC", "pH", "log10_NOx", "log10_PO4", "Sr",  "log10_suva_254", "log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "temp",  "wet_pc_uall")
designfileBasic.9.log10 = subset(congo_syn_meta.out1.9, select=BasicVars.9_log10)

designfileBasic.9.log10$log10_wet_pc_uall <- log10(1+designfileBasic.9.log10$wet_pc_uall)
designfileBasic.9.log10 <- subset(designfileBasic.9.log10, select = -wet_pc_uall)

# run Spearmann correlations
#M_Basic<-cor(designfileBasic, use="pairwise.complete.obs", method=c("spearman"))
M_Basic.9_linear<-cor(designfileBasic.9.linear, use="pairwise.complete.obs", method=c("spearman"))
M_Basic.9_log10<-cor(designfileBasic.9.log10, use="pairwise.complete.obs", method=c("spearman"))


# Square plots of Spearman correlation coefficients
corrplot(M_Basic, method="circle", tl.col="black", tl.cex=0.5)

pdf("output/spearmann_basic_lienar.pdf")
corrplot(M_Basic_linear, method="circle", tl.col="black", tl.cex=0.5)
dev.off()

pdf("output/spearman_basic_9_log10.pdf")
corrplot(M_Basic.9_log10, method="circle", tl.col="black", tl.cex=0.5)
dev.off()

# Function that calculates P values for spearman correlations and puts them in a matrix
cor.mtest <- function(mat, ...) {
    method = c("spearman")
    mat <- as.matrix(mat)
    n <- ncol(mat)
    p.mat<- matrix(NA, n, n)
    diag(p.mat) <- 0
    for (i in 1:(n - 1)) {
        for (j in (i + 1):n) {
            tmp <- cor.test(mat[, i], mat[, j], ...)
            p.mat[i, j] <- p.mat[j, i] <- tmp$p.value
        }
    }
  colnames(p.mat) <- rownames(p.mat) <- colnames(mat)
  p.mat
}

# matrix of the p-value of the correlation
p.mat_Basic_linear <- cor.mtest(M_Basic.9_linear)
p.mat_Basic_log10 <- cor.mtest(M_Basic.9_log10)
p.mat_Basic_log10 <- as.matrix(p.mat_Basic_log10)

# Combine correlation tables and p-values to create triangle matrices where insignificant relationships are left blank
pdf(file="output/corrplot_basic.9_linear.pdf")
corrplot(M_Basic.9_linear, type="upper", order="hclust", tl.col="black", tl.cex=0.85,
         p.mat = p.mat_Basic_linear, sig.level = 0.05, insig = "blank")
dev.off()

# do NOT use hclust
pdf(file="output/corrplot_basic.9_log10.pdf")
corrplot(M_Basic.9_log10, type="upper", tl.col="black", tl.cex=0.85,
         p.mat = p.mat_Basic_log10, sig.level = 0.05, insig = "blank")
dev.off()

#########
# set 8 rivers (all 21 Congo synoptic sites) AND using TT No Res
########
library(corrplot)
# This chunk reads environmental data into tables and calculates Spearman correlation matrices.  Note that "use = complete.obs" means that rows with NA in any of the cells are excluded for all pairwise comparisons.  If you switch to "use = pairwise.complete.obs" then rows are eliminated individually for each pairwise comparison.
# get variables
#BasicVars=c("q_daily_mean", "log10_q_daily_mean", "DIST_UP_KM", "log10_DIST_UP_KM", "UPLAND_SKM", "log10_UPLAND_SKM", "doc_mcc_l", "log10_DOC", "pH", "Nox", "log10_NOx", "TDN_mgN_l", "log10_TDN", "PO4", "log10_PO4", "Sr", "suva245_mgc_l_m", "log10_suva_254", "do_mg_l", "log10_do_mg_l", "CO2_pct_sat", "log10_CO2_pct_sat", "CH4_pct_sat", "log10_CH4_pct_sat", "temp", "tt_res_hr", "log10_tt", "wet_pc_uall")
#designfileBasic = subset(congo_syn_meta.out1.9, select=BasicVars)

BasicVars.8_linear=c("q_daily_mean", "DIST_UP_KM", "UPLAND_SKM", "tt_nores_hr","doc_mcc_l", "pH", "Nox", "PO4", "Sr", "suva245_mgc_l_m", "do_mg_l", "CO2_pct_sat", "CH4_pct_sat", "temp", "wet_pc_uall")
designfileBasic.8.linear = subset(congo_syn_meta.coa.out1, select=BasicVars.8_linear)

BasicVars.8_log10=c("log10_q_daily_mean",  "log10_DIST_UP_KM", "log10_UPLAND_SKM", "log10_tt_nores", "log10_DOC", "pH", "log10_NOx", "log10_PO4", "Sr",  "log10_suva_254", "log10_do_mg_l", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "temp",  "wet_pc_uall")
designfileBasic.8.log10 = subset(congo_syn_meta.out1.9, select=BasicVars.8_log10)
designfileBasic.8.log10$log10_wet_pc_uall <- log10(1+designfileBasic.8.log10$wet_pc_uall)
designfileBasic.8.log10 <- subset(designfileBasic.8.log10, select = -wet_pc_uall)



# run Spearmann correlations
#M_Basic<-cor(designfileBasic, use="pairwise.complete.obs", method=c("spearman"))
spear_ord_linear <- c("q_daily_mean", "DIST_UP_KM", "UPLAND_SKM", "tt_nores_hr", "pH", "Sr", "suva245_mgc_l_m", "doc_mcc_l", "do_mg_l", "Nox", "PO4","CO2_pct_sat", "CH4_pct_sat", "temp", "wet_pc_uall")
M_Basic.8_linear<-cor(designfileBasic.8.linear [ , spear_ord_linear], use="pairwise.complete.obs", method=c("spearman"))

#spear_ord_log10 <- c("log10_q_daily_mean", "log10_DIST_UP_KM", "log10_UPLAND_SKM", "log10_tt_nores", "pH", "Sr", "log10_suva_254", "log10_DOC", "log10_do_mg_l", "log10_NOx", "log10_PO4", "log10_CO2_pct_sat", "log10_CH4_pct_sat", "temp", "log10_wet_pc_uall")
#M_Basic.8_log10<-cor(designfileBasic.8.log10 [ , spear_ord_log10], use="pairwise.complete.obs", method=c("spearman"))
M_Basic.8_log10<-cor(designfileBasic.8.log10, use="pairwise.complete.obs", method=c("spearman"))

# Square plots of Spearman correlation coefficients
#corrplot(M_Basic, method="circle", tl.col="black", tl.cex=0.5)

#pdf("output/spearmann_basic_lienar.8.pdf")
#corrplot(M_Basic_linear.8, method="circle", tl.col="black", tl.cex=0.5)
#dev.off()

pdf("output/spearman_basic.8_log10.pdf")
corrplot(M_Basic.8_log10, method="circle", tl.col="black", tl.cex=0.5)
dev.off()

# Function that calculates P values for spearman correlations and puts them in a matrix
cor.mtest <- function(mat, ...) {
    method = c("spearman")
    mat <- as.matrix(mat)
    n <- ncol(mat)
    p.mat<- matrix(NA, n, n)
    diag(p.mat) <- 0
    for (i in 1:(n - 1)) {
        for (j in (i + 1):n) {
            tmp <- cor.test(mat[, i], mat[, j], ...)
            p.mat[i, j] <- p.mat[j, i] <- tmp$p.value
        }
    }
  colnames(p.mat) <- rownames(p.mat) <- colnames(mat)
  p.mat
}

# matrix of the p-value of the correlation
p.mat_Basic.8_linear <- cor.mtest(M_Basic.8_linear)
p.mat_Basic.8_log10 <- cor.mtest(M_Basic.8_log10)

# Combine correlation tables and p-values to create triangle matrices where insignificant relationships are left blank
pdf(file="output/corrplot_basic_linear.8.pdf")
corrplot(M_Basic.8_linear, type="upper", tl.col="black", tl.cex=0.85,
         p.mat = p.mat_Basic.8_linear, sig.level = 0.05, insig = "blank")
dev.off()

pdf(file="output/corrplot_basic_log10.8.pdf")
corrplot(M_Basic.8_log10, type="upper", tl.col="black", tl.cex=0.85,
         p.mat = p.mat_Basic.8_log10, sig.level = 0.05, insig = "blank")
dev.off()

pdf(file="output/corrplot_basic_log10.8.hclust.pdf")
corrplot(M_Basic.8_log10, type="upper", order="hclust", tl.col="black", tl.cex=0.85,
         p.mat = p.mat_Basic.8_log10, sig.level = 0.05, insig = "blank")
dev.off()

# troubleshooting - difference between circle plot and upper plot
# try complete.observations
M_Basic.8_log10.com.obs<-cor(designfileBasic.8.log10, use="pairwise.complete.obs", method=c("spearman"))
M_Basic.8_log10.com.obs <- as.matrix(M_Basic.8_log10.com.obs)
# circle plot
pdf("output/spearman_basic.8_log10.com.obs.pdf")
corrplot(M_Basic.8_log10.com.obs, method="circle", order="hclust", addCoef.col = "black", tl.col="black", tl.cex=0.5)
dev.off()
# p-mat
p.mat_Basic.8_log10.com.obs <- cor.mtest(M_Basic.8_log10.com.obs)
p.mat_Basic.8_log10.com.obs <- as.matrix(p.mat_Basic.8_log10.com.obs)
# upper plot
pdf(file="output/corrplot_basic_log10.8.final.pdf")
corrplot(M_Basic.8_log10.com.obs, method="circle", type="upper", tl.col="black", tl.cex=0.85,
         p.mat = p.mat_Basic.8_log10.com.obs, sig.level = 0.05, insig = "blank")
dev.off()

corrplot(M_Basic.8_log10.com.obs, tl.cex=0.85, p.mat=p.mat_Basic.8_log10.com.obs, insig = "p-value", sig.level = -1)
```


# Relative abundance - congo synoptic
```{r}
# microbiome package: https://microbiome.github.io/tutorials/Composition.html
# tutorial: https://www.yanh.org/2021/01/01/microbiome-r/
library(tidyverse)
library(microbiome)

congo_syn.ra <- congo_syn %>%
  plot_composition(sample.sort="tt_res_hr") +
  scale_fill_manual("Phylum", values = default_colors("Phylum")[taxa(congo_syn)]) +
  scale_y_continuous(label = scales::percent)
print(congo_syn.ra)

# using phyloseq object
congo_syn.ra = transform_sample_counts(congo_syn, function(x) x/sum(x)*100)

# phylum
glom_phylum_congo_syn<-tax_glom(congo_syn.ra, taxrank='Phylum', NArm=FALSE)
melt_phylum_congo_syn <- psmelt(glom_phylum_congo_syn)

melt_phylum_congo_syn <- melt_phylum_congo_syn %>%
  group_by(ORD_STRA, Phylum) %>%
  mutate(median=median(Abundance))
# select group median > 1 and group rare taxa into < 1% grouping
keep_phy_congo_syn <- unique(melt_phylum_congo_syn$Phylum[melt_phylum_congo_syn$median>1])
melt_phylum_congo_syn$Phylum[!(melt_phylum_congo_syn$Phylum %in% keep_phy_congo_syn)] <- "<1%"
# to get same rows together
melt_phylum_congo_syn_sum <- melt_phylum_congo_syn %>%
  group_by(Sample, river, Phylum, tt_res_hr) %>%
  summarize(Abundance=sum(Abundance))
# make sure tt is numeric
melt_phylum_congo_syn_sum$tt_res_hr <- as.numeric(melt_phylum_congo_syn_sum$tt_res_hr)

write.csv(melt_phylum_congo_syn_sum, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/output/melt_phylum_congo_syn_sum.csv")

# plot
ggplot(melt_phylum_congo_syn_sum, aes(x=river, y=Abundance, fill=Phylum))+
  geom_bar(stat="identity", aes(fill=Phylum))+
  labs(x="", y="%")+
  #facet_wrap(~project, scales="free_x", nrow=1)+
  theme_classic()+
  theme(strip.background=element_blank(), axis.text.x.bottom=element_text(angle=-90))
  
# class
glom_class_congo_syn<-tax_glom(congo_syn.ra, taxrank='Class', NArm=FALSE)
melt_class_congo_syn <- psmelt(glom_class_congo_syn)

melt_class_congo_syn <- melt_class_congo_syn %>%
  group_by(tt_res_hr, Class) %>%
  mutate(median=median(Abundance))
# select group median > 1 and group rare taxa into < 1% grouping
keep_cla_congo_syn <- unique(melt_class_congo_syn$Class[melt_class_congo_syn$median>1])
melt_class_congo_syn$Class[!(melt_class_congo_syn$Class %in% keep_cla_congo_syn)] <- "<1%"
# to get same rows together
melt_class_congo_syn_sum <- melt_class_congo_syn %>%
  group_by(Sample, river, Class, tt_res_hr) %>%
  summarize(Abundance=sum(Abundance))
# make sure tt is numeric
melt_class_congo_syn_sum$tt_res_hr <- as.numeric(melt_class_congo_syn_sum$tt_res_hr)

write.csv(melt_class_congo_syn_sum, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/output/melt_class_congo_syn_sum.csv")

# order - to explore Order Gammaproteobacteria
glom_ord_congo_syn<-tax_glom(congo_syn.ra, taxrank='Order', NArm=FALSE)
melt_ord_congo_syn <- psmelt(glom_ord_congo_syn)

melt_ord_congo_syn <- melt_ord_congo_syn %>% # cannot use log transformed data from congo_syn_meta df, because this sub-pipeline relies on phyloseq objects (e.g. congo_syn). So re-fransform in Excel
  group_by(pH, doc_mcc_l, CH4_pct_sat, wet_pc_u05, wet_pc_c05, wet_pc_ug2, Order) %>%
  mutate(median=median(Abundance))
# select group median > 1 and group rare taxa into < 1% grouping
keep_ord_congo_syn <- unique(melt_ord_congo_syn$Order[melt_ord_congo_syn$median>1])
melt_ord_congo_syn$Order[!(melt_ord_congo_syn$Order %in% keep_ord_congo_syn)] <- "<1%"
# to get same rows together
melt_ord_congo_syn_sum <- melt_ord_congo_syn %>%
  group_by(Sample, river, Class, Order,  pH, doc_mcc_l, CH4_pct_sat, wet_pc_u05, wet_pc_c05, wet_pc_ug2, tt_res_hr) %>%
  summarize(Abundance=sum(Abundance))
# make sure these variables are numeric
melt_ord_congo_syn_sum$wet_pc_u05 <- as.numeric(melt_ord_congo_syn_sum$wet_pc_u05)
melt_ord_congo_syn_sum$wet_pc_c05 <- as.numeric(melt_ord_congo_syn_sum$wet_pc_c05)
melt_ord_congo_syn_sum$wet_pc_ug2 <- as.numeric(melt_ord_congo_syn_sum$wet_pc_ug2)
melt_ord_congo_syn_sum$tt_res_hr <- as.numeric(melt_ord_congo_syn_sum$tt_res_hr)

write.csv(melt_ord_congo_syn_sum, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/melt_ord_congo_syn_sum.csv")

# family - to explore Order Burkholderiales
glom_fam_congo_syn<-tax_glom(congo_syn.ra, taxrank='Family', NArm=FALSE)
melt_fam_congo_syn <- psmelt(glom_fam_congo_syn)

melt_fam_congo_syn <- melt_fam_congo_syn %>% # cannot use log transformed data from congo_syn_meta df, because this sub-pipeline relies on phyloseq objects (e.g. congo_syn). So re-fransform in Excel
  group_by(pH, doc_mcc_l, CH4_pct_sat, wet_pc_u05, wet_pc_c05, wet_pc_ug2, Family) %>%
  mutate(median=median(Abundance))
# select group median > 1 and group rare taxa into < 1% grouping
keep_fam_congo_syn <- unique(melt_fam_congo_syn$Family[melt_fam_congo_syn$median>1])
melt_fam_congo_syn$Family[!(melt_fam_congo_syn$Family %in% keep_fam_congo_syn)] <- "<1%"
# to get same rows together
melt_fam_congo_syn_sum <- melt_fam_congo_syn %>%
  group_by(Sample, river, Order, Family, pH, doc_mcc_l, CH4_pct_sat, wet_pc_u05, wet_pc_c05, wet_pc_ug2, tt_res_hr) %>%
  summarize(Abundance=sum(Abundance))
# make sure these variables are numeric
melt_fam_congo_syn_sum$wet_pc_u05 <- as.numeric(melt_fam_congo_syn_sum$wet_pc_u05)
melt_fam_congo_syn_sum$wet_pc_c05 <- as.numeric(melt_fam_congo_syn_sum$wet_pc_c05)
melt_fam_congo_syn_sum$wet_pc_ug2 <- as.numeric(melt_fam_congo_syn_sum$wet_pc_ug2)
melt_fam_congo_syn_sum$tt_res_hr <- as.numeric(melt_fam_congo_syn_sum$tt_res_hr)

write.csv(melt_fam_congo_syn_sum, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/melt_fam_congo_syn_sum.csv")

# Genus - to explore Family Comamonadaceae
glom_gen_congo_syn<-tax_glom(congo_syn.ra, taxrank='Genus', NArm=FALSE)
melt_gen_congo_syn <- psmelt(glom_gen_congo_syn)

melt_gen_congo_syn <- melt_gen_congo_syn %>% # cannot use log transformed data from congo_syn_meta df, because this sub-pipeline relies on phyloseq objects (e.g. congo_syn). So re-fransform in Excel
  group_by(pH, doc_mcc_l, CH4_pct_sat, wet_pc_u05, wet_pc_c05, wet_pc_ug2, Genus) %>%
  mutate(median=median(Abundance))
# select group median > 1 and group rare taxa into < 1% grouping
keep_gen_congo_syn <- unique(melt_gen_congo_syn$Genus[melt_gen_congo_syn$median>1])
melt_gen_congo_syn$Genus[!(melt_gen_congo_syn$Genus %in% keep_gen_congo_syn)] <- "<1%"
# to get same rows together
melt_gen_congo_syn_sum <- melt_gen_congo_syn %>%
  group_by(Sample, river, Family, Genus, pH, doc_mcc_l, CH4_pct_sat, wet_pc_u05, wet_pc_c05, wet_pc_ug2, tt_res_hr) %>%
  summarize(Abundance=sum(Abundance))
# make sure these variables are numeric
melt_gen_congo_syn_sum$wet_pc_u05 <- as.numeric(melt_gen_congo_syn_sum$wet_pc_u05)
melt_gen_congo_syn_sum$wet_pc_c05 <- as.numeric(melt_gen_congo_syn_sum$wet_pc_c05)
melt_gen_congo_syn_sum$wet_pc_ug2 <- as.numeric(melt_gen_congo_syn_sum$wet_pc_ug2)
melt_gen_congo_syn_sum$tt_res_hr <- as.numeric(melt_gen_congo_syn_sum$tt_res_hr)

write.csv(melt_gen_congo_syn_sum, "/Users/ted/Library/Mobile Documents/com~apple~CloudDocs/Documents/CEOAS/MB668/Congo/mekong/R/100_ASVs/output/melt_gen_congo_syn_sum.csv")


# basic psmelt with phyloseq object congo_syn.9
# transform metadata variables
sample_data(congo_syn.9)$wet_pc_u04 <- as.numeric(sample_data(congo_syn.9)$wet_pc_u04)
sample_data(congo_syn.9)$wet_pc_u05 <- as.numeric(sample_data(congo_syn.9)$wet_pc_u05)
sample_data(congo_syn.9)$CH4_pct_sat <- as.numeric(sample_data(congo_syn.9)$CH4_pct_sat)
sample_data(congo_syn.9)$do_mg_l <- as.numeric(sample_data(congo_syn.9)$do_mg_l)

sample_data(congo_syn.9)$wet_pc_uall <- sample_data(congo_syn.9)$wet_pc_u04+sample_data(congo_syn.9)$wet_pc_u05
sample_data(congo_syn.9)$log10_CH4_pct_sat <- log10(1+sample_data(congo_syn.9)$CH4_pct_sat)
sample_data(congo_syn.9)$log10_do_mg_l <- log10(1+sample_data(congo_syn.9)$do_mg_l)

congo_syn.9.ra <- transform_sample_counts(congo_syn.9, function(x) x/sum(x)*100)

glom_phylum_congo.syn.9 <- tax_glom(congo_syn.9.ra, taxrank = "Phylum")
glom_class_congo.syn.9 <- tax_glom(congo_syn.9.ra, taxrank = "Class")
glom_order_congo.syn.9 <- tax_glom(congo_syn.9.ra, taxrank = "Order")

melt.congo.syn.9.phy <- psmelt(glom_phylum_congo.syn.9)
melt.congo.syn.9.cls <- psmelt(glom_class_congo.syn.9)
melt.congo.syn.9.ord <- psmelt(glom_order_congo.syn.9)
# bind
tax.grps.congo.syn.9 <- bind_rows(melt.congo.syn.9.phy, melt.congo.syn.9.cls, melt.congo.syn.9.ord)

tax.grps.congo.syn.9 <- tax.grps.congo.syn.9 %>%
  group_by(Sample, river, Phylum, Class, Order, wet_pc_uall, log10_CH4_pct_sat, log10_do_mg_l) %>%
  summarize(Abundance = sum(Abundance, na.rm=TRUE))



# using phyloseq object
congo_syn.ra = transform_sample_counts(congo_syn, function(x) x/sum(x)*100)

# phylum
glom_phylum_congo_syn<-tax_glom(congo_syn.ra, taxrank='Phylum', NArm=FALSE)
melt_phylum_congo_syn <- psmelt(glom_phylum_congo_syn)

melt_phylum_congo_syn <- melt_phylum_congo_syn %>%
  group_by(ORD_STRA, Phylum) %>%
  mutate(median=median(Abundance))
# select group median > 1 and group rare taxa into < 1% grouping
keep_phy_congo_syn <- unique(melt_phylum_congo_syn$Phylum[melt_phylum_congo_syn$median>1])
melt_phylum_congo_syn$Phylum[!(melt_phylum_congo_syn$Phylum %in% keep_phy_congo_syn)] <- "<1%"
# to get same rows together
melt_phylum_congo_syn_sum <- melt_phylum_congo_syn %>%
  group_by(Sample, river, Phylum, tt_res_hr) %>%
  summarize(Abundance=sum(Abundance))
# make sure tt is numeric
melt_phylum_congo_syn_sum$tt_res_hr <- as.numeric(melt_phylum_congo_syn_sum$tt_res_hr)


```

# indicator species analysis using RA
```{r}
# entire Congo synoptic dataset (n=23 including large rivers) - to examine taxa in outliers (Mopanzi, Lefini) and large rivers (Congo, Sangha, Oubangui)
# phyloseq object is congo_syn (n=23); metadata table is congo_syn_meta
# convert to RA
library(phylosmith)
congo_syn_ra <- relative_abundance(congo_syn, sig_fig=6)
#com_rare_ra <- relative_abundance(community_rare, sig_fig=6)
otu_congo_syn_ra <- otu_table(congo_syn_ra)

otu_congo_syn_ra_t<-as.data.frame(t(otu_congo_syn_ra)) # transpose OTU table, so rows=samples, columns=OTUs
otu_congo_syn_ra_t$sample.number<-row.names(otu_congo_syn_ra_t) # append sample numbers as new column at the end of the OTU table (last column)
end_otu_congo_syn_ra_t<-ncol(otu_congo_syn_ra_t)-1




# set 9: synoptic small streams (n=18)
# use original phyloseq object to eliminate samples from n=23 to n=18
congo_syn.9 <- ps_filter(congo_syn, sample_id!="CONGO_003", sample_id!="CONGO_019", sample_id!="CONGO_017", sample_id!="CONGO_031", sample_id!="CONGO_044")

```

